ccl/changefeedccl: TestChangefeedRandomExpressions failed [test needs adjustment]

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/7997c18f-6e7a-43c9-89c6-b37de47e69ca?testReportRun=22&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [41da9eccc216413c09cc66ad4c6bfa5f92c506e3](https://github.com/cockroachdb/cockroach/commits/41da9eccc216413c09cc66ad4c6bfa5f92c506e3):


```
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT (-1.144455258637275):::FLOAT8 AS " čol_95", ARRAY[1801709019:::OID, 4068909200:::OID, 1256423764:::OID, 1899201244:::OID] AS 😿col_96 FROM d.public.seed@[0] AS tab_26 WHERE EXISTS (SELECT e'\'YxNUGEL\':392B,422B,803A \'edFx\' \'lULBDR\':583C \'nEji\' \'ndrSVaBgL\' \'wPROhg\'':::TSVECTOR AS col_92, seed.tableoid AS col_93, NULL AS col_94 FROM d.public.seed@[0] AS "t'ab_27" ORDER BY seed._interval NULLS FIRST))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT (-1.144455258637275):::FLOAT8 AS " čol_95", ARRAY[1801709019:::OID, 4068909200:::OID, 1256423764:::OID, 1899201244:::OID] AS 😿col_96 FROM d.public.seed@[0] AS tab_26 WHERE EXISTS (SELECT e'\'YxNUGEL\':392B,422B,803A \'edFx\' \'lULBDR\':583C \'nEji\' \'ndrSVaBgL\' \'wPROhg\'':::TSVECTOR AS col_92, seed.tableoid AS col_93, NULL AS col_94 FROM d.public.seed@[0] AS "t'ab_27" ORDER BY seed._interval NULLS FIRST)) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE array_position(ARRAY['1981-04-07':::DATE, '1986-10-06':::DATE, '1990-03-26':::DATE]::DATE[], seed._date::DATE)::INT8::INT8 IN (seed._int8,)
    changefeed_test.go:1120: Skipping predicate array_position(ARRAY['1981-04-07':::DATE, '1986-10-06':::DATE, '1990-03-26':::DATE]::DATE[], seed._date::DATE)::INT8::INT8 IN (seed._int8,) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT '18:27:56.639516':::TIME AS col_136 FROM d.public.seed@[0] AS tab_46)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT '18:27:56.639516':::TIME AS col_136 FROM d.public.seed@[0] AS tab_46) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool OR (seed._string::STRING % seed._string::STRING)::BOOL)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool OR (seed._string::STRING % seed._string::STRING)::BOOL) to emit 76 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true OR EXISTS (SELECT seed._bool AS col̵_192 FROM d.public.seed@[0] AS " tab_66" WHERE true))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true OR EXISTS (SELECT seed._bool AS col̵_192 FROM d.public.seed@[0] AS " tab_66" WHERE true)) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE jsonb_exists_any(seed._jsonb::JSONB, ARRAY['':::STRING, e'S0>2E\x02':::STRING]::STRING[])::BOOL
    changefeed_test.go:1120: Skipping predicate jsonb_exists_any(seed._jsonb::JSONB, ARRAY['':::STRING, e'S0>2E\x02':::STRING]::STRING[])::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (NOT NULL))
    changefeed_test.go:1120: Skipping predicate (seed._bool AND (NOT NULL)) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (NOT (seed._string LIKE seed._string)))
    changefeed_test.go:1120: Skipping predicate (seed._bool AND (NOT (seed._string LIKE seed._string))) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_coveredby(seed._string::STRING, '':::STRING::STRING)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_coveredby(seed._string::STRING, '':::STRING::STRING)::BOOL because error pq: st_coveredby(): error parsing EWKT: syntax error: unexpected NUM at line 1, pos 0
        LINE 1: 1
                ^
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_within(st_multilinefromtext(seed._string::STRING, seed._int4::INT8)::GEOMETRY::GEOMETRY, '01060000000200000001030000000100000004000000000000000000444000000000000044400000000000003440000000000080464000000000008046400000000000003E4000000000000044400000000000004440010300000002000000060000000000000000003440000000000080414000000000000024400000000000003E40000000000000244000000000000024400000000000003E4000000000000014400000000000804640000000000000344000000000000034400000000000804140040000000000000000003E40000000000000344000000000000034400000000000002E40000000000000344000000000000039400000000000003E400000000000003440':::GEOMETRY::GEOMETRY)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_within(st_multilinefromtext(seed._string::STRING, seed._int4::INT8)::GEOMETRY::GEOMETRY, '01060000000200000001030000000100000004000000000000000000444000000000000044400000000000003440000000000080464000000000008046400000000000003E4000000000000044400000000000004440010300000002000000060000000000000000003440000000000080414000000000000024400000000000003E40000000000000244000000000000024400000000000003E4000000000000014400000000000804640000000000000344000000000000034400000000000804140040000000000000000003E40000000000000344000000000000034400000000000002E40000000000000344000000000000039400000000000003E400000000000003440':::GEOMETRY::GEOMETRY)::BOOL because error pq: st_multilinefromtext(): error parsing EWKT: syntax error: unexpected NUM at line 1, pos 0
        LINE 1: 1
                ^
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_coveredby(seed._string::STRING, seed._string::STRING)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_coveredby(seed._string::STRING, seed._string::STRING)::BOOL because error pq: st_coveredby(): error parsing EWKT: syntax error: unexpected NUM at line 1, pos 0
        LINE 1: 1
                ^
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT EXISTS (SELECT 'f89fba60-a327-47e5-b6a3-515540aa3dd9':::UUID AS "c ol_335" FROM d.public.seed@[0] AS tab_122 WHERE seed._bool GROUP BY seed._bool ORDER BY seed._bool NULLS FIRST, seed._bool DESC NULLS FIRST, seed._bool))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT EXISTS (SELECT 'f89fba60-a327-47e5-b6a3-515540aa3dd9':::UUID AS "c ol_335" FROM d.public.seed@[0] AS tab_122 WHERE seed._bool GROUP BY seed._bool ORDER BY seed._bool NULLS FIRST, seed._bool DESC NULLS FIRST, seed._bool)) to emit 54 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL OR false)
    changefeed_test.go:1120: Skipping predicate (NULL OR false) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE inet_contained_by_or_equals('31.140.231.100/14':::INET::INET, seed._inet::INET)::BOOL
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE inet_contained_by_or_equals('31.140.231.100/14':::INET::INET, seed._inet::INET)::BOOL to emit 22 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed.tableoid::OID NOT IN (seed.tableoid, seed.tableoid, oid(seed._string::STRING)::OID, seed.tableoid)
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed.tableoid::OID NOT IN (seed.tableoid, seed.tableoid, oid(seed._string::STRING)::OID, seed.tableoid) because error pq: oid(): invalid input syntax for type oid: "e'\\x00'"
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT _st_equals('01040000400500000001010000402EB8D6E1B429F4C1F0FE0CF00F72D541185D89C9F24CFAC1010100004032B59A08DAECF8C13E7EC6867A82F8415CA3E88CC692E3C10101000040354E57427FFCF2C1CCC99EF6DF2ADDC1D9094EA2A1ABF5C1010100004021851F842FB800C2D242BE50F157F3418042740D5907FC410101000040348B59295879F24160E8BEE12191F14112E7E9E49B95F5C1':::GEOMETRY::GEOMETRY, '010300000000000000':::GEOMETRY::GEOMETRY)::BOOL)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT _st_equals('01040000400500000001010000402EB8D6E1B429F4C1F0FE0CF00F72D541185D89C9F24CFAC1010100004032B59A08DAECF8C13E7EC6867A82F8415CA3E88CC692E3C10101000040354E57427FFCF2C1CCC99EF6DF2ADDC1D9094EA2A1ABF5C1010100004021851F842FB800C2D242BE50F157F3418042740D5907FC410101000040348B59295879F24160E8BEE12191F14112E7E9E49B95F5C1':::GEOMETRY::GEOMETRY, '010300000000000000':::GEOMETRY::GEOMETRY)::BOOL) to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._int2::INT8 # 6932519481858025320:::INT8::INT8)::INT8::INT8 NOT IN ()
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._int2::INT8 # 6932519481858025320:::INT8::INT8)::INT8::INT8 NOT IN () to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND ((e'\'':::STRING ~* seed._string) AND true))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND ((e'\'':::STRING ~* seed._string) AND true)) to emit 7 events
    changefeed_test.go:1141: invalid regular expression: error parsing regexp: missing closing ): ``(?i:(FC]@)``
    --- FAIL: TestChangefeedRandomExpressions/kafka (6.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36522

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/e0d516de-9e96-4a3b-8b9c-87d9fd892cfe?testReportRun=4&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [ebf92d22a7e4afe2423e681d49c615104b5f0098](https://github.com/cockroachdb/cockroach/commits/ebf92d22a7e4afe2423e681d49c615104b5f0098):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (16860:::INT8, (-128):::INT8, (-2147483648):::INT8, 1.5127230882644653:::FLOAT8, (-1.0611824199688036):::FLOAT8, '1986-08-03':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '2020-06-03 09:26:00.000756+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, '-Infinity':::DECIMAL, e'(%\x18':::STRING, '\x22':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[0.22314025158294343, "bar", false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-18027):::INT8, 1:::INT8, 0:::INT8, 1.2257072925567627:::FLOAT8, 0.6897435707190757:::FLOAT8, '4714-11-24 BC':::DATE, '1999-04-30 21:23:47.000023':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-18 years -11 mons -430 days -23:40:13.915916':::INTERVAL, true, 275.3360760432773009:::DECIMAL, NULL, NULL, '00000000-0000-0000-0000-000000000000':::UUID, '25.21.151.15/30':::INET, '[{"Nb2hcFk": null, "a": [0.19384045768817026, true], "bar": null, "baz": 0.46897000199137023}, "b", true, "foobar"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (8820:::INT8, 32767:::INT8, (-128):::INT8, (-1.0):::FLOAT8, '-Inf':::FLOAT8, '1995-02-06':::DATE, '2011-01-29 05:16:33.000838':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '-14 years -8 mons -444 days -07:24:37.641641':::INTERVAL, true, (-1.234E+401):::DECIMAL, NULL, '\x58':::BYTES, '0ad3d2b3-11fa-4a4b-992a-a94e9902ecef':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '["JZN", [1.43315626668601, "a", [null], 1.2664058920360262], ["lpkbu1oo"], 0.7521916342714189]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, 0:::INT8, 127:::INT8, (-0.31100112199783325):::FLOAT8, 1.401298464324817e-45:::FLOAT8, '-infinity':::DATE, '2001-08-13 20:10:09.000198':::TIMESTAMP, '2006-09-14 17:15:43.000393+00':::TIMESTAMPTZ, '63 years 11 mons 475 days 12:18:03.274085':::INTERVAL, true, (-39.3844881252334193):::DECIMAL, e'\'':::STRING, '\x085c6c766527':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '50.69.14.10/2':::INET, '[true, true]':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._string::STRING % e'\x02\x1a\x04W6P':::STRING::STRING)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._string::STRING % e'\x02\x1a\x04W6P':::STRING::STRING)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT B'1' AS col_3 FROM d.public.seed@[0] AS " tab_4" ORDER BY seed._enum ASC NULLS FIRST, seed._uuid DESC NULLS LAST, seed._int8 ASC NULLS LAST)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT B'1' AS col_3 FROM d.public.seed@[0] AS " tab_4" ORDER BY seed._enum ASC NULLS FIRST, seed._uuid DESC NULLS LAST, seed._int8 ASC NULLS LAST) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '18:45:15.558897+06:36':::TIMETZ::TIMETZ IN (SELECT (('23:58:49.603199':::TIME::TIME - (NULL::INTERVAL + NULL::TIME)::TIME::TIME)::INTERVAL::INTERVAL + '03:53:46.002561+09:17':::TIMETZ::TIMETZ)::TIMETZ::TIMETZ AS "c""ol-_9" FROM d.public.seed@[0] AS tab_6 GROUP BY seed._jsonb, seed._float8, seed.crdb_internal_mvcc_timestamp, seed.tableoid, seed._date HAVING seed._bool ORDER BY seed.crdb_internal_mvcc_timestamp NULLS FIRST)
    changefeed_test.go:1120: Skipping predicate '18:45:15.558897+06:36':::TIMETZ::TIMETZ IN (SELECT (('23:58:49.603199':::TIME::TIME - (NULL::INTERVAL + NULL::TIME)::TIME::TIME)::INTERVAL::INTERVAL + '03:53:46.002561+09:17':::TIMETZ::TIMETZ)::TIMETZ::TIMETZ AS "c""ol-_9" FROM d.public.seed@[0] AS tab_6 GROUP BY seed._jsonb, seed._float8, seed.crdb_internal_mvcc_timestamp, seed.tableoid, seed._date HAVING seed._bool ORDER BY seed.crdb_internal_mvcc_timestamp NULLS FIRST) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE parse_time('$':::STRING::STRING, e'"\x1diq\x16':::STRING::STRING)::TIME::TIME IN (SELECT '24:00:00':::TIME::TIME AS col_28 FROM d.public.seed@[0] AS "\\u632Ctab�_11")
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE parse_time('$':::STRING::STRING, e'"\x1diq\x16':::STRING::STRING)::TIME::TIME IN (SELECT '24:00:00':::TIME::TIME AS col_28 FROM d.public.seed@[0] AS "\\u632Ctab�_11") because error pq: parse_time(): unknown DateStyle parameter: "�iq
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._inet::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT false)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT false) to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET << seed._inet::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._inet::INET << seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE isnan(2.021500261309974:::FLOAT8::FLOAT8)::BOOL
    changefeed_test.go:1120: Skipping predicate isnan(2.021500261309974:::FLOAT8::FLOAT8)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 48 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> 'ee0b:b314:994b:6d8d:bed0:c8fe:f579:d961/84':::INET::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._inet::INET >> 'ee0b:b314:994b:6d8d:bed0:c8fe:f579:d961/84':::INET::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT ARRAY['BOX(-10 -10,10 10)':::BOX2D, 'BOX(-0.8767145881798835 -2.178936479131635,-0.21987745699864836 0.6946927902683193)':::BOX2D, 'BOX(-1.0969258426590665 -0.12337920815230818,-0.4964131646405418 0.2750661940514598)':::BOX2D] AS "col
        _169", seed._timestamp AS "😙col _170", seed._float8 AS col_171 FROM d.public.seed@[0] AS t😄ab_52 ORDER BY seed._inet ASC NULLS LAST, seed._timestamp ASC)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT ARRAY['BOX(-10 -10,10 10)':::BOX2D, 'BOX(-0.8767145881798835 -2.178936479131635,-0.21987745699864836 0.6946927902683193)':::BOX2D, 'BOX(-1.0969258426590665 -0.12337920815230818,-0.4964131646405418 0.2750661940514598)':::BOX2D] AS "col
        _169", seed._timestamp AS "😙col _170", seed._float8 AS col_171 FROM d.public.seed@[0] AS t😄ab_52 ORDER BY seed._inet ASC NULLS LAST, seed._timestamp ASC) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '00:00:00':::TIME::TIME NOT IN ('08:54:08.951744':::TIME, '21:40:19.000964':::TIME, '02:27:55.646337':::TIME)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE '00:00:00':::TIME::TIME NOT IN ('08:54:08.951744':::TIME, '21:40:19.000964':::TIME, '02:27:55.646337':::TIME) to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET << (seed._int2::INT8 + seed._inet::INET)::INET::INET)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET << (seed._int2::INT8 + seed._inet::INET)::INET::INET)::BOOL because error pq: result out of range
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '1974-12-09':::DATE::DATE IN (SELECT seed._date::DATE AS "c ol_247" FROM d.public.seed@[0] AS " tab_65" JOIN d.public.seed@[0] AS "t%p😶ab_66" ON (" tab_65"._interval) = ("t%p😶ab_66"._interval) WHERE NULL ORDER BY seed._jsonb, seed._int4 ASC NULLS FIRST)
    changefeed_test.go:1120: Skipping predicate '1974-12-09':::DATE::DATE IN (SELECT seed._date::DATE AS "c ol_247" FROM d.public.seed@[0] AS " tab_65" JOIN d.public.seed@[0] AS "t%p😶ab_66" ON (" tab_65"._interval) = ("t%p😶ab_66"._interval) WHERE NULL ORDER BY seed._jsonb, seed._int4 ASC NULLS FIRST) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._enum AS c😆ol_261, B'001111011000110000' AS col_262, seed._enum AS col_263 FROM d.public.seed@[0] AS "t|a%vb_69" ORDER BY seed._float4 NULLS FIRST, seed._inet)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._enum AS c😆ol_261, B'001111011000110000' AS col_262, seed._enum AS col_263 FROM d.public.seed@[0] AS "t|a%vb_69" ORDER BY seed._float4 NULLS FIRST, seed._inet) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._float4::FLOAT8 NOT IN (ceil((seed._date::DATE - seed._date::DATE)::INT8::INT8)::FLOAT8, 0.2850700589982581:::FLOAT8)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._float4::FLOAT8 NOT IN (ceil((seed._date::DATE - seed._date::DATE)::INT8::INT8)::FLOAT8, 0.2850700589982581:::FLOAT8) to emit 73 events
    changefeed_test.go:1141: cannot subtract infinite dates
    --- FAIL: TestChangefeedRandomExpressions/kafka (2.87s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/93e0e2c6-1e1b-4480-b257-144ddc8ca767?testReportRun=25&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [7488e090daa588c4d7c0f828c8006bb9b13a90f6](https://github.com/cockroachdb/cockroach/commits/7488e090daa588c4d7c0f828c8006bb9b13a90f6):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 676680355:::INT8, (-4031933384664920236):::INT8, 1.401298464324817e-45:::FLOAT8, NULL, '1990-02-25':::DATE, '1978-10-07 13:54:11.000845':::TIMESTAMP, '2005-02-14 04:31:45.000673+00':::TIMESTAMPTZ, '290 years':::INTERVAL, true, (-1.234E+401):::DECIMAL, e'\x00':::STRING, '\xff':::BYTES, '2b753c7e-eeef-4a59-800a-daaf647e61e2':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-23220):::INT8, 0:::INT8, 0:::INT8, (-0.15930914878845215):::FLOAT8, 0.08349584512906827:::FLOAT8, '1976-01-14':::DATE, '2017-01-25 02:00:24.00052':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '51 years 6 mons 482 days 04:44:49.515295':::INTERVAL, NULL, (-5.323318726735347158E+27):::DECIMAL, '':::STRING, '\x27':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '172.122.91.210/21':::INET, '{"0biJ7": null, "a": [null, {"BuFWxO": "SNCSCJ6Gl"}, true, null], "b": "a", "baz": [true]}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-816219628):::INT8, NULL, 0.0:::FLOAT8, 0.4190266845288343:::FLOAT8, '5874897-12-31':::DATE, '2033-03-03 23:33:05.000957':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, (-1):::DECIMAL, '':::STRING, '\x00':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'da4a:3927:390e:a408:e53e:4588:b65c:beb4/46':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-20481):::INT8, 32767:::INT8, (-128):::INT8, (-0.0):::FLOAT8, 0.4035411916281586:::FLOAT8, '1992-02-01':::DATE, '2011-03-07 17:39:09.000717':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 1:::DECIMAL, '"':::STRING, '\xe29883':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '249.68.39.168/29':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (4755:::INT8, 1:::INT8, (-9223372036854775807):::INT8, (-0.30003392696380615):::FLOAT8, 1.3572994763386133:::FLOAT8, '1988-05-26':::DATE, '2004-12-30 13:32:06.000408':::TIMESTAMP, '1972-04-18 23:28:54.000702+00':::TIMESTAMPTZ, '25 years 11 mons 632 days 17:48:07.439408':::INTERVAL, true, (-26395225889554869.30):::DECIMAL, '"':::STRING, '\x00':::BYTES, '9dbc4454-d640-4fda-bf04-c5b7e9ddc87b':::UUID, '109.140.57.105/27':::INET, '[{"6hi": "Nc33F"}, [true, 0.9925963070107657], null, false, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-26721):::INT8, 2002576534:::INT8, 2147483647:::INT8, (-0.13262015581130981):::FLOAT8, (-0.21575581262294385):::FLOAT8, '1970-01-01':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '1974-06-29 17:11:04.000802+00':::TIMESTAMPTZ, '6 years 2 mons 826 days 02:02:30.685173':::INTERVAL, NULL, (-467539981424229.2609):::DECIMAL, 'X':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[["rTbNG"], [true, 0.37288362421829224], "foobar", 0.7399271467914521]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-605857701):::INT8, 32767:::INT8, (-0.15109345316886902):::FLOAT8, (-0.0):::FLOAT8, '-infinity':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '-13 years -10 mons -443 days -24:44:52.323621':::INTERVAL, false, 7.069780961316660012:::DECIMAL, e'\x00':::STRING, '\x116d5e2db4411e':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-492333230):::INT8, 3552657091767350682:::INT8, 'NaN':::FLOAT8, 'NaN':::FLOAT8, '5874897-12-31':::DATE, '2011-12-12 04:14:47.000363':::TIMESTAMP, '2030-11-06 03:49:36.000016+00':::TIMESTAMPTZ, '55 years 11 mons 621 days 13:19:30.104437':::INTERVAL, false, 0:::DECIMAL, e'\'':::STRING, '\x9e513b7925026a61':::BYTES, 'b3741c6b-9acd-49d3-a977-20ef9d338e5c':::UUID, '184.177.207.181/19':::INET, '{"7DRIwP46Z": {"baz": null}, "D2": [["c"], "b"], "DorD4g3": "b", "b": 5.369285270711828, "c": false, "n3bpD9k": [{"a": [false, true], "mjrgK": null}, 1.3986674114483741, "q9S8SF", null]}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (32767:::INT8, 2147483647:::INT8, 127:::INT8, (-0.2683912515640259):::FLOAT8, 1.002009446530752:::FLOAT8, '1981-12-01':::DATE, '2010-08-18 07:26:00.000934':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '290 years':::INTERVAL, false, '-Infinity':::DECIMAL, e'\'':::STRING, '\xcdb8432e3f':::BYTES, 'd352b574-cbdc-4811-bab6-4f90c491e91f':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, (-128):::INT8, (-4608631781319849820):::INT8, 'NaN':::FLOAT8, 0.43978724257188095:::FLOAT8, '1980-08-26':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2032-06-13 13:04:23.000161+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, 8.21690870724977903E+18:::DECIMAL, '':::STRING, '\xfa07d2':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-20849):::INT8, 1690813959:::INT8, (-2585213655154962621):::INT8, (-1.0):::FLOAT8, '-Inf':::FLOAT8, '1973-11-08':::DATE, '1995-06-15 21:39:39.000444':::TIMESTAMP, NULL, '-27 years -9 mons -894 days -08:23:48.316451':::INTERVAL, NULL, 0:::DECIMAL, NULL, '\x':::BYTES, '80cc65fd-301f-4abc-88fa-03653a2c7ef1':::UUID, 'ea6:9c3c:118f:6bb4:dd21:ca14:5fae:ce70/69':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (29917:::INT8, 1795601356:::INT8, (-128):::INT8, '+Inf':::FLOAT8, '+Inf':::FLOAT8, '-infinity':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2015-12-24 01:57:00.000507+00':::TIMESTAMPTZ, NULL, true, 0:::DECIMAL, e'\U00002603':::STRING, '\xe29883':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, NULL, 32767:::INT8, '-Inf':::FLOAT8, 1.176837990027004:::FLOAT8, '-infinity':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2012-12-26 11:35:43.000998+00':::TIMESTAMPTZ, '290 years':::INTERVAL, true, (-852024.2276101646846):::DECIMAL, 'iMF':::STRING, '\x31556580':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '175e:ba11:4181:4176:e6c2:2ff:c2e8:2f4d/113':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 1102513335:::INT8, (-9223372036854775807):::INT8, '-Inf':::FLOAT8, 'NaN':::FLOAT8, '1981-01-04':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-42 years -5 mons -257 days -14:24:05.549692':::INTERVAL, NULL, 1.0:::DECIMAL, e'\x1d\'\rU':::STRING, '\xff':::BYTES, '011dee23-0cb0-4348-8b2e-c47ae01da8a5':::UUID, '64.125.235.50/23':::INET, '{"b": null, "foobar": 0.6505975379537954}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (20561:::INT8, 720498638:::INT8, (-2844575063548101673):::INT8, (-1.0):::FLOAT8, 0.940682099785647:::FLOAT8, '1985-08-18':::DATE, '2021-12-31 09:00:34.000395':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '290 years':::INTERVAL, false, '-Infinity':::DECIMAL, 'C':::STRING, '\x00':::BYTES, 'a4893e1a-7681-45dd-a9a8-e6eaf42ac9eb':::UUID, 'cc2f:c4fe:ad00:81b0:f4c0:3bd8:8be3:279e/47':::INET, '[[[[2.2559057285609843], false, true], 0.14033627743709254, "Nc"], "bar", "c"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-15670):::INT8, 0:::INT8, (-8621195337977681273):::INT8, '-Inf':::FLOAT8, 0.8605217508300289:::FLOAT8, NULL, '1990-02-04 00:03:39.000079':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-6.69422383027121623E+32):::DECIMAL, e'\'':::STRING, '\xace2fced69ad2e75c6':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '255.119.2.54/10':::INET, '[[{"037Wce": "b", "baz": false}, 0.5658007166688831], ["bjEJhlyf5", false, null], null]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (23406:::INT8, 720406777:::INT8, (-128):::INT8, 0.0:::FLOAT8, (-1.0):::FLOAT8, '1972-03-09':::DATE, '1997-04-14 21:14:42.000876':::TIMESTAMP, '2007-05-13 11:28:29.000443+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 1.0:::DECIMAL, '"':::STRING, '\x58':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '147.203.12.113/16':::INET, '{}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 49 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 45 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_covers(st_geogfromtext(e'\x1a-R)fJ$':::STRING::STRING)::GEOGRAPHY::GEOGRAPHY, '01060000A0E61000000600000001030000800100000009000000E49958BBD1485CC088F752D52CE450C0943B9C65747100425C034D5C8B0B58C094E10969FC4650C0847A29656E6CEA4148AB2FBE7E994A40E009711A510F51C06C4C0B7FAE49DEC194D3CF4CEAC66140F2AD91BDE82E54C0F038D7B14D86E3C1CA03BF7DA2606640D0E8457995E827404010BF50E305FB41785B24EFBC184440208EFECD716A1640B4DD6D48F1AF0142B0464CEB297A2BC064C93EDB59A33740B072A6653F49D1C16F8AEE682E7D56C07C0EEA7C85634840500061079C9FECC1E49958BBD1485CC088F752D52CE450C0943B9C657471004201030000800100000008000000041403C9D60A5FC0F808E480F2993A40001065B0E31E6E41F34F9D7E891857C0BC56BDA1041533404045B9D97466C5410495790025324A4040ACDD7D7F3609C0C962DCE3D7B4F8C1F2DF573596E06340C80E36AC7C983540300A8A7A01BCD5C198873A950EE53B40D66380D20D7541406011D044F9A5C9C18456E0F9843059409E8271FE997355405498F39075BEEF416ABBF60DD9B14BC026AA756A5CA442400E171CB41E8AF341041403C9D60A5FC0F808E480F2993A40001065B0E31E6E410103000080010000000C00000069648106688751C009B54E44792153C04674988B1768F3C1E03C2D5E89B02DC0F0F2F105488437C0CC13413FF565FB4160CE61412B671F408898AE03BDA83EC0D44456DA2146F54100B1554996911D40004ED70AD12925400A16C22C8F6D0242F004FA85F0083C4078860E9A7BAE31C0CB2AD3D0CD8AF1C1A4B48B77A35A5540690A2C83696E49C0F2548FA7FE8F014232C79B3CC4806040226AC6C58E405540E0F7DFD81091ED419062D0A978114D40922969460C555040A2837E102FC4E3C1B86D56058AB94C401434969776695140D849686D13E1FC41E168618CD5C250C08844C279B7F24F40700DC4EE12DCD341C6C34CDF7AFF61C088C44213DF3847409895214F3F4EEF4169648106688751C009B54E44792153C04674988B1768F3C10103000080010000000600000029B203081C5B66C0E1DBD06CBC384EC0F60FEF6502ED0142045FECE1C9E04440FC8C948DAB7238C0F5ADB38FE152F0C1DCC9A6EDCD225A40E025AF79F6E42B4018A13063D284F3C162B54F7165E56040E0905C3BE589334030469D91EC75D241D29A668BF2D752C02C85E796AAD34040D08BAB9E9132CBC129B203081C5B66C0E1DBD06CBC384EC0F60FEF6502ED01420103000080010000000900000033325D1CAA1266C0F1AB5848438A55C0BA725432E3A3F4C1302BAFCE73CC5540D19EDE3BD47F54C09602393A084EF8411C3A518B253460407C3FA781692256C000302BAA91A5AC419E093F687E826440315F43AC2A0655C0EAF8FAE1AFA0FBC17826AFDB0F415C406C67E68E936F5440C83F9ECD6966E5C148EBE8A04FE5534076803618F13B50403C28B567A23700C280A06F953BDD21C0805B30B84672E4BF3638CE69DBD2F8C10044BE10535D05C0E83DE13571462FC080162A2422A19A4133325D1CAA1266C0F1AB5848438A55C0BA725432E3A3F4C1010300008001000000090000001D3D64F93CBF62C0BC63C1926EDE44C0F6B610BED81E0042661557AC52DC65C02D761F51B9074AC09065E3ACB1D7D4C1C0231AC04F0342409E9AB1F0992542C0B451C2A22F2EFB41E03D4082C9095E40813AE71F787454C05FD446922E51FCC150B7A8DBBC963740186F6256AFA72940F06FA3CCCCB200C28C69EABFABEF534010DEA37B4A174B408E5714377B0902420C4F235A02ED404020D95D31D10E4C40D9B4FBFB7B42F4C1E6CA7761B57466C084B72935D98D5140425843B9CCB6F9C11D3D64F93CBF62C0BC63C1926EDE44C0F6B610BED81E0042':::GEOGRAPHY::GEOGRAPHY)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_covers(st_geogfromtext(e'\x1a-R)fJ$':::STRING::STRING)::GEOGRAPHY::GEOGRAPHY, '01060000A0E61000000600000001030000800100000009000000E49958BBD1485CC088F752D52CE450C0943B9C65747100425C034D5C8B0B58C094E10969FC4650C0847A29656E6CEA4148AB2FBE7E994A40E009711A510F51C06C4C0B7FAE49DEC194D3CF4CEAC66140F2AD91BDE82E54C0F038D7B14D86E3C1CA03BF7DA2606640D0E8457995E827404010BF50E305FB41785B24EFBC184440208EFECD716A1640B4DD6D48F1AF0142B0464CEB297A2BC064C93EDB59A33740B072A6653F49D1C16F8AEE682E7D56C07C0EEA7C85634840500061079C9FECC1E49958BBD1485CC088F752D52CE450C0943B9C657471004201030000800100000008000000041403C9D60A5FC0F808E480F2993A40001065B0E31E6E41F34F9D7E891857C0BC56BDA1041533404045B9D97466C5410495790025324A4040ACDD7D7F3609C0C962DCE3D7B4F8C1F2DF573596E06340C80E36AC7C983540300A8A7A01BCD5C198873A950EE53B40D66380D20D7541406011D044F9A5C9C18456E0F9843059409E8271FE997355405498F39075BEEF416ABBF60DD9B14BC026AA756A5CA442400E171CB41E8AF341041403C9D60A5FC0F808E480F2993A40001065B0E31E6E410103000080010000000C00000069648106688751C009B54E44792153C04674988B1768F3C1E03C2D5E89B02DC0F0F2F105488437C0CC13413FF565FB4160CE61412B671F408898AE03BDA83EC0D44456DA2146F54100B1554996911D40004ED70AD12925400A16C22C8F6D0242F004FA85F0083C4078860E9A7BAE31C0CB2AD3D0CD8AF1C1A4B48B77A35A5540690A2C83696E49C0F2548FA7FE8F014232C79B3CC4806040226AC6C58E405540E0F7DFD81091ED419062D0A978114D40922969460C555040A2837E102FC4E3C1B86D56058AB94C401434969776695140D849686D13E1FC41E168618CD5C250C08844C279B7F24F40700DC4EE12DCD341C6C34CDF7AFF61C088C44213DF3847409895214F3F4EEF4169648106688751C009B54E44792153C04674988B1768F3C10103000080010000000600000029B203081C5B66C0E1DBD06CBC384EC0F60FEF6502ED0142045FECE1C9E04440FC8C948DAB7238C0F5ADB38FE152F0C1DCC9A6EDCD225A40E025AF79F6E42B4018A13063D284F3C162B54F7165E56040E0905C3BE589334030469D91EC75D241D29A668BF2D752C02C85E796AAD34040D08BAB9E9132CBC129B203081C5B66C0E1DBD06CBC384EC0F60FEF6502ED01420103000080010000000900000033325D1CAA1266C0F1AB5848438A55C0BA725432E3A3F4C1302BAFCE73CC5540D19EDE3BD47F54C09602393A084EF8411C3A518B253460407C3FA781692256C000302BAA91A5AC419E093F687E826440315F43AC2A0655C0EAF8FAE1AFA0FBC17826AFDB0F415C406C67E68E936F5440C83F9ECD6966E5C148EBE8A04FE5534076803618F13B50403C28B567A23700C280A06F953BDD21C0805B30B84672E4BF3638CE69DBD2F8C10044BE10535D05C0E83DE13571462FC080162A2422A19A4133325D1CAA1266C0F1AB5848438A55C0BA725432E3A3F4C1010300008001000000090000001D3D64F93CBF62C0BC63C1926EDE44C0F6B610BED81E0042661557AC52DC65C02D761F51B9074AC09065E3ACB1D7D4C1C0231AC04F0342409E9AB1F0992542C0B451C2A22F2EFB41E03D4082C9095E40813AE71F787454C05FD446922E51FCC150B7A8DBBC963740186F6256AFA72940F06FA3CCCCB200C28C69EABFABEF534010DEA37B4A174B408E5714377B0902420C4F235A02ED404020D95D31D10E4C40D9B4FBFB7B42F4C1E6CA7761B57466C084B72935D98D5140425843B9CCB6F9C11D3D64F93CBF62C0BC63C1926EDE44C0F6B610BED81E0042':::GEOGRAPHY::GEOGRAPHY)::BOOL because error pq: st_geogfromtext(): error parsing EWKT: syntax error: invalid character at line 1, pos 0
        LINE 1: �-R)fJ$
                ^
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE not_similar_to_escape(seed._string::STRING, seed._string::STRING, (COALESCE('':::STRING, '':::STRING)::STRING || seed._timestamp::TIMESTAMP)::STRING::STRING)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE not_similar_to_escape(seed._string::STRING, seed._string::STRING, (COALESCE('':::STRING, '':::STRING)::STRING || seed._timestamp::TIMESTAMP)::STRING::STRING)::BOOL because error pq: not_similar_to_escape(): invalid escape string
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (false AND (';':::STRING::STRING % seed._string::STRING)::BOOL))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT (false AND (';':::STRING::STRING % seed._string::STRING)::BOOL)) to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE json_valid(seed._string::STRING)::BOOL
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE json_valid(seed._string::STRING)::BOOL to emit 5 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL::STRING % e'\x02Q\x0f}\x1e?A:':::STRING::STRING)::BOOL
    changefeed_test.go:1120: Skipping predicate (NULL::STRING % e'\x02Q\x0f}\x1e?A:':::STRING::STRING)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE bool(seed.crdb_internal_mvcc_timestamp::DECIMAL)::BOOL
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE bool(seed.crdb_internal_mvcc_timestamp::DECIMAL)::BOOL to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._float4::FLOAT8 NOT IN ((-0.28991876627247354):::FLOAT8,)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._float4::FLOAT8 NOT IN ((-0.28991876627247354):::FLOAT8,) to emit 91 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._bool OR not_ilike_escape('2qP':::STRING::STRING, 'yv':::STRING::STRING, seed._string::STRING)::BOOL))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._bool OR not_ilike_escape('2qP':::STRING::STRING, 'yv':::STRING::STRING, seed._string::STRING)::BOOL)) to emit 49 events
    changefeed_test.go:1141: not_ilike_escape(): invalid escape string
    --- FAIL: TestChangefeedRandomExpressions/kafka (3.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=25</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/d9d2197a-8684-408b-aa5a-ee7f91757517?testReportRun=10&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [a3ac7ebf958f25201c2696a17f996c0b9f86830f](https://github.com/cockroachdb/cockroach/commits/a3ac7ebf958f25201c2696a17f996c0b9f86830f):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 127:::INT8, 543258836076629686:::INT8, 0.5528786778450012:::FLOAT8, '+Inf':::FLOAT8, NULL, '4714-11-24 00:00:00 BC':::TIMESTAMP, '2032-03-05 22:34:21.000227+00':::TIMESTAMPTZ, '74 years 7 mons 570 days 01:01:01.439523':::INTERVAL, true, 'NaN':::DECIMAL, e'cuq2\b\f\x05OD':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 1215762550:::INT8, 9177021824219827915:::INT8, 1.7296892404556274:::FLOAT8, (-0.7744041736933392):::FLOAT8, 'infinity':::DATE, '2027-08-28 07:07:27.00075':::TIMESTAMP, '2030-08-05 18:52:21.000415+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 8807562578179228.556:::DECIMAL, '':::STRING, '\x22':::BYTES, 'b1b5465f-3909-4fd4-9547-1cd5313265a5':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 709596827:::INT8, (-32768):::INT8, (-1.1331731081008911):::FLOAT8, NULL, '1970-01-01':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, NULL, '-3 years -3 mons -423 days -02:46:42.219054':::INTERVAL, false, 6.60866113902190526E+31:::DECIMAL, e'\x12\x06[|':::STRING, '\xe29883':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '103.54.86.19/14':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 127:::INT8, 2872712967898812524:::INT8, '-Inf':::FLOAT8, 1.0:::FLOAT8, 'infinity':::DATE, '1971-11-08 02:55:31.000222':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '-9 years -2 mons -186 days -13:15:08.618455':::INTERVAL, true, (-448478503.1793896464):::DECIMAL, e'\x1a.\\eK]f\x18':::STRING, '\xff':::BYTES, '5268b991-58da-4036-8862-2a2c1888abef':::UUID, '0.0.0.0/0':::INET, '[{"PBNKPv": {"c": null, "foo": false}, "Xyg": [false, false], "a": false, "b": [null, false, false], "cWJ": [true, false], "fK": null, "foobar": false}, null, "YEX", 6.295569997700887]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-1):::INT8, 8839684014644661496:::INT8, '+Inf':::FLOAT8, 0.359943889466381:::FLOAT8, '1970-01-01':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2003-03-21 15:54:17.000808+00':::TIMESTAMPTZ, '-4 years -6 mons -396 days -21:35:36.246442':::INTERVAL, false, 75970289553640838.61:::DECIMAL, e'F\x16Y\x10]&ro':::STRING, '\x4fc78d75':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '46.22.210.146/19':::INET, '[[[2.6788755250173844, false], ["baz"], true], {"C1bI": [true], "a": 1.2676163644819658, "baz": "baz"}, {"dshea": 4.921680751513673, "sKPVhkW": false}, 0.6780459944272341, 2.5886665030881426]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, 2147483647:::INT8, 9223372036854775807:::INT8, 0.9585562348365784:::FLOAT8, '-Inf':::FLOAT8, '1977-02-26':::DATE, NULL, '1975-03-18 13:43:28.000707+00':::TIMESTAMPTZ, '-45 years -11 mons -194 days -06:43:42.476118':::INTERVAL, true, (-67766226.33957327621):::DECIMAL, e'\'':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '200.104.10.224/18':::INET, '{"0DkvuYo20": 0.9728463034981589, "b": [[0.054032935896483725], [null], null, "VbhM66TxW"]}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-8061):::INT8, (-128):::INT8, NULL, (-1.0):::FLOAT8, (-0.07335014456521805):::FLOAT8, '-infinity':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '1995-11-19 04:10:22.00075+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 'Infinity':::DECIMAL, e'g~\x1ba\x0430\x10@':::STRING, NULL, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '113.189.246.217/15':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, (-1):::INT8, (-128):::INT8, '-Inf':::FLOAT8, 5e-324:::FLOAT8, '4714-11-24 BC':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2016-10-23 05:25:52.000366+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-6.140424774120247585E+28):::DECIMAL, e'\x00':::STRING, '\x00':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'bb69:ed56:7487:ad1:7307:85ce:e0c:550f/91':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1076:::INT8, (-128):::INT8, (-8610397860526808979):::INT8, (-1.0):::FLOAT8, NULL, '1990-07-30':::DATE, '2014-04-06 18:47:59.000473':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, NULL, '"':::STRING, '\x00':::BYTES, NULL, '0.0.0.0/0':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (14755:::INT8, (-1):::INT8, 2541896873967215883:::INT8, (-0.14304833114147186):::FLOAT8, 1.0:::FLOAT8, '1980-04-16':::DATE, '1992-10-17 05:37:05.000426':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 5.75018421267186944E+36:::DECIMAL, '':::STRING, '\x':::BYTES, '99c48e16-bfec-4c4e-9463-efe3a99aa1fa':::UUID, 'e9fe:28de:e6e0:f00f:329b:e172:264a:1abc/110':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 0:::INT8, 32767:::INT8, 0.01827017031610012:::FLOAT8, (-0.003746371826499284):::FLOAT8, '-infinity':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, NULL, '290 years':::INTERVAL, true, NULL, e'\'':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-26223):::INT8, 127:::INT8, (-1):::INT8, 1.401298464324817e-45:::FLOAT8, (-1.0):::FLOAT8, '-infinity':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '67 years 4 mons 610 days 03:23:53.31009':::INTERVAL, false, NULL, '"':::STRING, '\xff':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ad94:1819:7bf4:19e5:5054:d5b4:37ac:99b1/9':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-30898):::INT8, (-1992666849):::INT8, 0:::INT8, 0.08901289105415344:::FLOAT8, 0.25519045645888166:::FLOAT8, '4714-11-24 BC':::DATE, '2018-02-16 15:55:02.000248':::TIMESTAMP, '1995-12-03 22:56:58.00006+00':::TIMESTAMPTZ, NULL, true, (-2.300502348697162075E+19):::DECIMAL, '4s0OU':::STRING, '\x7e543d402bd1':::BYTES, 'ad07ee93-1769-4d64-a79c-7e2c1b30435a':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (14412:::INT8, 32767:::INT8, (-1):::INT8, '+Inf':::FLOAT8, 0.13396311553306312:::FLOAT8, '1995-09-23':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2023-09-13 12:18:48.000501+00':::TIMESTAMPTZ, '1 day':::INTERVAL, true, 0.6514830960202630065:::DECIMAL, e'\x02t]fO,\x18':::STRING, '\xbde3e2f3a95900':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (26658:::INT8, (-857493741):::INT8, 32767:::INT8, 1.0:::FLOAT8, 'NaN':::FLOAT8, '1990-11-05':::DATE, NULL, '1980-07-21 09:44:18.000095+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, NULL, 4.380469126634882964E+21:::DECIMAL, '@Azk':::STRING, '\x27':::BYTES, '09e90659-19c2-43f7-94ea-36d1531b9144':::UUID, '0.0.0.0/0':::INET, '{"q0": 0.7018179414544257}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (27008:::INT8, NULL, 32767:::INT8, 0.0:::FLOAT8, '+Inf':::FLOAT8, '4714-11-24 BC':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '1981-02-01 07:24:33.000713+00':::TIMESTAMPTZ, '290 years':::INTERVAL, true, 48655435457279822.4:::DECIMAL, 'SAxp;q:c':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '91.29.117.72/0':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 1420234277:::INT8, (-9223372036854775807):::INT8, 0.8680716753005981:::FLOAT8, 0.0036660931773788707:::FLOAT8, '-infinity':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '1989-05-13 01:37:53.000081+00':::TIMESTAMPTZ, '71 years 9 mons 684 days 24:12:52.994946':::INTERVAL, NULL, 8.541972843276900168E+24:::DECIMAL, '':::STRING, '\x58':::BYTES, '1d2a7c48-c1a1-42c1-9ce6-1b6167f2e0dd':::UUID, '62.39.131.18/17':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, NULL, (-5327639608747399426):::INT8, 0.1961871236562729:::FLOAT8, 1.0:::FLOAT8, '4714-11-24 BC':::DATE, '2023-12-21 05:18:01.000547':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '81 years 8 mons 979 days 10:40:42.414984':::INTERVAL, false, 1.0:::DECIMAL, '':::STRING, '\x7abba98a017a024bb9':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 1548666133:::INT8, 2147483647:::INT8, 3.4028234663852886e+38:::FLOAT8, (-1.3614223697969212):::FLOAT8, 'infinity':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-73718171085585.54507):::DECIMAL, '"':::STRING, '\x58':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{"4hnVgd": true, "c": null, "mIEYiD0": {"5R": false}, "xUW8": {"foo": true}}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (18432:::INT8, (-1):::INT8, 1449841412961451611:::INT8, 'NaN':::FLOAT8, (-0.0):::FLOAT8, '1993-06-05':::DATE, '1982-06-20 11:46:56.000661':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '-63 years -6 mons -233 days -09:01:30.727304':::INTERVAL, false, (-925057374526947.748):::DECIMAL, e'\U00002603':::STRING, '\x':::BYTES, '9b59fda1-45b7-4b8e-9d3b-29431bc5f7d1':::UUID, 'f38a:c192:9c78:2d4e:3ef7:f6c6:7db3:aa18/83':::INET, '{"a": false, "baz": true, "h1u": {"O3H": 0.5310180987498656, "bar": 0.6618234414791979}, "hgYqM": "DAG"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (26122:::INT8, (-32768):::INT8, (-6780562896921611608):::INT8, (-0.9538408517837524):::FLOAT8, 5e-324:::FLOAT8, '1975-01-12':::DATE, '2018-05-01 08:01:21.000664':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '-15 years -674 days -00:33:23.453818':::INTERVAL, true, 1.822518732700908205E+35:::DECIMAL, '':::STRING, '\x58':::BYTES, '85e804ca-d53e-40e4-985f-77a20b673b53':::UUID, '198.37.142.142/18':::INET, '[{"CpvVdoP": null, "JO6": 0.4857087540602436, "bar": null, "c": "2Zm"}, 1.050328540023902]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (8883:::INT8, (-951429048):::INT8, 2147483647:::INT8, (-0.8565292358398438):::FLOAT8, (-0.5664727216075702):::FLOAT8, 'infinity':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-2.342328566448825962):::DECIMAL, '$NF':::STRING, NULL, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '490a:3333:2620:1fee:5431:871:77fd:3643/37':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-31575):::INT8, 1776809612:::INT8, (-32768):::INT8, 0.0:::FLOAT8, '-Inf':::FLOAT8, '1987-09-17':::DATE, '2006-09-06 11:39:31.000591':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '-58 years -8 mons -564 days -07:14:35.568627':::INTERVAL, false, NULL, e'\x00':::STRING, '\x2db0':::BYTES, '5bbf7d95-38f0-4b24-8b2e-9b2aa2e836ba':::UUID, '0.0.0.0/0':::INET, '["a", {"a": 1.327427994323051, "foo": "CjwJp", "jhW7V65Y": {"a": 5.417870079381686, "b": 1.093959348891574}}, {"a": null}, ["foobar"], true, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-1126942374):::INT8, (-6022005077904107784):::INT8, 0.3832065463066101:::FLOAT8, 5e-324:::FLOAT8, '1989-02-16':::DATE, '2017-09-21 23:02:30.000519':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '9 years 1 mon 809 days 13:53:52.818404':::INTERVAL, true, (-1):::DECIMAL, e'``QfT\x17\x11|T$':::STRING, '\xc0c26b8bd491':::BYTES, '4036f3c7-877e-4beb-b02c-12bac0811a0a':::UUID, '2aa0:af8:f0d2:9b26:9ee2:7d9f:97f3:39ed/24':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, 1112747075:::INT8, 32767:::INT8, (-1.1490213871002197):::FLOAT8, 0.6498124854295931:::FLOAT8, '1970-01-01':::DATE, '1979-07-07 05:56:46.00026':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, true, (-1.0):::DECIMAL, '':::STRING, '\x22':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-640537122):::INT8, 127:::INT8, (-0.0):::FLOAT8, 5e-324:::FLOAT8, '1977-02-13':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '1998-07-14 14:06:43.000603+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-331.5854086614966725):::DECIMAL, '':::STRING, '\x0003':::BYTES, '389946bc-bb53-4dbe-88ca-d8090eb3cfef':::UUID, '151.164.245.72/6':::INET, '{"baz": {"PpcoP": null, "tViNh": null, "xrv": {"bar": null}}, "qgZtStKW9": 1.689033963917013}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-32768):::INT8, (-2147483648):::INT8, 0.40900948643684387:::FLOAT8, 1.0:::FLOAT8, '5874897-12-31':::DATE, '1978-07-14 03:18:24.000827':::TIMESTAMP, '1997-02-01 03:58:28.000024+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, 0:::DECIMAL, '?%Z8L':::STRING, '\x27':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '176.86.137.90/17':::INET, '[{"XfbTG": "Xfc", "baz": [null]}, {"X2Shg7bdf": null, "a": true}, false, "foo", null, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 1559229782:::INT8, 8840909812378170706:::INT8, 0.0:::FLOAT8, (-0.46523520340685054):::FLOAT8, '5874897-12-31':::DATE, '2032-06-13 13:30:58.000752':::TIMESTAMP, '2021-12-30 20:17:36.000044+00':::TIMESTAMPTZ, '-28 years -6 mons -517 days -01:22:25.301393':::INTERVAL, true, 0:::DECIMAL, e'Fcv\b':::STRING, '\xaa4e16343879':::BYTES, NULL, '202.101.50.53/17':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, (-399265779):::INT8, NULL, 1.722324252128601:::FLOAT8, 0.8946693520292838:::FLOAT8, '1995-03-11':::DATE, '2032-05-22 09:11:36.000857':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 4.83804958212550090E+27:::DECIMAL, e'\x00':::STRING, NULL, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1883:::INT8, 32767:::INT8, (-2147483648):::INT8, (-0.032988034188747406):::FLOAT8, '-Inf':::FLOAT8, '1970-01-01':::DATE, '2022-07-13 08:07:34.000275':::TIMESTAMP, '2012-01-08 06:30:02.000978+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-1.0):::DECIMAL, e'koiE(\x07\x1dV\x1a':::STRING, '\x':::BYTES, 'a099ace4-1504-41b0-8f0b-38eec2cdea51':::UUID, '247.23.216.166/7':::INET, '{"94s025P": {"Fape3qZA": null, "foo": [false, 0.1587261394621503, false]}, "b": true, "kwR": 0.03891581942790076}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-19041):::INT8, (-1):::INT8, 9223372036854775807:::INT8, '-Inf':::FLOAT8, '+Inf':::FLOAT8, NULL, '294276-12-31 23:59:59.999999':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '53 years 7 mons 96 days 04:44:02.002182':::INTERVAL, true, (-6311298182048.501843):::DECIMAL, e'7fK+\x078gQ\'':::STRING, '\x58':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '7978:c813:cab1:59bd:62e5:e99b:2c6c:346/6':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-373426635):::INT8, 0:::INT8, 0.06306002289056778:::FLOAT8, '+Inf':::FLOAT8, 'infinity':::DATE, '1970-11-01 08:42:55.000202':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '7 years 271 days 08:07:27.07466':::INTERVAL, false, (-28400341131.83404714):::DECIMAL, 'X':::STRING, '\x1168aaf55f8c5f90':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, NULL, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, NULL, 9223372036854775807:::INT8, 1.6370717287063599:::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '5874897-12-31':::DATE, NULL, '1975-12-15 05:40:02.000443+00':::TIMESTAMPTZ, '-41 years -8 mons -353 days -11:19:45.429586':::INTERVAL, true, 7001.983058922763896:::DECIMAL, e'M\x7f7\x14C\n\x05\x04':::STRING, '\xe29883':::BYTES, '9ec4ed4f-49ac-4112-a187-26d0d24a973c':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 46 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._inet::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e'9/$n}2\n':::REFCURSOR::REFCURSOR NOT IN ()
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e'9/$n}2\n':::REFCURSOR::REFCURSOR NOT IN () because error pq: unsupported comparison operator: refcursor NOT IN tuple
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (((seed._string LIKE seed._string) OR seed._bool) OR st_isvalidtrajectory('01060000C00600000001030000C00100000008000000154CD25F802801C260E29A67821EDF413E26E3BC1CBEF4C132BAFB446D8C0042CC1795650C16F7C1401E7B5C0B89C641C8642D7640ACEF419409966F7EF5F6C130782914D6A6C6410A6C0328AED8FCC1545FCA6F0C1CDDC128635BDC4DC3004290949B756DB7C2C1C05754E2828601428CAAFEA46D3B024230961D53D5BAED4116A3B29BDA8BEEC1E043616A0F42014264B768281804F9414834EC1B74D60042E63295E8AC2002C2EA6B17C0307A024262AD4CA4CD7EE0C13E68388281B70042F4B229A39F8AFDC172971CED81CAF64156B0306CBA04F44164BDDFFF7C70D7C1154CD25F802801C260E29A67821EDF413E26E3BC1CBEF4C132BAFB446D8C004201030000C0010000000A00000008E93BAF1482D0C16AAF852C5E33E0C1D8DEFCE7CA1EFA4154F3C9404B990142A0773C74AB0ED74157D7041FE706F0C164995DE2BC10FB41BC9491451822E1C1181AC4D6C555FF411C402545AFD201C2CB39D4C933A3F3C1709319C25B6BCAC17A6917A109C7F4417D745D2A9415F6C1503A3CB4AC15DEC194F845F5EF3FF6419CA544AC9829FD41A07D81FA949EC341CEAF47A4A9EEFFC1D02BEC0134F3E6C1F446187C9270E241ACDB3D2D6FE3F441945119BB88CCED41FC447474A8E7FD4180C65D48E9EBBAC178CF4891EE55D2412FA56605529CFCC17697FBEB34F901421A2D5B1F4AAFF2C1E059D3B0C998E1417C1AAE93DA23EBC12CAFB4FD2A94E341E4F634905E58FFC140E250E71FB2C741B3B3229DB880F2C1AA4B050E42BEFFC108E93BAF1482D0C16AAF852C5E33E0C1D8DEFCE7CA1EFA4154F3C9404B99014201030000C001000000040000002C0958B0A290E941CDF0E534FDC600C2E078DA0A05C1D2C1D291EF8E6F6DF841FCDD3EB97CE0E5C110E59DD3CFD1CEC1D43B7C9152AAFC4146E14C91B4B2FBC1C045A49B0B00A8C10AE164EF8A00F2C1B66DCA4D7AC5E7C1D069EA7E08C8CEC12C0958B0A290E941CDF0E534FDC600C2E078DA0A05C1D2C1D291EF8E6F6DF84101030000C001000000070000001295E753C4F0F2C1211D68E10919FBC1D0D8A5AB1D000142E00E96D80207E9411C35CB051B65FD41769745AE8164EEC14E1EBA87B85DF2416835BD24DFB0FE41344DAF35A378FE41FEDCA4ECB01DEEC1979903059C59F4C1627354303E76F941D299B5000F1EFAC1F0D39F27F21BDC41AC93A0D89FF2F04100497853CECBA641F12E9F8245F400C2580D7C7EE4ABDB41E4FC99DE1B5EE341A028B748AC960042B4FD05726DF7ECC1B284F82C5124E0C1B8AD09889D1E02C207BB2D17C62BF0C11295E753C4F0F2C1211D68E10919FBC1D0D8A5AB1D000142E00E96D80207E94101030000C001000000070000001EA8B4D8383CFCC1C4421BB1072CEB41505E272B254BC4C1D85BE7D6FF73F941A1A5CBE3FF5BFEC18096926901BBDC416DAEEF449FBAFAC1C6F3A234349BF2418E1436C17B34F7C150DBD42845E7CF415F9B675688D900C25CA3B6651495EB4130CAE7577E85C941C09D37A24D5EE54196F236271492F24198ACEC10AD75D241BC33F510481B01428AF70BB60DFA0142800EAA1FD672D0C1CCF93A6FF1E1EAC100C3675D86D39A41820C31ED8968F941156C0A9C94F3F5C110A5F37A7258CB411EA8B4D8383CFCC1C4421BB1072CEB41505E272B254BC4C1D85BE7D6FF73F94101030000C0010000000B000000FC1596EEEDABE4C14E41F90F2EDBE0C1BC71B2182AFDE941D8BA212947BFFCC1CD7EC5C5B23CF0C1C49AF72ABB84F0C18398E1C1CBD601C250B3D1D886B3E3C1DC739FD8E1D0E5414061593D1A4D01C2247B9B2AF5FB01C2B81F40A43F30F441F8D5E72803DAFB4141E64FF3834202C250458AE418D5CCC154CA0AC72E30FCC1C87F8B5731C3F141725AC59FB83BE9C19770AD68422EF5C1CA2F81A1D4530242480B07E8B6FAFD41E021B4F11EA9D0C19478DE012491E741209C03E9F850CFC114DCD4DBD77FE841E0F81EBE8803FC41748603A7D87CE541B10A0695B1FAF9C162D0EADDC606EEC138B9B978B618EB415D2EB6D144E700C2D2DAAE8ACC4BF541F80AB36C5A0BFBC16E55AFE13929F2419860BE3FE8A1E3C14E6E90C74DD201C2BE7741AFAD26F5C1901EBC284CB9E541EE5AC19629FEEBC11868CA72DCD1D241FC1596EEEDABE4C14E41F90F2EDBE0C1BC71B2182AFDE941D8BA212947BFFCC1':::GEOMETRY::GEOMETRY)::BOOL))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (((seed._string LIKE seed._string) OR seed._bool) OR st_isvalidtrajectory('01060000C00600000001030000C00100000008000000154CD25F802801C260E29A67821EDF413E26E3BC1CBEF4C132BAFB446D8C0042CC1795650C16F7C1401E7B5C0B89C641C8642D7640ACEF419409966F7EF5F6C130782914D6A6C6410A6C0328AED8FCC1545FCA6F0C1CDDC128635BDC4DC3004290949B756DB7C2C1C05754E2828601428CAAFEA46D3B024230961D53D5BAED4116A3B29BDA8BEEC1E043616A0F42014264B768281804F9414834EC1B74D60042E63295E8AC2002C2EA6B17C0307A024262AD4CA4CD7EE0C13E68388281B70042F4B229A39F8AFDC172971CED81CAF64156B0306CBA04F44164BDDFFF7C70D7C1154CD25F802801C260E29A67821EDF413E26E3BC1CBEF4C132BAFB446D8C004201030000C0010000000A00000008E93BAF1482D0C16AAF852C5E33E0C1D8DEFCE7CA1EFA4154F3C9404B990142A0773C74AB0ED74157D7041FE706F0C164995DE2BC10FB41BC9491451822E1C1181AC4D6C555FF411C402545AFD201C2CB39D4C933A3F3C1709319C25B6BCAC17A6917A109C7F4417D745D2A9415F6C1503A3CB4AC15DEC194F845F5EF3FF6419CA544AC9829FD41A07D81FA949EC341CEAF47A4A9EEFFC1D02BEC0134F3E6C1F446187C9270E241ACDB3D2D6FE3F441945119BB88CCED41FC447474A8E7FD4180C65D48E9EBBAC178CF4891EE55D2412FA56605529CFCC17697FBEB34F901421A2D5B1F4AAFF2C1E059D3B0C998E1417C1AAE93DA23EBC12CAFB4FD2A94E341E4F634905E58FFC140E250E71FB2C741B3B3229DB880F2C1AA4B050E42BEFFC108E93BAF1482D0C16AAF852C5E33E0C1D8DEFCE7CA1EFA4154F3C9404B99014201030000C001000000040000002C0958B0A290E941CDF0E534FDC600C2E078DA0A05C1D2C1D291EF8E6F6DF841FCDD3EB97CE0E5C110E59DD3CFD1CEC1D43B7C9152AAFC4146E14C91B4B2FBC1C045A49B0B00A8C10AE164EF8A00F2C1B66DCA4D7AC5E7C1D069EA7E08C8CEC12C0958B0A290E941CDF0E534FDC600C2E078DA0A05C1D2C1D291EF8E6F6DF84101030000C001000000070000001295E753C4F0F2C1211D68E10919FBC1D0D8A5AB1D000142E00E96D80207E9411C35CB051B65FD41769745AE8164EEC14E1EBA87B85DF2416835BD24DFB0FE41344DAF35A378FE41FEDCA4ECB01DEEC1979903059C59F4C1627354303E76F941D299B5000F1EFAC1F0D39F27F21BDC41AC93A0D89FF2F04100497853CECBA641F12E9F8245F400C2580D7C7EE4ABDB41E4FC99DE1B5EE341A028B748AC960042B4FD05726DF7ECC1B284F82C5124E0C1B8AD09889D1E02C207BB2D17C62BF0C11295E753C4F0F2C1211D68E10919FBC1D0D8A5AB1D000142E00E96D80207E94101030000C001000000070000001EA8B4D8383CFCC1C4421BB1072CEB41505E272B254BC4C1D85BE7D6FF73F941A1A5CBE3FF5BFEC18096926901BBDC416DAEEF449FBAFAC1C6F3A234349BF2418E1436C17B34F7C150DBD42845E7CF415F9B675688D900C25CA3B6651495EB4130CAE7577E85C941C09D37A24D5EE54196F236271492F24198ACEC10AD75D241BC33F510481B01428AF70BB60DFA0142800EAA1FD672D0C1CCF93A6FF1E1EAC100C3675D86D39A41820C31ED8968F941156C0A9C94F3F5C110A5F37A7258CB411EA8B4D8383CFCC1C4421BB1072CEB41505E272B254BC4C1D85BE7D6FF73F94101030000C0010000000B000000FC1596EEEDABE4C14E41F90F2EDBE0C1BC71B2182AFDE941D8BA212947BFFCC1CD7EC5C5B23CF0C1C49AF72ABB84F0C18398E1C1CBD601C250B3D1D886B3E3C1DC739FD8E1D0E5414061593D1A4D01C2247B9B2AF5FB01C2B81F40A43F30F441F8D5E72803DAFB4141E64FF3834202C250458AE418D5CCC154CA0AC72E30FCC1C87F8B5731C3F141725AC59FB83BE9C19770AD68422EF5C1CA2F81A1D4530242480B07E8B6FAFD41E021B4F11EA9D0C19478DE012491E741209C03E9F850CFC114DCD4DBD77FE841E0F81EBE8803FC41748603A7D87CE541B10A0695B1FAF9C162D0EADDC606EEC138B9B978B618EB415D2EB6D144E700C2D2DAAE8ACC4BF541F80AB36C5A0BFBC16E55AFE13929F2419860BE3FE8A1E3C14E6E90C74DD201C2BE7741AFAD26F5C1901EBC284CB9E541EE5AC19629FEEBC11868CA72DCD1D241FC1596EEEDABE4C14E41F90F2EDBE0C1BC71B2182AFDE941D8BA212947BFFCC1':::GEOMETRY::GEOMETRY)::BOOL)) to emit 46 events
    changefeed_test.go:1141: st_isvalidtrajectory(): expected LineString, got MultiPolygonZM
    --- FAIL: TestChangefeedRandomExpressions/kafka (4.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/6c388c68-b7e8-4c4f-b47d-159a54ff1853?testReportRun=23&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [7166724ff02b9b7f0d5787f74d919487ca347bd0](https://github.com/cockroachdb/cockroach/commits/7166724ff02b9b7f0d5787f74d919487ca347bd0):


```
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT NULL)
    changefeed_test.go:1120: Skipping predicate (NOT NULL) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._inet::INET - get_bit(seed._bytes::BYTES, seed._int8::INT8)::INT8::INT8)::INET::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._inet::INET - get_bit(seed._bytes::BYTES, seed._int8::INT8)::INT8::INT8)::INET::INET >> seed._inet::INET)::BOOL because error pq: get_bit(): bit index 127 out of valid range (0..63)
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._float8 AS col_100 FROM d.public.seed@[0] AS " tab""_46" WHERE seed._bool ORDER BY seed._int2)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._float8 AS col_100 FROM d.public.seed@[0] AS " tab""_46" WHERE seed._bool ORDER BY seed._int2) to emit 47 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT st_overlaps('01050000C00100000001020000C0060000000C97389B8F7CEFC138DE63EB6A34E441F1C0CDB1D520FCC16C7F81118328FAC120ED8E0426A4BEC1C08D1FE7968DA7C1694CF239865DF3C140E93A59C59AF741445652E9FD04FF41971584032C02F8C136F03D1E5DB4F14170C8B7740354F1C17E0CC04AA99AFA41DEF01B58823D00423C946980B710F7C161CD8E2D467500C22C8E5E99899AF2C11242557832990142004A1290E5C9B3C12A823E69DD12E7C1C3551FF0A72F00C2788F0356FE5EFF41EC71B8368B2E02425880DD73BCE8E3C1':::GEOMETRY::GEOMETRY, st_force3dz('010700000000000000':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT st_overlaps('01050000C00100000001020000C0060000000C97389B8F7CEFC138DE63EB6A34E441F1C0CDB1D520FCC16C7F81118328FAC120ED8E0426A4BEC1C08D1FE7968DA7C1694CF239865DF3C140E93A59C59AF741445652E9FD04FF41971584032C02F8C136F03D1E5DB4F14170C8B7740354F1C17E0CC04AA99AFA41DEF01B58823D00423C946980B710F7C161CD8E2D467500C22C8E5E99899AF2C11242557832990142004A1290E5C9B3C12A823E69DD12E7C1C3551FF0A72F00C2788F0356FE5EFF41EC71B8368B2E02425880DD73BCE8E3C1':::GEOMETRY::GEOMETRY, st_force3dz('010700000000000000':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._inet::INET >> seed._inet::INET)::BOOL OR true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((seed._inet::INET >> seed._inet::INET)::BOOL OR true) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND true) to emit 47 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string SIMILAR TO postgis_liblwgeom_version()::STRING
    changefeed_test.go:1120: Skipping predicate seed._string SIMILAR TO postgis_liblwgeom_version()::STRING because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (EXISTS (SELECT B'0011011001110100100101110100100' AS col_181 FROM d.public.seed@[0] AS "%ptab_90" JOIN d.public.seed AS "tab\\uA770_91" JOIN d.public.seed AS "Tab&_92" ON ("tab\\uA770_91"._int8) = ("Tab&_92"._int8) JOIN d.public.seed@[0] AS tab_93 ON (("tab\\uA770_91"._date) = (tab_93._date)) AND (("tab\\uA770_91"._float8) = (tab_93._float4)) JOIN d.public.seed AS "t\\u35C7ab_94" JOIN d.public.seed AS tab_95 ON ("t\\u35C7ab_94"._jsonb) = (tab_95._jsonb) ON (tab_93._jsonb) = ("t\\u35C7ab_94"._jsonb) ON ((("%ptab_90".tableoid) = (tab_93.tableoid)) AND (("%ptab_90"._int8) = ("Tab&_92"._int8))) AND (("%ptab_90"._date) = ("Tab&_92"._date))) OR (NOT true))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (EXISTS (SELECT B'0011011001110100100101110100100' AS col_181 FROM d.public.seed@[0] AS "%ptab_90" JOIN d.public.seed AS "tab\\uA770_91" JOIN d.public.seed AS "Tab&_92" ON ("tab\\uA770_91"._int8) = ("Tab&_92"._int8) JOIN d.public.seed@[0] AS tab_93 ON (("tab\\uA770_91"._date) = (tab_93._date)) AND (("tab\\uA770_91"._float8) = (tab_93._float4)) JOIN d.public.seed AS "t\\u35C7ab_94" JOIN d.public.seed AS tab_95 ON ("t\\u35C7ab_94"._jsonb) = (tab_95._jsonb) ON (tab_93._jsonb) = ("t\\u35C7ab_94"._jsonb) ON ((("%ptab_90".tableoid) = (tab_93.tableoid)) AND (("%ptab_90"._int8) = ("Tab&_92"._int8))) AND (("%ptab_90"._date) = ("Tab&_92"._date))) OR (NOT true)) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string LIKE seed._string
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string LIKE seed._string to emit 97 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT seed._bool) AND false)
    changefeed_test.go:1120: Skipping predicate ((NOT seed._bool) AND false) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._date::DATE - seed._date::DATE)::INT8::INT8 IN (seed._int2, NULL, 127:::INT8)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._date::DATE - seed._date::DATE)::INT8::INT8 IN (seed._int2, NULL, 127:::INT8) to emit 12 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1141: cannot subtract infinite dates
    --- FAIL: TestChangefeedRandomExpressions/kafka (7.24s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=23</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/5909c881-23b7-41e6-9f82-0bd62caf9d6a?testReportRun=17&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [2f498d0abcf4130e5c872b73200086d6a3af9f02](https://github.com/cockroachdb/cockroach/commits/2f498d0abcf4130e5c872b73200086d6a3af9f02):


```
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._inet AS col_106, '757071AD/4C1FCF92':::PG_LSN AS col_107, seed.tableoid AS col_108) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false::BOOL NOT IN (SELECT (EXISTS (SELECT NULL AS "col _142" FROM d.public.seed AS "-ta😰b_53" JOIN d.public.seed AS "t.ab_54" ON ((("-ta😰b_53"._int8) = ("t.ab_54"._int8)) AND (("-ta😰b_53"._date) = ("t.ab_54"._date))) AND (("-ta😰b_53"._float8) = ("t.ab_54"._float8))) AND false)::BOOL AS col_143 FROM d.public.seed AS " tab_51" JOIN d.public.seed AS "}tab_52" ON (((" tab_51"._int8) = ("}tab_52"._int8)) AND ((" tab_51"._date) = ("}tab_52"._date))) AND ((" tab_51"._float8) = ("}tab_52"._float8)) WHERE ('4e19:4e6a:1a03:9630:8243:6fc3:da08:6c35/6':::INET::INET << '0.0.0.0/0':::INET::INET)::BOOL)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE false::BOOL NOT IN (SELECT (EXISTS (SELECT NULL AS "col _142" FROM d.public.seed AS "-ta😰b_53" JOIN d.public.seed AS "t.ab_54" ON ((("-ta😰b_53"._int8) = ("t.ab_54"._int8)) AND (("-ta😰b_53"._date) = ("t.ab_54"._date))) AND (("-ta😰b_53"._float8) = ("t.ab_54"._float8))) AND false)::BOOL AS col_143 FROM d.public.seed AS " tab_51" JOIN d.public.seed AS "}tab_52" ON (((" tab_51"._int8) = ("}tab_52"._int8)) AND ((" tab_51"._date) = ("}tab_52"._date))) AND ((" tab_51"._float8) = ("}tab_52"._float8)) WHERE ('4e19:4e6a:1a03:9630:8243:6fc3:da08:6c35/6':::INET::INET << '0.0.0.0/0':::INET::INET)::BOOL) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (seed._bool AND seed._bool))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT (seed._bool AND seed._bool)) to emit 39 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_coveredby('01040000C00200000001010000C04ADEC1A2B65DE1C11BF12D7C4F47FBC14E0BC755FD5900C25839AB2CFE0502C201010000C0B0E037FBD73AC9C198214E9EF56CD9416CA2F68DC1CCF841A823EB9CE20EDF41':::GEOMETRY::GEOMETRY, st_makepoint((-2.7174618785320352):::FLOAT8::FLOAT8, (-0.5971368254407063):::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::BOOL
    changefeed_test.go:1120: Skipping predicate _st_coveredby('01040000C00200000001010000C04ADEC1A2B65DE1C11BF12D7C4F47FBC14E0BC755FD5900C25839AB2CFE0502C201010000C0B0E037FBD73AC9C198214E9EF56CD9416CA2F68DC1CCF841A823EB9CE20EDF41':::GEOMETRY::GEOMETRY, st_makepoint((-2.7174618785320352):::FLOAT8::FLOAT8, (-0.5971368254407063):::FLOAT8::FLOAT8)::GEOMETRY::GEOMETRY)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT true)
    changefeed_test.go:1120: Skipping predicate (NOT true) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true OR false)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true OR false) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (false OR EXISTS (SELECT seed._int4 AS "c
        ol _235" FROM d.public.seed AS "%vta
        b_93" JOIN d.public.seed AS tab_94 ON ("%vta
        b_93"._jsonb) = (tab_94._jsonb) WHERE NULL))
    changefeed_test.go:1120: Skipping predicate (false OR EXISTS (SELECT seed._int4 AS "c
        ol _235" FROM d.public.seed AS "%vta
        b_93" JOIN d.public.seed AS tab_94 ON ("%vta
        b_93"._jsonb) = (tab_94._jsonb) WHERE NULL)) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_dfullywithinexclusive('0101000000605BF0C479E4CF41101DFAD1FB86C2C1':::GEOMETRY::GEOMETRY, '01040000000800000001010000009E288574A1280142088D1BC8FB9EFB4101010000001C797C711FC4E8C110AB2831154CE5410101000000E0BFD4419915BC41CC2188633187FE4101010000004E30DA8FB3EAFEC1F8CA312B40E0D3C1010100000064851711BDB3F1C1378416DF847B01C2010100000068DA0DD2F4BCFC41A023D103799BF8410101000000123DCE575075E3C1F0C5CCFE9A93E94101010000009032863D2B1FF1C1C947632998ACF2C1':::GEOMETRY::GEOMETRY, seed._float8::FLOAT8)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_dfullywithinexclusive('0101000000605BF0C479E4CF41101DFAD1FB86C2C1':::GEOMETRY::GEOMETRY, '01040000000800000001010000009E288574A1280142088D1BC8FB9EFB4101010000001C797C711FC4E8C110AB2831154CE5410101000000E0BFD4419915BC41CC2188633187FE4101010000004E30DA8FB3EAFEC1F8CA312B40E0D3C1010100000064851711BDB3F1C1378416DF847B01C2010100000068DA0DD2F4BCFC41A023D103799BF8410101000000123DCE575075E3C1F0C5CCFE9A93E94101010000009032863D2B1FF1C1C947632998ACF2C1':::GEOMETRY::GEOMETRY, seed._float8::FLOAT8)::BOOL because error pq: st_dfullywithinexclusive(): dwithin distance cannot be less than zero
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL::STRING % 'NF3,S&"d4':::STRING::STRING)::BOOL
    changefeed_test.go:1120: Skipping predicate (NULL::STRING % 'NF3,S&"d4':::STRING::STRING)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT e'!\'KbRAnyqTSQ\' | \'NkjaT\' <-> \'CIapAWanbX\' | \'WTkTw\'':::TSQUERY AS "co?l_284", e'\'Ei\':54B \'Gn\' \'HVxgV\':518A,632A,733B,912B \'L\' \'bRKRq\':150C,820C,948C \'nTnQ\' \'ougRSJT\' \'sOGYUGsz\':110C,257B,333C \'xa\'':::TSVECTOR AS col_285 FROM d.public.seed AS tab̀_115 WHERE seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT e'!\'KbRAnyqTSQ\' | \'NkjaT\' <-> \'CIapAWanbX\' | \'WTkTw\'':::TSQUERY AS "co?l_284", e'\'Ei\':54B \'Gn\' \'HVxgV\':518A,632A,733B,912B \'L\' \'bRKRq\':150C,820C,948C \'nTnQ\' \'ougRSJT\' \'sOGYUGsz\':110C,257B,333C \'xa\'':::TSVECTOR AS col_285 FROM d.public.seed AS tab̀_115 WHERE seed._bool) to emit 53 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true AND true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true AND true) to emit 100 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (((-8607119618291217877):::INT8::INT8 | seed._int4::INT8)::INT8 < (seed._date::DATE - seed._date::DATE)::INT8)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (((-8607119618291217877):::INT8::INT8 | seed._int4::INT8)::INT8 < (seed._date::DATE - seed._date::DATE)::INT8) to emit 87 events
    testfeed_test.go:283: creating external connection
    testfeed_test.go:286: ran create external connection
    changefeed_test.go:1141: cannot subtract infinite dates
    --- FAIL: TestChangefeedRandomExpressions/kafka (4.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #117050 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16157004?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16157004?buildTab=artifacts#/) on master @ [8def76b7b83df90d7e2bdf07309d61da8bfcf1d3](https://github.com/cockroachdb/cockroach/commits/8def76b7b83df90d7e2bdf07309d61da8bfcf1d3):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (29110:::INT8, (-231142125):::INT8, NULL, (-1.0):::FLOAT8, 0.0:::FLOAT8, '1987-11-05':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '40 years 8 mons 859 days 13:11:37.784917':::INTERVAL, true, (-2.832314707625368494E+26):::DECIMAL, '"':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[1.6702382013684691, null]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-1057272699):::INT8, (-1500295690125651295):::INT8, 1.0:::FLOAT8, 3.4028234663852886e+38:::FLOAT8, '4714-11-24 BC':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '1 day':::INTERVAL, false, (-26034335.63966049349):::DECIMAL, 'X':::STRING, '\x0db244':::BYTES, '266d435d-6c49-49f1-b1e5-3b6155d629f4':::UUID, 'd3f4:c6e0:8e50:cc20:8b49:4ddc:4623:66dd/76':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1273:::INT8, (-1):::INT8, 127:::INT8, (-0.0):::FLOAT8, (-1.9878061314565811):::FLOAT8, '-infinity':::DATE, '2004-10-07 22:17:20.000729':::TIMESTAMP, '2026-06-18 18:15:18.000966+00':::TIMESTAMPTZ, '-8 years -3 mons -514 days -05:27:48.708846':::INTERVAL, true, (-4.180426722556408815E+31):::DECIMAL, e'\x18#':::STRING, '\x27':::BYTES, 'dbb3d85c-3435-436a-b83e-c3c9384ea676':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-27626):::INT8, 1:::INT8, NULL, 1.2053967714309692:::FLOAT8, 1.549064067356011:::FLOAT8, '1980-10-23':::DATE, '2005-12-13 13:09:16.000089':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, false, (-5.67061285717253291E+19):::DECIMAL, e'>\x18pi':::STRING, '\xed':::BYTES, '0299f572-1dea-492c-9871-4104e95a465c':::UUID, '93.244.177.233/18':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (2209:::INT8, NULL, (-3890616800968222515):::INT8, 'NaN':::FLOAT8, (-0.15201059828683583):::FLOAT8, 'infinity':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2001-04-12 14:45:57.000625+00':::TIMESTAMPTZ, '20 years 8 mons 449 days 08:38:37.039684':::INTERVAL, false, 1.0:::DECIMAL, e'GlYj~\t':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '120.21.55.218/19':::INET, '{"0YKld": {"ZUX": {"bar": "foo"}, "yl3b3g08A": true}, "aQgmd9O": [{"9JYSxFCm": null, "c": true, "iq": [null]}, 0.19002930424802233, "E88r", "h8"], "foo": null, "gpYR0aM": null}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (7216:::INT8, 319418379:::INT8, (-32768):::INT8, 0.9015381932258606:::FLOAT8, 0.0:::FLOAT8, '5874897-12-31':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '1984-08-21 16:39:03.0007+00':::TIMESTAMPTZ, NULL, true, 16190010047622658.6:::DECIMAL, NULL, '\x':::BYTES, 'ae6b1398-e618-46b2-9a0e-89dde7cfaf2d':::UUID, '248.154.214.87/0':::INET, '{"c": {"VUkRGmF": 0.8163915447490805, "b": {"F3yaM": false, "h5mr": null}, "c": [null, 0.032612264521197865, 0.181375353758013]}, "chCDO": "eLeI", "foobar": {"foobar": true}, "x6Je": [0.2120257419363895, true]}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (13894:::INT8, (-1240300481):::INT8, (-1):::INT8, '+Inf':::FLOAT8, 1.7976931348623157e+308:::FLOAT8, NULL, '2001-01-01 00:00:00 BC':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-179301490.3439535758):::DECIMAL, 'V2B t':::STRING, '\x499833fb':::BYTES, '84fddd6f-a719-49a2-ac5a-7e6f1fa02085':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 127:::INT8, 6151155622921558535:::INT8, '+Inf':::FLOAT8, 1.5633490219990045:::FLOAT8, '1997-03-01':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-12 years -4 mons -210 days -03:25:08.936774':::INTERVAL, false, (-1.234E+401):::DECIMAL, e'\x1f\x0ey\x1c\n':::STRING, '\xe29883':::BYTES, '9477a094-1e6e-4318-a6a0-61919781c5fc':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (5434:::INT8, 1:::INT8, 127:::INT8, (-1.9510574340820312):::FLOAT8, 0.7734965005023559:::FLOAT8, '1979-12-27':::DATE, '1993-02-11 09:37:53.000609':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '-51 years -3 mons -928 days -12:26:43.737895':::INTERVAL, true, 9.55011537612707814E+25:::DECIMAL, 'X':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '1a9a:a2e9:597c:db61:d306:8690:fb5d:f4f1/126':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-11010):::INT8, 32767:::INT8, 0:::INT8, (-0.7021192312240601):::FLOAT8, 1.401298464324817e-45:::FLOAT8, '5874897-12-31':::DATE, '2019-02-16 20:20:34.00038':::TIMESTAMP, NULL, '21 years 9 mons 906 days 02:35:38.588137':::INTERVAL, false, (-9197638472500460.13):::DECIMAL, 'X':::STRING, '\xd8':::BYTES, 'a2298b65-0a3b-4be5-83ff-2036dbfd5e44':::UUID, '0.0.0.0/0':::INET, '[false, true, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, (-31192028):::INT8, NULL, (-1.2613052129745483):::FLOAT8, 1.0:::FLOAT8, '1994-07-16':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '68 years 2 mons 917 days 20:02:36.595452':::INTERVAL, true, NULL, e'\'':::STRING, NULL, 'bc3ef1f1-d4e9-436f-bc77-9cfe1fb3f943':::UUID, '159.55.138.204/17':::INET, '{"c": "79I", "fHlVx": false}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-9592):::INT8, 1942060786:::INT8, (-7125507220343678299):::INT8, 0.1361234039068222:::FLOAT8, 0.0:::FLOAT8, 'infinity':::DATE, '2004-11-10 02:31:46.000441':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '-61 years -5 mons -464 days -08:31:15.174007':::INTERVAL, NULL, (-1263680.944729915718):::DECIMAL, 'D':::STRING, '\x853914786002541c':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '[true, [[false], null, false], [null, 0.36388316706654295], ["JQh"], null, "a"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-2147483648):::INT8, 2363927101478130887:::INT8, '-Inf':::FLOAT8, 0.2431344590686635:::FLOAT8, NULL, '1978-12-29 07:47:47.000178':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-6.60271032884238147E+31):::DECIMAL, e'7\x17\x0e\x0f\x1f:':::STRING, '\x58':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{"baz": "bar", "foobar": null, "lG": {"Hdg": false, "Ue": true, "vqtH": true}}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-21736):::INT8, 535900974:::INT8, NULL, 0.4565350115299225:::FLOAT8, NULL, '1994-11-15':::DATE, '2019-06-09 01:56:25.000489':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-53 years -6 mons -421 days -22:31:36.412028':::INTERVAL, false, (-5.43219888648437382E+19):::DECIMAL, e'\'':::STRING, '\xd39fb03ac3139b94':::BYTES, '08c1deaf-02fc-4106-8f8e-ed7e1a62cbec':::UUID, '69.112.117.93/25':::INET, '[[[false], "foobar"], {"4UzAxQ": null, "Axrv": 2.205051016603078, "tOJwTjSj6": null}, [true], {"b": false}, true, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 731371110:::INT8, 8369398558026996708:::INT8, 0.33274099230766296:::FLOAT8, NULL, '1972-02-26':::DATE, '2005-03-16 07:37:44.000027':::TIMESTAMP, '2027-05-07 21:13:52.000021+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, '-Infinity':::DECIMAL, e'\x00':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, 1539976422:::INT8, 2147483647:::INT8, 0.0:::FLOAT8, 'NaN':::FLOAT8, '1995-11-16':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '64 years 1 mon 743 days 02:52:22.861141':::INTERVAL, false, 1.0:::DECIMAL, e'\U00002603':::STRING, '\x0567a6ab73a2eedec2':::BYTES, '1581116c-9b19-45fb-8416-dabfe00d0f00':::UUID, '0.0.0.0/0':::INET, '["b", {"b": {"Ye": true, "a": "b"}, "bar": 1.5904773919793835, "foo": [null]}, null, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 1:::INT8, 127:::INT8, NULL, 0.0:::FLOAT8, '1993-06-27':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2007-10-08 12:25:54.00025+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 231993.3863883511375:::DECIMAL, e'\x7f':::STRING, '\x0253b870ffd55e6cef':::BYTES, 'c71578b7-470f-4a4a-8402-18e76face654':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"1l1DUUBTk": {"YE18F26hC": false}, "544W": [0.9790609143660904, "b"], "a9vIN61": {"DruMyJ": {"foo": false}, "b": ["Ra0KP", null], "baz": null, "fRX59": "O9PjfC"}, "c": null}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 32767:::INT8, (-32768):::INT8, '+Inf':::FLOAT8, 0.15841356838633014:::FLOAT8, 'infinity':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 2.569077850738698765E+23:::DECIMAL, '"':::STRING, '\x3c238d5a27':::BYTES, 'f6e3b912-0416-4fd5-807c-a8501398efc0':::UUID, '72a6:1018:2784:3722:378c:9f3b:b307:3c4b/99':::INET, '[{"foo": true, "npP": "L4JAy"}, [["6m5"], null], false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (8051:::INT8, (-32768):::INT8, 1:::INT8, NULL, (-0.9321247170939508):::FLOAT8, '1992-01-26':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '78 years 8 mons 346 days 10:17:58.451762':::INTERVAL, true, (-1.234E+401):::DECIMAL, '"':::STRING, '\x6b13':::BYTES, '7fa134b6-78a5-4698-9875-6f60b9950ca2':::UUID, '0.0.0.0/0':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-8144):::INT8, NULL, (-2676854692462966351):::INT8, 0.0:::FLOAT8, (-1.5536194104038836):::FLOAT8, '-infinity':::DATE, NULL, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '4 years 1 mon 34 days 23:35:46.497676':::INTERVAL, false, (-4.745651178800580647E+34):::DECIMAL, '':::STRING, '\x522e29559562164ffb':::BYTES, '38ced364-4eb0-4291-8321-c3ad3b9e66f5':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[[true, null, null], 2.9896941571966256, null]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (25963:::INT8, (-1):::INT8, 6583494780062226745:::INT8, NULL, (-1.0):::FLOAT8, '5874897-12-31':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '2013-10-30 17:28:08.00073+00':::TIMESTAMPTZ, '-10 years -4 mons -138 days -00:36:28.985643':::INTERVAL, false, (-6.734250540022707961E+24):::DECIMAL, e'\U00002603':::STRING, '\x07c3668f52f0217058':::BYTES, 'c389a752-648c-4554-a37e-89f797f28f3b':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 339933568:::INT8, 1:::INT8, (-1.0365873575210571):::FLOAT8, (-1.9342580155074174):::FLOAT8, '1995-08-10':::DATE, '2025-01-24 21:11:36.00068':::TIMESTAMP, NULL, '1 mon 1 day':::INTERVAL, NULL, 0:::DECIMAL, '"':::STRING, '\xe29883':::BYTES, 'b3bf8926-dda3-4ddf-b318-9d22b5e6a2ab':::UUID, '225c:f8e2:71d9:d41f:2086:8d6d:a786:9a58/106':::INET, '1':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '':::STRING NOT ILIKE seed._string
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE '':::STRING NOT ILIKE seed._string to emit 85 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 48 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 46 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('6bdc:8a83:1e71:3750:81fb:24e:5a5a:403d/10':::INET::INET >> NULL::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate ('6bdc:8a83:1e71:3750:81fb:24e:5a5a:403d/10':::INET::INET >> NULL::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._date = '1975-03-21':::DATE) OR false)
    changefeed_test.go:1120: Skipping predicate ((seed._date = '1975-03-21':::DATE) OR false) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (CASE WHEN e'\U00002603':::REFCURSOR::REFCURSOR NOT IN (SELECT refcursor(seed._string::STRING)::REFCURSOR::REFCURSOR AS col_221 FROM d.public.seed AS tab_72 JOIN d.public.seed AS t̏ab_73 ON (tab_72._int8) = (t̏ab_73._int8) JOIN d.public.seed AS tab_74 JOIN d.public.seed AS "tab(_75" JOIN d.public.seed AS tab_76 ON ("tab(_75"._int8) = (tab_76._int8) ON ((tab_74._int4) = ("tab(_75"._int8)) AND ((tab_74._date) = (tab_76._date)) ON ((tab_72._int8) = (tab_74._int8)) AND ((tab_72._date) = ("tab(_75"._date)) WHERE seed._bool ORDER BY seed._int8) THEN NULL ELSE (e'\x14\\_\x10N':::STRING::STRING || '':::REFCURSOR::REFCURSOR)::STRING END::STRING % '':::STRING::STRING)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (CASE WHEN e'\U00002603':::REFCURSOR::REFCURSOR NOT IN (SELECT refcursor(seed._string::STRING)::REFCURSOR::REFCURSOR AS col_221 FROM d.public.seed AS tab_72 JOIN d.public.seed AS t̏ab_73 ON (tab_72._int8) = (t̏ab_73._int8) JOIN d.public.seed AS tab_74 JOIN d.public.seed AS "tab(_75" JOIN d.public.seed AS tab_76 ON ("tab(_75"._int8) = (tab_76._int8) ON ((tab_74._int4) = ("tab(_75"._int8)) AND ((tab_74._date) = (tab_76._date)) ON ((tab_72._int8) = (tab_74._int8)) AND ((tab_72._date) = ("tab(_75"._date)) WHERE seed._bool ORDER BY seed._int8) THEN NULL ELSE (e'\x14\\_\x10N':::STRING::STRING || '':::REFCURSOR::REFCURSOR)::STRING END::STRING % '':::STRING::STRING)::BOOL because error pq: unsupported comparison operator: refcursor NOT IN tuple{refcursor}
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._jsonb AS col_235 FROM d.public.seed AS tab_83 ORDER BY seed._jsonb NULLS FIRST, seed._enum DESC NULLS FIRST, seed.tableoid ASC NULLS LAST)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._jsonb AS col_235 FROM d.public.seed AS tab_83 ORDER BY seed._jsonb NULLS FIRST, seed._enum DESC NULLS FIRST, seed.tableoid ASC NULLS LAST) to emit 100 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT '[4.9200415e+10,-1.2421617e+23,-3.8097295e-19,-3.63415e-08,1.6950994e-34,-0.006601956,1.30253585e+17,-1.1428921e+35,3.9249717e-38,1.2230643e+23,2.1146801e-24,1.0780316e-10,-9.100592e+06,7.9875337e-13,-3.4022208e-23,-2.8944225e+35,-1.9507581e-37,-766.1268,0.0134363305,1.01356904e+08,3698.9912,-1.7836827e+28,1.9325878e-05,1.5571662e-16,7.5498006e+20,-1.3608007e-28,2.3528127e+34,-1.83878e-16,908.0314,9.711173e+36,-2.5482058e-12,6.895188e+35,-2.1259066e+29,-1.5131006e-31,3.9948338e+37,-4.3993973e+15,-5.126992e-32,4.523338e+34,1.1789111e+26,1.1301209,-1.2156481e-21,1.4648947e+32,-2.0430346e+23,-32.735313,-8.8857e+10,1.1660043e-35,-4.9926193e-26,-1.7695127e+35,-1.7720468e-26,2.5216551e+33,-2.2046558e+23,-2.9021696e-21,-2.533231e-18,-9.269392e+26,2.0269698e+29,3.689879e+30,-7.8519863e+24,6.625526e-23,-9.47506e-38,4.4734916e-18,-101661.99,-1.787038e+13,-8.843488e-30,-4.8237588e-36,-2.442928e+27,-1.8803454e+30,1.7407595e-34,1.0536649e+36,3.3081952e-38,3.5627215e-09,-1.2885769e-08,4.5327474e+15,-7.381949e-19,6.415445e+12,2.2588755e+11,-4.2460285e+27,-0.12661026,1.018393e+32,-6.9712824e-35,-4.7905848e-37,-2.2812528e+11,0.005754913,0.00033756925,-4.5503463e+28,-40.87682,1.1268082e+38,-0.014860287,-1.187346e+38,2.0084141e+15,-4.2570518e+30,-1.094047e+14,-2.3859528e-05,-1.6419095e-06,9.6418816e+29,1.14803676e-29,-2.7192343e-08,-2.6954185e-06,-90313.34,-5530.0293,0.19483827,3.445386e-07,-2.0867045e+33,5.8471132e-06,-0.22336684,1.9691273e+15,-6.2981145e-09,-6.3043605e+06,2.9438323e+38,-4.8987274e-29,1.02575865e-35,-5.3791393e-14,3.1984896e-14,-0.0027087238,-2.4997793e+28,4.3539557e+19,1.7830244e+21,-225.15442,-11.471054,-0.0018444138,-9.211328e+13,-8.0167857e-22,-88785.03,-8.480019e+15,-2.8793945e-21,-1.7236387e-23,-8.292595,-1.3420761e-14,2.6714036e-35,-2.6693663e-15,-0.0052280435,48.866375,5.3858304e-07,-5.0194014e-36,827726,-13.953986,-2.6995655e-17,1.2172024e-17,-3.0083635e-33,1.4198154e-15,-2.244507e-24,5.166529e-17,-3.335641e+24,4.7108237e+08,2.0602474e-32,2.344219e+24,1.7379508e-08,3.670296e-23,-2.3053763e+25,9.953676e-31,30.97987,-1.3450299e-38,15.446057,4.3370206e+26,18.580767,-2.0038951e-28,-5.702115e-26,-2.2805216e-31,-11734.308,2.3574587e-09,-2.5029376e-13,-0.0013743767,2.4910062e+34,1.1414531e+17,1.0054673e+10,-168030.39,-1.6843547e-31,-6.057697e-36,-2.7662675e+36,-26.424412,-3.4267672e+14,-2.6199677e-31,-4.7908554e-05,-2.0324258e+32,-2.886432e+31,4.500856e-20,3.9707752e-23,-1.2453457e+35,-1.78253e-25,-3.2106893e-24,2.3520093e-07,9.644719e-15]':::VECTOR AS col_239, (-5305413186402659845):::INT8 AS "_co?l_240", 'F4C0D6F5/406E8B15':::PG_LSN AS col_241 FROM d.public.seed AS "ta!b._87" WHERE (seed._inet::INET << seed._inet::INET)::BOOL)
    changefeed_test.go:1120: Skipping predicate EXISTS (SELECT '[4.9200415e+10,-1.2421617e+23,-3.8097295e-19,-3.63415e-08,1.6950994e-34,-0.006601956,1.30253585e+17,-1.1428921e+35,3.9249717e-38,1.2230643e+23,2.1146801e-24,1.0780316e-10,-9.100592e+06,7.9875337e-13,-3.4022208e-23,-2.8944225e+35,-1.9507581e-37,-766.1268,0.0134363305,1.01356904e+08,3698.9912,-1.7836827e+28,1.9325878e-05,1.5571662e-16,7.5498006e+20,-1.3608007e-28,2.3528127e+34,-1.83878e-16,908.0314,9.711173e+36,-2.5482058e-12,6.895188e+35,-2.1259066e+29,-1.5131006e-31,3.9948338e+37,-4.3993973e+15,-5.126992e-32,4.523338e+34,1.1789111e+26,1.1301209,-1.2156481e-21,1.4648947e+32,-2.0430346e+23,-32.735313,-8.8857e+10,1.1660043e-35,-4.9926193e-26,-1.7695127e+35,-1.7720468e-26,2.5216551e+33,-2.2046558e+23,-2.9021696e-21,-2.533231e-18,-9.269392e+26,2.0269698e+29,3.689879e+30,-7.8519863e+24,6.625526e-23,-9.47506e-38,4.4734916e-18,-101661.99,-1.787038e+13,-8.843488e-30,-4.8237588e-36,-2.442928e+27,-1.8803454e+30,1.7407595e-34,1.0536649e+36,3.3081952e-38,3.5627215e-09,-1.2885769e-08,4.5327474e+15,-7.381949e-19,6.415445e+12,2.2588755e+11,-4.2460285e+27,-0.12661026,1.018393e+32,-6.9712824e-35,-4.7905848e-37,-2.2812528e+11,0.005754913,0.00033756925,-4.5503463e+28,-40.87682,1.1268082e+38,-0.014860287,-1.187346e+38,2.0084141e+15,-4.2570518e+30,-1.094047e+14,-2.3859528e-05,-1.6419095e-06,9.6418816e+29,1.14803676e-29,-2.7192343e-08,-2.6954185e-06,-90313.34,-5530.0293,0.19483827,3.445386e-07,-2.0867045e+33,5.8471132e-06,-0.22336684,1.9691273e+15,-6.2981145e-09,-6.3043605e+06,2.9438323e+38,-4.8987274e-29,1.02575865e-35,-5.3791393e-14,3.1984896e-14,-0.0027087238,-2.4997793e+28,4.3539557e+19,1.7830244e+21,-225.15442,-11.471054,-0.0018444138,-9.211328e+13,-8.0167857e-22,-88785.03,-8.480019e+15,-2.8793945e-21,-1.7236387e-23,-8.292595,-1.3420761e-14,2.6714036e-35,-2.6693663e-15,-0.0052280435,48.866375,5.3858304e-07,-5.0194014e-36,827726,-13.953986,-2.6995655e-17,1.2172024e-17,-3.0083635e-33,1.4198154e-15,-2.244507e-24,5.166529e-17,-3.335641e+24,4.7108237e+08,2.0602474e-32,2.344219e+24,1.7379508e-08,3.670296e-23,-2.3053763e+25,9.953676e-31,30.97987,-1.3450299e-38,15.446057,4.3370206e+26,18.580767,-2.0038951e-28,-5.702115e-26,-2.2805216e-31,-11734.308,2.3574587e-09,-2.5029376e-13,-0.0013743767,2.4910062e+34,1.1414531e+17,1.0054673e+10,-168030.39,-1.6843547e-31,-6.057697e-36,-2.7662675e+36,-26.424412,-3.4267672e+14,-2.6199677e-31,-4.7908554e-05,-2.0324258e+32,-2.886432e+31,4.500856e-20,3.9707752e-23,-1.2453457e+35,-1.78253e-25,-3.2106893e-24,2.3520093e-07,9.644719e-15]':::VECTOR AS col_239, (-5305413186402659845):::INT8 AS "_co?l_240", 'F4C0D6F5/406E8B15':::PG_LSN AS col_241 FROM d.public.seed AS "ta!b._87" WHERE (seed._inet::INET << seed._inet::INET)::BOOL) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND ((seed._bool OR (seed._string !~* seed._string)) AND seed._bool))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND ((seed._bool OR (seed._string !~* seed._string)) AND seed._bool)) to emit 48 events
    changefeed_test.go:1141: invalid regular expression: error parsing regexp: invalid escape sequence: ``\xg"``
    --- FAIL: TestChangefeedRandomExpressions/kafka (7.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1.2-rc release-blocker]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/f3c75a3a-4d52-4d9f-b3c0-5ba8f873f099?testReportRun=5&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [7917e91eadb2f622faadf635a3e13c384e42070e](https://github.com/cockroachdb/cockroach/commits/7917e91eadb2f622faadf635a3e13c384e42070e):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-640846228):::INT8, (-5208206897757348346):::INT8, 0.596921443939209:::FLOAT8, 1.401298464324817e-45:::FLOAT8, '-infinity':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '1996-03-20 06:33:38.000598+00':::TIMESTAMPTZ, '-70 years -8 mons -277 days -02:36:15.594156':::INTERVAL, true, (-219633504356688747.9):::DECIMAL, e'\'':::STRING, '\x1c8a841f1b1a':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-128):::INT8, NULL, 1.401298464324817e-45:::FLOAT8, 1.0:::FLOAT8, '1984-12-22':::DATE, NULL, '2009-11-25 06:04:57.000629+00':::TIMESTAMPTZ, '-72 years -2 mons -794 days -18:06:09.427574':::INTERVAL, false, 1.0:::DECIMAL, 'X':::STRING, '\x':::BYTES, '8ebe54d1-53ba-4eee-8327-4b3c8fb6c880':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[[["c", "lI2AmFAM", false, false, null], 1.4484210689571526, 0.5426043532233706, null], [false], true, "KC"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-9338):::INT8, 1:::INT8, (-32768):::INT8, 'NaN':::FLOAT8, '-Inf':::FLOAT8, '4714-11-24 BC':::DATE, '1972-05-13 01:23:52.000122':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-2.295886575482022244E+24):::DECIMAL, e'\x7f\x1bS+2JZ\t':::STRING, '\x0b3871eb':::BYTES, '8bbe8adf-c4ed-42c2-b809-5f3cf0f1bcdc':::UUID, '9687:a439:ca4c:132f:c5f1:e092:667f:783/68':::INET, '[[0.7546002666736815, null], [true, 1.8637892305315782, false], 0.2535264689356948, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 1:::INT8, (-6394737122537883719):::INT8, '+Inf':::FLOAT8, 'NaN':::FLOAT8, '-infinity':::DATE, '2028-04-30 06:42:12.000894':::TIMESTAMP, '1996-06-13 17:12:32.000271+00':::TIMESTAMPTZ, '45 years 846 days 07:31:58.492503':::INTERVAL, NULL, (-1.434226260599546696E+34):::DECIMAL, '"':::STRING, '\x27':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 696899806:::INT8, (-1759470889835438104):::INT8, (-1.5164581537246704):::FLOAT8, (-0.731947733956679):::FLOAT8, '1991-08-12':::DATE, '1984-04-12 01:25:24.000769':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '70 years 5 mons 48 days 18:16:02.650436':::INTERVAL, NULL, (-1.0):::DECIMAL, e'\x00':::STRING, '\xcb':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'c673:f49f:c532:b0a:96ea:342a:3acb:e64a':::INET, '[{"baz": true, "dmZMK": {"PLJjB3B": {"OEt5": null}, "a": false}, "wTTP4LBx": 0.420053652770824}, {"Sqz3UMl08": true, "bIh": [false]}, "ffsMLj", true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-21711):::INT8, 1:::INT8, NULL, 0.45461106300354004:::FLOAT8, 1.9915058393901903:::FLOAT8, 'infinity':::DATE, '1999-04-27 04:27:18.000573':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, NULL, false, (-4.76747621861092104E+25):::DECIMAL, '':::STRING, NULL, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 477959497:::INT8, 127:::INT8, 0.0:::FLOAT8, 'NaN':::FLOAT8, '1973-03-07':::DATE, '2020-12-08 10:53:22.000473':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '68 years 2 mons 517 days 24:16:08.861158':::INTERVAL, true, (-1.481088084040991344E+30):::DECIMAL, e'O\x1f\x1c*\x0b[;f':::STRING, '\x':::BYTES, '9f5bfd34-4169-444b-9c4b-8d19fd34488e':::UUID, '8b8a:4e2f:31c6:6a7e:6ba5:8fa9:3990:37a1/72':::INET, '{"94XB6": {"c": false}, "bar": 0.08260941035469338, "c": true, "yzGgW1Q": false}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (24212:::INT8, 1:::INT8, 1:::INT8, 0.0:::FLOAT8, 1.0:::FLOAT8, '5874897-12-31':::DATE, '1990-09-16 04:15:08.00008':::TIMESTAMP, '1994-06-17 01:12:29.000321+00':::TIMESTAMPTZ, '53 years 5 mons 278 days 16:17:22.468945':::INTERVAL, true, (-32502976.49954690343):::DECIMAL, e'R_LD\x1et*[\t':::STRING, '\x7d':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 584402588:::INT8, (-1):::INT8, (-0.9985977411270142):::FLOAT8, 0.9555209539223892:::FLOAT8, '1987-08-28':::DATE, '1999-12-06 09:19:38.000018':::TIMESTAMP, '1976-03-20 19:46:09.000887+00':::TIMESTAMPTZ, '65 years 2 mons 582 days 21:17:09.791185':::INTERVAL, false, 463957491.5247195922:::DECIMAL, e'\U00002603':::STRING, '\x00':::BYTES, '2fa646ab-2f5c-4e74-ac7e-d5671b247dda':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[[0.257882341198081, false, true], [{"ZXowI4SWI": true}, null, false], false, true, 0.06622745853550271, 0.4774382052726832]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (22621:::INT8, 32767:::INT8, (-4842944780597835175):::INT8, '+Inf':::FLOAT8, '+Inf':::FLOAT8, '1994-12-02':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 8176905.781015033184:::DECIMAL, e'uv\x1a(cC\x1eO':::STRING, '\x22':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[[null, true, "JP5rgCW", "iT7gPR3zr"], [false], false, 0.19191616491421648]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-24959):::INT8, 1:::INT8, 127:::INT8, 0.0:::FLOAT8, 5e-324:::FLOAT8, NULL, '1991-12-25 18:29:54.00048':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '-53 years -973 days -14:14:59.534219':::INTERVAL, true, NULL, 'I':::STRING, '\x27':::BYTES, '3369cbe8-9b65-4430-a3b5-410aa6156071':::UUID, '25.139.219.64/28':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, 1:::INT8, (-9223372036854775807):::INT8, 3.4028234663852886e+38:::FLOAT8, (-0.14918193260809054):::FLOAT8, '5874897-12-31':::DATE, '2026-05-22 07:13:47.000063':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-5439.083742154038342):::DECIMAL, 'X':::STRING, '\xd2fd411f2ca78eb2':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"LxJj2m5": {"6Z": "foobar", "c": true, "v2": {"9wH5dua": null, "JMi": null, "baz": null}}, "a": false, "bar": true, "c": {"sbba": null}, "gadDk": [{"VprqDv4AM": 0.017567144719356775, "XlZn1a": "WN0k", "b": null}, false], "yubTbt": {"bar": false, "ydJpO": null}}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-2147483648):::INT8, (-2358582934367810483):::INT8, (-0.2239374816417694):::FLOAT8, 'NaN':::FLOAT8, '-infinity':::DATE, '2016-12-07 16:11:09.000371':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '290 years':::INTERVAL, false, (-4.140084630616985462E+28):::DECIMAL, e'\'':::STRING, '\x22':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '227.116.143.65/30':::INET, 'true':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, (-1):::INT8, (-2147483648):::INT8, 2.578482151031494:::FLOAT8, 0.990002272568898:::FLOAT8, '5874897-12-31':::DATE, '1999-01-28 22:38:14.000593':::TIMESTAMP, '2021-11-09 20:38:15.000997+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-444.6565515885129436):::DECIMAL, e'\x14]1:g4':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '253.88.80.234/5':::INET, '{"4ffH": 0.1398289419050902}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (17026:::INT8, (-32768):::INT8, 2147483647:::INT8, (-1.0):::FLOAT8, 1.0:::FLOAT8, '-infinity':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2008-03-31 01:11:11.000018+00':::TIMESTAMPTZ, '-11 years -150 days -05:46:29.2644':::INTERVAL, false, 'NaN':::DECIMAL, e'\x00':::STRING, '\x00':::BYTES, 'bae1765e-0289-499a-809a-ac8ee71c6699':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (32767:::INT8, 127:::INT8, (-9223372036854775807):::INT8, '+Inf':::FLOAT8, (-1.2044126607630856):::FLOAT8, 'infinity':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2002-04-13 07:46:49.000217+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 1.0:::DECIMAL, '"':::STRING, '\x7a7ddb07a3':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[1.583315990218754, [{"foo": "baz", "l8ShaC0hN": 0.7164494627490645}, {"0zdrgcjY": 0.33439720916992816, "4nHJL": "P5NLB"}, false], "YI6", 0.5735658595782984]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, NULL, 1:::INT8, 0.5080723166465759:::FLOAT8, 0.5609831316865346:::FLOAT8, '1970-01-01':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2015-02-18 07:32:18.000999+00':::TIMESTAMPTZ, '1 day':::INTERVAL, true, 'Infinity':::DECIMAL, e'\x1e':::STRING, '\xe29883':::BYTES, '26a1b7fa-c45a-445a-9872-113799ae88fc':::UUID, '0.0.0.0/0':::INET, '["foobar", 1.0264323214918993, 2.150850001652179]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (14297:::INT8, 0:::INT8, 2142467738598419176:::INT8, NULL, 1.102715696489447:::FLOAT8, '1994-12-31':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '2 years 8 mons 673 days 08:25:55.685088':::INTERVAL, true, (-1):::DECIMAL, '"':::STRING, '\x22':::BYTES, '0525d049-178f-48c6-8abb-7050d37ef8c9':::UUID, '142.149.60.238/30':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 1775517509:::INT8, (-1915056282129118503):::INT8, '-Inf':::FLOAT8, (-0.0):::FLOAT8, '1970-01-01':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '32 years 1 mon 516 days 11:06:16.614405':::INTERVAL, false, 1.0:::DECIMAL, '':::STRING, '\xe29883':::BYTES, '82587886-c15b-4a8e-a7af-3045402db625':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-9678):::INT8, (-429359960):::INT8, NULL, 'NaN':::FLOAT8, (-1.137194814479531):::FLOAT8, '1993-04-11':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-7.700209096399050047E+33):::DECIMAL, e'gp7\x01bl"9C':::STRING, '\xff':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, (-1699022963):::INT8, 9127409494421254667:::INT8, 0.0:::FLOAT8, 0.4735741094045104:::FLOAT8, '1989-03-05':::DATE, '2010-09-27 23:12:31.000335':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, NULL, true, NULL, e'\'':::STRING, '\xf494512952b9a8':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-28048):::INT8, 32767:::INT8, (-7224251794364407001):::INT8, '+Inf':::FLOAT8, '-Inf':::FLOAT8, 'infinity':::DATE, '1994-11-26 16:56:21.000421':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '-53 years -5 mons -330 days -19:12:09.947378':::INTERVAL, false, 7.591172704534166688E+27:::DECIMAL, e'\U00002603':::STRING, '\xb5c2872a1e':::BYTES, '0930725e-a652-4526-a088-d39197351259':::UUID, '0.2.2.19/2':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, NULL, 1724077099250464414:::INT8, NULL, 0.0:::FLOAT8, '5874897-12-31':::DATE, '2005-05-28 01:45:57.000997':::TIMESTAMP, '2020-06-23 22:51:57.000305+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 1.0:::DECIMAL, '':::STRING, '\xa0c99108':::BYTES, '33b5ff64-635d-419e-af07-53d46c5ab1b3':::UUID, '89.95.184.184/21':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, 2147483647:::INT8, NULL, 0.4510746896266937:::FLOAT8, 2.303269122162093:::FLOAT8, '1970-01-01':::DATE, '2008-03-12 13:23:08.000023':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '17 years 3 mons 145 days 06:51:47.253534':::INTERVAL, false, 7115036531.2734125:::DECIMAL, '"':::STRING, '\xff':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, NULL, '[{"4Kt2": 0.45946014976897837, "bar": {"bar": 0.24202991149480976, "foobar": null}, "ipjkeY": {"aiuF": true, "c": true}, "kc6btRFd": {"BGh": null}, "sVtEsOE": {"b": null, "buF16Y": false}}, 1.1619923214712533, 0.1706120115957788, true]':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '0105000060E610000000000000':::GEOGRAPHY::GEOGRAPHY IN (SELECT st_geographyfromtext((seed._string::STRING || seed._date::DATE)::STRING::STRING, seed._int8::INT8)::GEOGRAPHY::GEOGRAPHY AS "c\\x82ol_15" FROM d.public.seed AS "ta
b _7" WHERE false ORDER BY seed._string)
    changefeed_test.go:1120: Skipping predicate '0105000060E610000000000000':::GEOGRAPHY::GEOGRAPHY IN (SELECT st_geographyfromtext((seed._string::STRING || seed._date::DATE)::STRING::STRING, seed._int8::INT8)::GEOGRAPHY::GEOGRAPHY AS "c\\x82ol_15" FROM d.public.seed AS "ta
b _7" WHERE false ORDER BY seed._string) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 44 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._uuid::UUID IN (SELECT '6f73ff4b-e48c-46b5-be3f-68fbc1cb15a4':::UUID::UUID AS col_48 FROM d.public.seed AS "ta"" b_17" WHERE bool(seed._int4::INT8)::BOOL)) AND (seed._string !~ seed._string))
    changefeed_test.go:1120: Skipping predicate ((seed._uuid::UUID IN (SELECT '6f73ff4b-e48c-46b5-be3f-68fbc1cb15a4':::UUID::UUID AS col_48 FROM d.public.seed AS "ta"" b_17" WHERE bool(seed._int4::INT8)::BOOL)) AND (seed._string !~ seed._string)) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string NOT SIMILAR TO e'Rv\x18P\tj,':::STRING
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string NOT SIMILAR TO e'Rv\x18P\tj,':::STRING to emit 94 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e'\\x449d8a79a27255':::BYTES::BYTES NOT IN (SELECT seed._bytes::BYTES AS col_116 FROM d.public.seed AS tab_34 JOIN d.public.seed AS "ta'b_35" ON (tab_34._jsonb) = ("ta'b_35"._jsonb) WHERE true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE e'\\x449d8a79a27255':::BYTES::BYTES NOT IN (SELECT seed._bytes::BYTES AS col_116 FROM d.public.seed AS tab_34 JOIN d.public.seed AS "ta'b_35" ON (tab_34._jsonb) = ("ta'b_35"._jsonb) WHERE true) to emit 93 events
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('/5':::STRING::STRING % e'\x04(\x1fX7':::STRING::STRING)::BOOL
    changefeed_test.go:1120: Skipping predicate ('/5':::STRING::STRING % e'\x04(\x1fX7':::STRING::STRING)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e' +j^D#\x1a\x1c+':::REFCURSOR::REFCURSOR IN ('_':::REFCURSOR,)
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e' +j^D#\x1a\x1c+':::REFCURSOR::REFCURSOR IN ('_':::REFCURSOR,) because error pq: unsupported comparison operator: refcursor IN tuple{refcursor}
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._bool OR (st_issimple(st_force2d('01020000000200000088C83C97453AD3C15830FC8B0BE0F241063304CC4B3DF5C110384A950B4FFF41':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL OR seed._bool)))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._bool OR (st_issimple(st_force2d('01020000000200000088C83C97453AD3C15830FC8B0BE0F241063304CC4B3DF5C110384A950B4FFF41':::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL OR seed._bool))) to emit 44 events
    changefeed_test.go:1141: st_issimple(): geos: no locations to init GEOS
    --- FAIL: TestChangefeedRandomExpressions/kafka (2.49s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=5</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/d9cde2fd-81ff-425b-aa2e-2d6422e24117?testReportRun=9&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [9e48858c6c8b22af4ec1159bcff6e233e7bfddff](https://github.com/cockroachdb/cockroach/commits/9e48858c6c8b22af4ec1159bcff6e233e7bfddff):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (19068:::INT8, 127:::INT8, 9223372036854775807:::INT8, NULL, 0.7014923273238843:::FLOAT8, '1970-01-01':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '1992-01-28 02:46:52.000333+00':::TIMESTAMPTZ, '-77 years -8 mons -590 days -19:23:32.433748':::INTERVAL, false, (-1.234E+401):::DECIMAL, e'd$\ne[@':::STRING, '\x00':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '6b60:a74a:eec3:3d48:8213:58cf:4b4c:c9de/67':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-2147483648):::INT8, 1221162541647122670:::INT8, (-1.7818645238876343):::FLOAT8, 1.0478983287162198:::FLOAT8, '5874897-12-31':::DATE, '1989-07-16 22:37:16.000044':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, true, (-746.8566079228288262):::DECIMAL, '"':::STRING, '\xe29883':::BYTES, 'c64bdb00-6ac4-443b-8e38-ef4a21fdf918':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (22819:::INT8, (-1):::INT8, 1921650592892464692:::INT8, 1.401298464324817e-45:::FLOAT8, NULL, '1970-01-01':::DATE, '1981-09-27 06:57:15.000077':::TIMESTAMP, NULL, '290 years':::INTERVAL, true, (-4.032124230591880161):::DECIMAL, e'\x00':::STRING, '\xe29883':::BYTES, '34490e3a-fc5a-4d02-8a72-b215a09731ef':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, (-32768):::INT8, 2147483647:::INT8, '+Inf':::FLOAT8, NULL, NULL, '2006-08-02 19:52:40.000857':::TIMESTAMP, '2024-03-03 14:04:04.000084+00':::TIMESTAMPTZ, '82 years 1 mon 747 days 06:09:42.53897':::INTERVAL, false, 6.5558561309669543E+32:::DECIMAL, '':::STRING, '\x8603ddbd5728b4':::BYTES, 'f6a542b3-358f-4d17-a656-1db020b3b6c6':::UUID, 'acc3:6093:c644:3584:5635:9294:e8b:d3d2/3':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-28957):::INT8, 32767:::INT8, 8425399000461208426:::INT8, 1.7214328050613403:::FLOAT8, 1.401298464324817e-45:::FLOAT8, 'infinity':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '30 years 6 mons 8 days 17:27:57.565393':::INTERVAL, true, (-1.0):::DECIMAL, '!SL':::STRING, '\x1d5f3d50b350425215':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[{"Cd": {"5U": 0.5703034166522867, "c702FQG7": {"xvswQxvL": null}, "foo": false, "wkM67kzr": ["foo"]}, "OWl": "IuWlHooj", "baz": false, "eJkjDT": {"52vuM": null, "a": [0.7187205229415631, true], "c": {"KPA": true, "bar": true, "c": "oCnG6Im", "cGAYCjfjP": {"foo": null}}, "foo": "foo"}}, [{"0eF2A8": 0.9852244269317637, "2MgU": "f1", "DeD9ObifS": [[{"8agFGmn": "GS9XkApn"}, 2.324137063943965, false], null, null], "t5LWH": {"c": true}}, "mLbiIa0"], {"b": true, "foobar": false}, 2.35032378011595, 1.421343915740682, "zNOJeF"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 32767:::INT8, 5899900201805499270:::INT8, 0.5357957482337952:::FLOAT8, 0.8988384167038854:::FLOAT8, 'infinity':::DATE, '1970-08-21 13:50:31.000416':::TIMESTAMP, '1986-04-08 23:25:45.000341+00':::TIMESTAMPTZ, '290 years':::INTERVAL, true, 3.891448683658114779E+23:::DECIMAL, NULL, '\xd6b4':::BYTES, 'b5b1fa13-aa98-47fe-b916-211f421b7623':::UUID, '131.140.45.132/31':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (32767:::INT8, 1:::INT8, 0:::INT8, (-0.7468724846839905):::FLOAT8, NULL, 'infinity':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '2026-08-30 12:36:35.000355+00':::TIMESTAMPTZ, '52 years 10 mons 714 days 24:02:12.950054':::INTERVAL, true, (-5.355263455455206501E+28):::DECIMAL, e'\'':::STRING, '\x27':::BYTES, 'e67c2527-fc2e-4543-b029-eece67a4d5df':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[{"baz": true}, [{"bar": null}, "9zkV"], true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 310051984:::INT8, 0:::INT8, 0.06826823204755783:::FLOAT8, (-2.032604676052187):::FLOAT8, '-infinity':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-6.829333795827972245):::DECIMAL, e'}f\x174\x04:':::STRING, '\xe87845':::BYTES, 'fae5b9cb-8eba-48cf-9ef2-847870e59a2a':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-9456):::INT8, 127:::INT8, 0:::INT8, 1.0:::FLOAT8, (-0.0):::FLOAT8, '-infinity':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-28687030661506319.16):::DECIMAL, '"':::STRING, '\x344cefd31131':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{"BQzSxNM": [false], "b": 0.7128874048895545}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, NULL, 3368800846461225779:::INT8, '+Inf':::FLOAT8, 0.2934283857042308:::FLOAT8, '5874897-12-31':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '2017-11-03 09:02:13.000874+00':::TIMESTAMPTZ, '-46 years -2 mons -770 days -12:30:44.626473':::INTERVAL, true, (-2.171579612094919561E+36):::DECIMAL, '"':::STRING, '\xff':::BYTES, '11043c6d-0263-4be6-9e85-606ca74570d6':::UUID, '6c13:d25b:3e00:908a:a050:e312:37d7:9f46/33':::INET, '[[{"VIJXKraJ": null, "a": true}, false, {"b": false}, false], [null], {"eD": "zBJvMdzCo"}, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1371:::INT8, 1376675496:::INT8, (-2147483648):::INT8, 0.0:::FLOAT8, NULL, 'infinity':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '1987-08-29 15:06:26.000023+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, true, (-8163880516333138306):::DECIMAL, e'\U00002603':::STRING, '\x41':::BYTES, '040251e6-fc5b-495c-b95e-dd750441fe74':::UUID, '132.208.216.227/14':::INET, '[[false], null, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-5638):::INT8, (-1):::INT8, 32767:::INT8, '+Inf':::FLOAT8, '-Inf':::FLOAT8, '1973-02-02':::DATE, '1979-05-19 04:06:26.00039':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '9 years 3 mons 923 days 24:34:10.723368':::INTERVAL, true, 392518562706076375:::DECIMAL, e'?p\x17\x1a':::STRING, '\x4831b555f7f51526':::BYTES, 'c99d92e4-d0bc-48fb-8fdf-6303714e33cc':::UUID, '66ff:8989:b87c:d9a:686d:5c5d:4eda:9154/12':::INET, '[{"b36JJ": null, "bar": 0.18924980895540927}, false, 0.24668903019348232, "7cyx00I", true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (20295:::INT8, (-1722472790):::INT8, (-3666542968485808057):::INT8, 'NaN':::FLOAT8, (-0.737359461684166):::FLOAT8, '4714-11-24 BC':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2010-10-12 22:34:49.000856+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 1:::DECIMAL, '':::STRING, '\x16cd2225c86c':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, NULL, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-26692):::INT8, (-1935128141):::INT8, (-9115656869562611663):::INT8, 0.12614309787750244:::FLOAT8, (-0.10465651164133255):::FLOAT8, '1970-01-01':::DATE, '2002-01-15 16:51:12.00099':::TIMESTAMP, '2020-11-07 16:39:24.000723+00':::TIMESTAMPTZ, '1 day':::INTERVAL, true, (-7.270809556938363921E+34):::DECIMAL, '':::STRING, '\xec7f':::BYTES, NULL, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-722168251):::INT8, (-2147483648):::INT8, '+Inf':::FLOAT8, 0.43636503681206573:::FLOAT8, '1978-06-18':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, true, 1.343526502991143349:::DECIMAL, '=':::STRING, '\x56be9cbd24fa96':::BYTES, 'bc7f1dc6-6a4e-465a-94c6-ab3c6735d7b7':::UUID, '228.88.126.96/29':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 1:::INT8, 4926776731483964265:::INT8, 1.0317150354385376:::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '4714-11-24 BC':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '-6 years -4 mons -427 days -10:30:55.991285':::INTERVAL, false, (-1.0):::DECIMAL, e'\'':::STRING, '\x4db5e9ab51ac':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[[null], "soUJ"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-3800):::INT8, 127:::INT8, (-128):::INT8, (-1.0):::FLOAT8, 0.16328868773718708:::FLOAT8, NULL, '1983-05-28 14:32:44.000593':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-1.667944082764946980E+28):::DECIMAL, e'\U00002603':::STRING, '\x00':::BYTES, 'be7258d7-95ce-4a6a-80f8-70751683f23e':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"test": "json"}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL LIKE e'JC\x05\x1e':::STRING
    changefeed_test.go:1120: Skipping predicate NULL LIKE e'JC\x05\x1e':::STRING because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 47 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string !~* seed._string
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string !~* seed._string because error pq: invalid regular expression: error parsing regexp: missing closing ): ``(?i:P.�\)``
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (st_iscollection('0103000000010000000A000000284B95A8B62DFCC190E930E590A2C1C1EEFC571AC9B5FFC19612A3EAA19E01C200B665C857CA7EC168074821E4A601C29097D927C8DFEF418BF72D8F51DFF0C1E04C7BD0BAE6F54140F0661A2BD9B5C1C033217FCB04FC4118102D91FC64DD41742177AD9F67EE41147DA2A129570042406FE66DEDBBA74170E0D8CC19AFEC41F68D8DC92310FFC184665FA4CCBEF941284B95A8B62DFCC190E930E590A2C1C1':::GEOMETRY::GEOMETRY)::BOOL AND false)
    changefeed_test.go:1120: Skipping predicate (st_iscollection('0103000000010000000A000000284B95A8B62DFCC190E930E590A2C1C1EEFC571AC9B5FFC19612A3EAA19E01C200B665C857CA7EC168074821E4A601C29097D927C8DFEF418BF72D8F51DFF0C1E04C7BD0BAE6F54140F0661A2BD9B5C1C033217FCB04FC4118102D91FC64DD41742177AD9F67EE41147DA2A129570042406FE66DEDBBA74170E0D8CC19AFEC41F68D8DC92310FFC184665FA4CCBEF941284B95A8B62DFCC190E930E590A2C1C1':::GEOMETRY::GEOMETRY)::BOOL AND false) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT 1.0667773798989921:::FLOAT8 AS "co	l
_84" FROM (SELECT true AS col_70, seed._decimal AS "c%pol_71", '-Inf':::FLOAT8 AS col_72, NULL AS col_73, seed._int8 AS " col_74", seed._uuid AS col_75, seed._float8 AS "%7bcol_76" FROM d.public.seed AS _tab_33 JOIN d.public.seed AS tab_34 ON (_tab_33._float8) = (tab_34._float8) WHERE seed._bool) AS tab_35 (col_77, col_78, col_79, ")col_80", col_81, col_82, "c�ol_83") WHERE (('151.229.246.56/30':::INET::INET | (NULL::INET + seed.col_81::INT8)::INET::INET)::INET::INET << hostmask('197.13.248.112/8':::INET::INET)::INET::INET)::BOOL)
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT 1.0667773798989921:::FLOAT8 AS "co	l
_84" FROM (SELECT true AS col_70, seed._decimal AS "c%pol_71", '-Inf':::FLOAT8 AS col_72, NULL AS col_73, seed._int8 AS " col_74", seed._uuid AS col_75, seed._float8 AS "%7bcol_76" FROM d.public.seed AS _tab_33 JOIN d.public.seed AS tab_34 ON (_tab_33._float8) = (tab_34._float8) WHERE seed._bool) AS tab_35 (col_77, col_78, col_79, ")col_80", col_81, col_82, "c�ol_83") WHERE (('151.229.246.56/30':::INET::INET | (NULL::INET + seed.col_81::INT8)::INET::INET)::INET::INET << hostmask('197.13.248.112/8':::INET::INET)::INET::INET)::BOOL) because error pq: column "seed.col_81" does not exist
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_astext('01020000C000000000':::GEOMETRY::GEOMETRY)::STRING !~* seed._string
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_astext('01020000C000000000':::GEOMETRY::GEOMETRY)::STRING !~* seed._string because error pq: invalid regular expression: error parsing regexp: missing closing ): ``(?i:P.�\)``
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 46 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._bool OR (CASE WHEN true THEN seed._string ELSE seed._string END !~* seed._string)))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._bool OR (CASE WHEN true THEN seed._string ELSE seed._string END !~* seed._string))) to emit 47 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1141: invalid regular expression: error parsing regexp: missing closing ): ``(?i:@A��C(o)``
    --- FAIL: TestChangefeedRandomExpressions/kafka (11.81s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.2.0-rc release-blocker]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1.3-rc release-blocker]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16632867?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16632867?buildTab=artifacts#/) on master @ [53a31d236663047edb2966a73749a8842be25bd5](https://github.com/cockroachdb/cockroach/commits/53a31d236663047edb2966a73749a8842be25bd5):


```
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (('c4b5:baab:142c:b777:cae0:4d79:6433:2de9':::INET::INET >> '49c5:46f8:ccf6:d9c0:a86f:a45f:5ab1:c6e/97':::INET::INET)::BOOL OR (seed._string NOT LIKE seed._string)) to emit 2 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (false AND seed._bool)
    changefeed_test.go:1120: Skipping predicate (false AND seed._bool) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string ~* e'6\x0f\x11S\f\x05\x16\x0b\x14':::STRING
    changefeed_test.go:1120: Skipping predicate seed._string ~* e'6\x0f\x11S\f\x05\x16\x0b\x14':::STRING because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT seed._bool) to emit 49 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE B'01100101011110011100000001101001'::BIT(32) IN (B'01101101111111011010110111001000', CASE WHEN COALESCE(true, ('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET::INET >> inet(e':jof\x05A/':::STRING::STRING)::INET::INET)::BOOL) THEN B'11011010110100110011000111000001' ELSE B'10101000100110100001001100011011' END, B'01011011110101100000101100000100')
    changefeed_test.go:1120: Skipping predicate B'01100101011110011100000001101001'::BIT(32) IN (B'01101101111111011010110111001000', CASE WHEN COALESCE(true, ('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET::INET >> inet(e':jof\x05A/':::STRING::STRING)::INET::INET)::BOOL) THEN B'11011010110100110011000111000001' ELSE B'10101000100110100001001100011011' END, B'01011011110101100000101100000100') because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_generatepoints('010400000000000000':::GEOMETRY::GEOMETRY, seed._int8::INT4, seed._int8::INT4)::GEOMETRY::GEOMETRY NOT IN (SELECT '01070000000400000001040000000300000001010000009EF4E0A2B89900426E564508A26E01C20101000000E3B3EE8E200FF2C13CC476D659C1F7C10101000000FA71AEA01FDCF641FE6399D97445BDC1010200000003000000C19475C0E8E301C2A295377F35E8FBC1DF684A802C2D0042B8F5EF829C63F3C14C5387BFA368FF41BD969E4E5FF3E34101010000009601D4C55CEDF2C1FCE016DADA82FF410105000000090000000102000000020000006B8C61BF3C04E54191F50A19F8A7F6C10169C0818E79ED41A33DC41BD1D2BFC1010200000005000000562C8ADA5F49F941F58B2FBEB792FDC1F89992EF5F9B00428817B313973401C247091215F5C8F1419B8CB81FA0960242F9B208208804F141364055AA0373F741F126A8B548A5F041F0B3BDB140F7F8410102000000040000005FAE5237579DE941F80DE1EDAD1EBFC12A4E26C5736DF041DB9734CC0FC3E241CB2FBE1518F2F1C17852C701396AFA41776D4112E0B2ECC1C1EAA57D7651EA41010200000002000000B05DE84C2598EE41D80FB33D5637D5C18F0A00EC52D5FB4109D46DFF79BAF7410102000000020000000C635CB9BDE1D84124B8F3DA20BAE6C185CE0FC004D7F641FE349D0582B1FCC10102000000050000004AB1E12A090EF2C178A495F81928E2C1FAE2CCD1C4AE01C27031D93182D7F8C1D12046B5AD69ED41307909BBE04502C293B13187E35BF2410C6A82F380ABF1C18406BB42FC11F7411A32B23A469DE8C1010200000002000000DAE666EB1258F341B7D87D4CFE6ED1C151F0F826DD3AF6416E586199AA68FB410102000000040000000E476116C05600421B6BEF73E12FF4417EE8B2FE774FFC4176BE0EDFA8C9FD4105D0B48FF5F7F2418AA11B013F82FA41367D1FA1CB26FAC119A9532638E9FE410102000000030000007B995BC201DC01428EAC448F0520F541A0C804AA5E49D7410F4F8C7CB97B0242B56E4347CE7DC041694DF7970D470142':::GEOMETRY::GEOMETRY AS "co%c9l_119")
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_generatepoints('010400000000000000':::GEOMETRY::GEOMETRY, seed._int8::INT4, seed._int8::INT4)::GEOMETRY::GEOMETRY NOT IN (SELECT '01070000000400000001040000000300000001010000009EF4E0A2B89900426E564508A26E01C20101000000E3B3EE8E200FF2C13CC476D659C1F7C10101000000FA71AEA01FDCF641FE6399D97445BDC1010200000003000000C19475C0E8E301C2A295377F35E8FBC1DF684A802C2D0042B8F5EF829C63F3C14C5387BFA368FF41BD969E4E5FF3E34101010000009601D4C55CEDF2C1FCE016DADA82FF410105000000090000000102000000020000006B8C61BF3C04E54191F50A19F8A7F6C10169C0818E79ED41A33DC41BD1D2BFC1010200000005000000562C8ADA5F49F941F58B2FBEB792FDC1F89992EF5F9B00428817B313973401C247091215F5C8F1419B8CB81FA0960242F9B208208804F141364055AA0373F741F126A8B548A5F041F0B3BDB140F7F8410102000000040000005FAE5237579DE941F80DE1EDAD1EBFC12A4E26C5736DF041DB9734CC0FC3E241CB2FBE1518F2F1C17852C701396AFA41776D4112E0B2ECC1C1EAA57D7651EA41010200000002000000B05DE84C2598EE41D80FB33D5637D5C18F0A00EC52D5FB4109D46DFF79BAF7410102000000020000000C635CB9BDE1D84124B8F3DA20BAE6C185CE0FC004D7F641FE349D0582B1FCC10102000000050000004AB1E12A090EF2C178A495F81928E2C1FAE2CCD1C4AE01C27031D93182D7F8C1D12046B5AD69ED41307909BBE04502C293B13187E35BF2410C6A82F380ABF1C18406BB42FC11F7411A32B23A469DE8C1010200000002000000DAE666EB1258F341B7D87D4CFE6ED1C151F0F826DD3AF6416E586199AA68FB410102000000040000000E476116C05600421B6BEF73E12FF4417EE8B2FE774FFC4176BE0EDFA8C9FD4105D0B48FF5F7F2418AA11B013F82FA41367D1FA1CB26FAC119A9532638E9FE410102000000030000007B995BC201DC01428EAC448F0520F541A0C804AA5E49D7410F4F8C7CB97B0242B56E4347CE7DC041694DF7970D470142':::GEOMETRY::GEOMETRY AS "co%c9l_119") because error pq: integer out of range for type int4
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT true)
    changefeed_test.go:1120: Skipping predicate (NOT true) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_hasarc(st_force3dm(st_makeenvelope(COALESCE(seed._float8, seed._float8)::FLOAT8, NULL::FLOAT8, seed._float4::FLOAT8, seed._float4::FLOAT8, seed._int8::INT8)::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL
    changefeed_test.go:1120: Skipping predicate st_hasarc(st_force3dm(st_makeenvelope(COALESCE(seed._float8, seed._float8)::FLOAT8, NULL::FLOAT8, seed._float4::FLOAT8, seed._float4::FLOAT8, seed._int8::INT8)::GEOMETRY::GEOMETRY)::GEOMETRY::GEOMETRY)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_centroid('01020000E0E610000000000000':::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY NOT IN (st_project(st_centroid('01070000A0E61000000400000001060000A0E6100000040000000103000080010000000C00000017B51816E72435C09A193803483F45C077DC13CD63D4C5413E85A2E273593740CC640030C70951C0159464CB924D00C2687484CE3D735B40177CA51B33EB54C010519A1A99ECDDC140CDA7D7435166400964836D9CF853C0E9B22986A97CF0C1C31880914E385F40505691BDA2F83CC0FD3EB806B0E0F84111E3F2F35EBB5640F0E89639692BEC3FECB3D61CBEBDFAC1FF9B1FCAB54E64405FED0C6BB3B54140144072A5B03DE841BE7C8BFF1CE8534045B68F9F90613240BD0560B7BC4AF441AAF82F831FF35C40549FDBB612E75540970DEEB47725DCC1090ACDB8E28765C0E11849D91B0B564010BECD3FCF0EEC41851DA5B4571B5AC0F44F5724F24AD2BFC8210CAD9CC401C217B51816E72435C09A193803483F45C077DC13CD63D4C5410103000080010000000C00000036299A82C08B61C064F87275B2303CC0A9A98F364767A9416ED7FC4304205840C7B3B179F7B847C0E16FB0AC04A5F6C115E7791C5A0E55404E1E463B376C1BC0748D8D482A489CC124FBE0B7497066407A21DCEED93C4EC0E14471970A9DD4C1009BD1E2489962402A1E1E847AAE44C0FAEEF24194D3FCC1BE2E6065AAEA6540B12D7183DA415440DA828360075C014246F7FBDC85E351409C7E408D4AB82F400F20C65F6F3AF3C1B785922153164E40772A090AF9D445400227415595BE00425B85CA5E18234B400C464D2EDAAD4E40065F1AD997EEF5C10970868CDB27324082A1CCEAEEB554408754395BA30CF941DF8E6C94C5E23BC024ABA83CE6F72A40D063041D1B24F7C136299A82C08B61C064F87275B2303CC0A9A98F364767A941010300008001000000050000004D57A941D0345DC0649B78FA5B1A51C0BAD4F7756927E1C1CBC76BF7EC5A4E402A0D5D718AC428400F1B03D1638AE7C11E43A3BCCE9E2440AE327D5811382E406BB5D6FC0002FEC1382E68D5780D59C037D76744B3E35140651D851A542CECC14D57A941D0345DC0649B78FA5B1A51C0BAD4F7756927E1C101030000800100000006000000DE79C32DA8264CC036A2AB8ED57D4EC06091473969C5DE41FB75C62D31965240FB16B98A08C752C016BEA61CC717FAC1B31E49458F7B52C01E0B81FF6DEB4740A8B6C10FA578EAC1E66BA7EACF2962C0F86B76A7A33F54409C58836E8043EFC1E513902F18C064C0E0E91C6D0F3101407FC3FD83B6A8F241DE79C32DA8264CC036A2AB8ED57D4EC06091473969C5DE4101010000A0E6100000E972AB185A0A57C0E1D8D32A235920C0A76B1382222BE24101010000A0E61000004440177A1EE32C40F3AC2BFFE34F4140B6E807D82E00E4C101030000A0E6100000010000000A000000AD453BFEDFF765C0B4613F7B51B63240CAED187DF0A4D84178AE74FA4CBF62C0A692BBDB143D41C0A8BF88EF1C01C2418B6994030D2356407B26ECAFFBAC32C026449D850350FBC1C60BA95DBDDC614079E0806C8175FCBF54EF7915CDD9EF412069DDE5A4044D40DBF3E4DAD9BD4140265062F4EF8DEB4187B9B13277B65F40DAC099EFBB2C4D406AEB6C01746F02C2F99185607C155A40AF3DF5FAA22D4F40D6295E7BE419FDC14FC8082CDC075AC050D11EC09A1D5040A9B08F05A92100C24C5B19C07A5963C0F33B1B2871465440629B12DACBACF441AD453BFEDFF765C0B4613F7B51B63240CAED187DF0A4D841':::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY, 0.8161051597037724:::FLOAT8::FLOAT8, (-0.4534231086902766):::FLOAT8::FLOAT8)::GEOGRAPHY, st_s2covering('01020000E0E6100000070000009757A12650AE5CC0BAA9C48F49294CC02895C522A255C54179748F19BEA7FCC1D485A48121DA51C00991DAFF9C6D53C0AA9EFCC63CEAF24129A20934FD52FC41B67F858866EC5E40AF5A24CF78EA4FC060DE2487A947F6C1C9D067A68E86ECC1636C3D25A04D63405855CE19E3612FC0DDDDC9E7BA9BD841DD7BD12862F300C253CEC6AFDDCD64407BB51FA55A4536404BE321179BA9F6C111D901E268BEFA4154D681AFE7AF53409507665C36AE3640FA32A4B52CDCD0C19E8D023ED54CFC41D69C8BA93B784E4060350531D51538405E3A3C8B8FB9D941983010173099EFC1':::GEOGRAPHY::GEOGRAPHY, e't\x16Z\x1a':::STRING::STRING)::GEOGRAPHY)
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_centroid('01020000E0E610000000000000':::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY NOT IN (st_project(st_centroid('01070000A0E61000000400000001060000A0E6100000040000000103000080010000000C00000017B51816E72435C09A193803483F45C077DC13CD63D4C5413E85A2E273593740CC640030C70951C0159464CB924D00C2687484CE3D735B40177CA51B33EB54C010519A1A99ECDDC140CDA7D7435166400964836D9CF853C0E9B22986A97CF0C1C31880914E385F40505691BDA2F83CC0FD3EB806B0E0F84111E3F2F35EBB5640F0E89639692BEC3FECB3D61CBEBDFAC1FF9B1FCAB54E64405FED0C6BB3B54140144072A5B03DE841BE7C8BFF1CE8534045B68F9F90613240BD0560B7BC4AF441AAF82F831FF35C40549FDBB612E75540970DEEB47725DCC1090ACDB8E28765C0E11849D91B0B564010BECD3FCF0EEC41851DA5B4571B5AC0F44F5724F24AD2BFC8210CAD9CC401C217B51816E72435C09A193803483F45C077DC13CD63D4C5410103000080010000000C00000036299A82C08B61C064F87275B2303CC0A9A98F364767A9416ED7FC4304205840C7B3B179F7B847C0E16FB0AC04A5F6C115E7791C5A0E55404E1E463B376C1BC0748D8D482A489CC124FBE0B7497066407A21DCEED93C4EC0E14471970A9DD4C1009BD1E2489962402A1E1E847AAE44C0FAEEF24194D3FCC1BE2E6065AAEA6540B12D7183DA415440DA828360075C014246F7FBDC85E351409C7E408D4AB82F400F20C65F6F3AF3C1B785922153164E40772A090AF9D445400227415595BE00425B85CA5E18234B400C464D2EDAAD4E40065F1AD997EEF5C10970868CDB27324082A1CCEAEEB554408754395BA30CF941DF8E6C94C5E23BC024ABA83CE6F72A40D063041D1B24F7C136299A82C08B61C064F87275B2303CC0A9A98F364767A941010300008001000000050000004D57A941D0345DC0649B78FA5B1A51C0BAD4F7756927E1C1CBC76BF7EC5A4E402A0D5D718AC428400F1B03D1638AE7C11E43A3BCCE9E2440AE327D5811382E406BB5D6FC0002FEC1382E68D5780D59C037D76744B3E35140651D851A542CECC14D57A941D0345DC0649B78FA5B1A51C0BAD4F7756927E1C101030000800100000006000000DE79C32DA8264CC036A2AB8ED57D4EC06091473969C5DE41FB75C62D31965240FB16B98A08C752C016BEA61CC717FAC1B31E49458F7B52C01E0B81FF6DEB4740A8B6C10FA578EAC1E66BA7EACF2962C0F86B76A7A33F54409C58836E8043EFC1E513902F18C064C0E0E91C6D0F3101407FC3FD83B6A8F241DE79C32DA8264CC036A2AB8ED57D4EC06091473969C5DE4101010000A0E6100000E972AB185A0A57C0E1D8D32A235920C0A76B1382222BE24101010000A0E61000004440177A1EE32C40F3AC2BFFE34F4140B6E807D82E00E4C101030000A0E6100000010000000A000000AD453BFEDFF765C0B4613F7B51B63240CAED187DF0A4D84178AE74FA4CBF62C0A692BBDB143D41C0A8BF88EF1C01C2418B6994030D2356407B26ECAFFBAC32C026449D850350FBC1C60BA95DBDDC614079E0806C8175FCBF54EF7915CDD9EF412069DDE5A4044D40DBF3E4DAD9BD4140265062F4EF8DEB4187B9B13277B65F40DAC099EFBB2C4D406AEB6C01746F02C2F99185607C155A40AF3DF5FAA22D4F40D6295E7BE419FDC14FC8082CDC075AC050D11EC09A1D5040A9B08F05A92100C24C5B19C07A5963C0F33B1B2871465440629B12DACBACF441AD453BFEDFF765C0B4613F7B51B63240CAED187DF0A4D841':::GEOGRAPHY::GEOGRAPHY)::GEOGRAPHY::GEOGRAPHY, 0.8161051597037724:::FLOAT8::FLOAT8, (-0.4534231086902766):::FLOAT8::FLOAT8)::GEOGRAPHY, st_s2covering('01020000E0E6100000070000009757A12650AE5CC0BAA9C48F49294CC02895C522A255C54179748F19BEA7FCC1D485A48121DA51C00991DAFF9C6D53C0AA9EFCC63CEAF24129A20934FD52FC41B67F858866EC5E40AF5A24CF78EA4FC060DE2487A947F6C1C9D067A68E86ECC1636C3D25A04D63405855CE19E3612FC0DDDDC9E7BA9BD841DD7BD12862F300C253CEC6AFDDCD64407BB51FA55A4536404BE321179BA9F6C111D901E268BEFA4154D681AFE7AF53409507665C36AE3640FA32A4B52CDCD0C19E8D023ED54CFC41D69C8BA93B784E4060350531D51538405E3A3C8B8FB9D941983010173099EFC1':::GEOGRAPHY::GEOGRAPHY, e't\x16Z\x1a':::STRING::STRING)::GEOGRAPHY) because error pq: st_centroid(): unhandled geography type GeometryCollectionZ
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_coveredby('01050000E0E610000000000000':::GEOGRAPHY::GEOGRAPHY, st_centroid('01010000E0E6100000626B76F941A64FC074D651248BCF4EC0C1AEDC5F0C3DF5C165FAE2903145B4C1':::GEOGRAPHY::GEOGRAPHY, ('':::STRING ~ e'{\f\f\x1b\x1f\x10>?\x0f':::STRING)::BOOL)::GEOGRAPHY::GEOGRAPHY)::BOOL
    changefeed_test.go:1120: Skipping predicate _st_coveredby('01050000E0E610000000000000':::GEOGRAPHY::GEOGRAPHY, st_centroid('01010000E0E6100000626B76F941A64FC074D651248BCF4EC0C1AEDC5F0C3DF5C165FAE2903145B4C1':::GEOGRAPHY::GEOGRAPHY, ('':::STRING ~ e'{\f\f\x1b\x1f\x10>?\x0f':::STRING)::BOOL)::GEOGRAPHY::GEOGRAPHY)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE inet_contained_by_or_equals('1178:2a48:52b4:cb7e:648f:a690:9551:6e3/79':::INET::INET, '888f:10b5:56d6:6bfd:776d:8039:ebac:7f05/121':::INET::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate inet_contained_by_or_equals('1178:2a48:52b4:cb7e:648f:a690:9551:6e3/79':::INET::INET, '888f:10b5:56d6:6bfd:776d:8039:ebac:7f05/121':::INET::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT ARRAY[2365901421:::OID, 777044083:::OID, 3293163044:::OID, 1740164623:::OID] AS "c%qol�_241")
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT ARRAY[2365901421:::OID, 777044083:::OID, 3293163044:::OID, 1740164623:::OID] AS "c%qol�_241") to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL::INET << 'ee1:50d:1d5c:e832:f9fe:44eb:8572:c9a2/81':::INET::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (NULL::INET << 'ee1:50d:1d5c:e832:f9fe:44eb:8572:c9a2/81':::INET::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('234.5.6.157/13':::INET::INET << seed.col_285::INET)::BOOL
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('234.5.6.157/13':::INET::INET << seed.col_285::INET)::BOOL because error pq: column "seed.col_285" does not exist
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '1989-11-20':::DATE::DATE IN (SELECT '1987-07-01':::DATE::DATE AS col_297 FROM d.public.seed AS "Tab_111" WHERE seed._bool ORDER BY seed._enum DESC NULLS LAST)
    changefeed_test.go:1120: Skipping predicate '1989-11-20':::DATE::DATE IN (SELECT '1987-07-01':::DATE::DATE AS col_297 FROM d.public.seed AS "Tab_111" WHERE seed._bool ORDER BY seed._enum DESC NULLS LAST) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT st_dfullywithinexclusive('0101000040D5E0C0593137EAC16CBE23190703004282D644B2572AF8C1':::GEOMETRY::GEOMETRY, NULL::GEOMETRY, seed._float8::FLOAT8)::BOOL)
    changefeed_test.go:1120: Skipping predicate (NOT st_dfullywithinexclusive('0101000040D5E0C0593137EAC16CBE23190703004282D644B2572AF8C1':::GEOMETRY::GEOMETRY, NULL::GEOMETRY, seed._float8::FLOAT8)::BOOL) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._bool AND seed._bool) AND (seed._bool OR _st_dwithin('01020000A0E610000002000000BA406BC1A4B8584059325D32AFB1454021449B91530AD641D7A44CCC4A315040C677EC31220955401FB070382AA7FDC1':::GEOGRAPHY::GEOGRAPHY, '01050000E0E61000000300000001020000C0030000003F36D1922FA451C0B8C59AA030FA53C05825C3D0EE6DA0C18DC41A2D265DC9410EC5B2CF77506340CEAFAFED923A1E40E160399DEF88F141B82811D4465E01C2A176641C4B8C61C0DA169BEC8A0E45C0A506CCC8B04701C2B0E40B873CAB96C101020000C003000000710D0F19FBEF4C40545E507AC59F5140DF1438C1D53902C2E58344DFADA101C29C8CBB5B8D0A3D40A112F6FA502053406C8863E520A3FE417B7492ABF369F241A1E044EA8D2847C0F42F4A8E256351409D099827C8B9F1C1C93C697E9EB8C4C101020000C003000000B2FA027EFA2061400D593EF4E19D4EC0537DA1E65FE9E14187CBD4CF0928014226AA009868056140DB9B80ED3C8139C02F943FA53BFFFAC14D71762C2564D0C1C97A564AAA0B364024DB283AFA402B40687EC83C8344C041FC77658475A8ED41':::GEOGRAPHY::GEOGRAPHY, seed._float8::FLOAT8)::BOOL))
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((seed._bool AND seed._bool) AND (seed._bool OR _st_dwithin('01020000A0E610000002000000BA406BC1A4B8584059325D32AFB1454021449B91530AD641D7A44CCC4A315040C677EC31220955401FB070382AA7FDC1':::GEOGRAPHY::GEOGRAPHY, '01050000E0E61000000300000001020000C0030000003F36D1922FA451C0B8C59AA030FA53C05825C3D0EE6DA0C18DC41A2D265DC9410EC5B2CF77506340CEAFAFED923A1E40E160399DEF88F141B82811D4465E01C2A176641C4B8C61C0DA169BEC8A0E45C0A506CCC8B04701C2B0E40B873CAB96C101020000C003000000710D0F19FBEF4C40545E507AC59F5140DF1438C1D53902C2E58344DFADA101C29C8CBB5B8D0A3D40A112F6FA502053406C8863E520A3FE417B7492ABF369F241A1E044EA8D2847C0F42F4A8E256351409D099827C8B9F1C1C93C697E9EB8C4C101020000C003000000B2FA027EFA2061400D593EF4E19D4EC0537DA1E65FE9E14187CBD4CF0928014226AA009868056140DB9B80ED3C8139C02F943FA53BFFFAC14D71762C2564D0C1C97A564AAA0B364024DB283AFA402B40687EC83C8344C041FC77658475A8ED41':::GEOGRAPHY::GEOGRAPHY, seed._float8::FLOAT8)::BOOL)) to emit 43 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1141: _st_dwithin(): dwithin distance cannot be less than zero
    --- FAIL: TestChangefeedRandomExpressions/kafka (4.61s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.2.0-rc release-blocker]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1.3-rc release-blocker]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/97be0377-f949-482a-a6a1-baf2c181b25c?testReportRun=20&testReportShard=10&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [5b1702fc72dba825e94c60ac581f7f2bdc04d1fb](https://github.com/cockroachdb/cockroach/commits/5b1702fc72dba825e94c60ac581f7f2bdc04d1fb):


```
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_contains('01010000007C50F230861C02421C428B064665EE41':::GEOMETRY::GEOMETRY, NULL::GEOMETRY)::BOOL
    changefeed_test.go:1120: Skipping predicate st_contains('01010000007C50F230861C02421C428B064665EE41':::GEOMETRY::GEOMETRY, NULL::GEOMETRY)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._string::STRING % seed._string::STRING)::BOOL
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._string::STRING % seed._string::STRING)::BOOL to emit 44 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true AND true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true AND true) to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND seed._bool) to emit 41 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (EXISTS (SELECT false AS "c'ol_121" FROM d.public.seed AS "' ta b_53" WHERE seed._bool GROUP BY seed._int8) AND similar_to_escape(seed._string::STRING, '?X':::STRING::STRING, json_typeof('{"1WYm": null, "CDccLhSFP": [["smSpdT6lL"], true, 2.4474133702748433], "H4iioYM": [true], "c": null}':::JSONB::JSONB)::STRING::STRING)::BOOL)
    changefeed_test.go:1113: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (EXISTS (SELECT false AS "c'ol_121" FROM d.public.seed AS "' ta b_53" WHERE seed._bool GROUP BY seed._int8) AND similar_to_escape(seed._string::STRING, '?X':::STRING::STRING, json_typeof('{"1WYm": null, "CDccLhSFP": [["smSpdT6lL"], true, 2.4474133702748433], "H4iioYM": [true], "c": null}':::JSONB::JSONB)::STRING::STRING)::BOOL) because error pq: similar_to_escape(): invalid escape string
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (EXISTS (SELECT seed._string AS col_138, ARRAY[e'9V3````1\x1b\x12\x11':::STRING, e'\x07X]N\'':::STRING, e'u\rQoy':::STRING, e'G2\x0b_0':::STRING] AS col_139, seed._jsonb AS "!col_140" FROM d.public.seed AS tab_63 JOIN d.public.seed AS "Tab_64" ON (tab_63._jsonb) = ("Tab_64"._jsonb) JOIN d.public.seed AS tab_65 ON (tab_63._jsonb) = (tab_65._jsonb) WHERE (seed._string::STRING || seed._timestamptz::TIMESTAMPTZ)::STRING SIMILAR TO postgis_geos_version()::STRING ORDER BY seed._uuid NULLS LAST) AND false)
    changefeed_test.go:1120: Skipping predicate (EXISTS (SELECT seed._string AS col_138, ARRAY[e'9V3````1\x1b\x12\x11':::STRING, e'\x07X]N\'':::STRING, e'u\rQoy':::STRING, e'G2\x0b_0':::STRING] AS col_139, seed._jsonb AS "!col_140" FROM d.public.seed AS tab_63 JOIN d.public.seed AS "Tab_64" ON (tab_63._jsonb) = ("Tab_64"._jsonb) JOIN d.public.seed AS tab_65 ON (tab_63._jsonb) = (tab_65._jsonb) WHERE (seed._string::STRING || seed._timestamptz::TIMESTAMPTZ)::STRING SIMILAR TO postgis_geos_version()::STRING ORDER BY seed._uuid NULLS LAST) AND false) because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1120: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool OR true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool OR true) to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT COALESCE('':::VOID, '':::VOID) AS col_201, NULL AS col_202 FROM d.public.seed AS tab_77 ORDER BY seed.tableoid DESC NULLS FIRST)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT COALESCE('':::VOID, '':::VOID) AS col_201, NULL AS col_202 FROM d.public.seed AS tab_77 ORDER BY seed.tableoid DESC NULLS FIRST) to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (((seed._bool <= EXISTS (SELECT seed._int4 AS col_244 FROM d.public.seed AS tab_98 JOIN d.public.seed AS tab_99 ON (tab_98._jsonb) = (tab_99._jsonb) JOIN d.public.seed AS t_ab_100 ON ((tab_99._jsonb) = (t_ab_100._jsonb)) AND ((tab_98._jsonb) = (t_ab_100._jsonb)) JOIN d.public.seed AS tab_101 JOIN d.public.seed AS tab_102 ON (tab_101._jsonb) = (tab_102._jsonb) JOIN d.public.seed AS tab_103 ON (tab_102._jsonb) = (tab_103._jsonb) ON (((t_ab_100._inet) = (tab_103._inet)) AND ((t_ab_100.crdb_internal_mvcc_timestamp) = (tab_103.crdb_internal_mvcc_timestamp))) AND ((t_ab_100._int8) = (tab_103._int8)) JOIN d.public.seed AS "tab _104" ON ((((tab_103._int8) = ("tab _104"._int4)) AND ((t_ab_100._decimal) = ("tab _104".crdb_internal_mvcc_timestamp))) AND ((tab_99._jsonb) = ("tab _104"._jsonb))) AND ((tab_103._int4) = ("tab _104"._int4)))) OR (seed._bool OR seed._bool)) OR true)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (((seed._bool <= EXISTS (SELECT seed._int4 AS col_244 FROM d.public.seed AS tab_98 JOIN d.public.seed AS tab_99 ON (tab_98._jsonb) = (tab_99._jsonb) JOIN d.public.seed AS t_ab_100 ON ((tab_99._jsonb) = (t_ab_100._jsonb)) AND ((tab_98._jsonb) = (t_ab_100._jsonb)) JOIN d.public.seed AS tab_101 JOIN d.public.seed AS tab_102 ON (tab_101._jsonb) = (tab_102._jsonb) JOIN d.public.seed AS tab_103 ON (tab_102._jsonb) = (tab_103._jsonb) ON (((t_ab_100._inet) = (tab_103._inet)) AND ((t_ab_100.crdb_internal_mvcc_timestamp) = (tab_103.crdb_internal_mvcc_timestamp))) AND ((t_ab_100._int8) = (tab_103._int8)) JOIN d.public.seed AS "tab _104" ON ((((tab_103._int8) = ("tab _104"._int4)) AND ((t_ab_100._decimal) = ("tab _104".crdb_internal_mvcc_timestamp))) AND ((tab_99._jsonb) = ("tab _104"._jsonb))) AND ((tab_103._int4) = ("tab _104"._int4)))) OR (seed._bool OR seed._bool)) OR true) to emit 100 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1127: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate (seed._inet::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('179.15.228.72':::INET::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1120: Skipping predicate ('179.15.228.72':::INET::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1120: Skipping predicate false because it returned no rows
    changefeed_test.go:1108: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((((NOT inet_same_family('0.0.0.0/0':::INET::INET, seed._inet::INET)::BOOL) OR ((seed._string LIKE seed._string) AND ('s':::STRING !~ seed._string))) AND (seed._bool OR seed._bool)) AND seed._bool)
    changefeed_test.go:1124: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((((NOT inet_same_family('0.0.0.0/0':::INET::INET, seed._inet::INET)::BOOL) OR ((seed._string LIKE seed._string) AND ('s':::STRING !~ seed._string))) AND (seed._bool OR seed._bool)) AND seed._bool) to emit 38 events
    testfeed_test.go:304: creating external connection
    testfeed_test.go:307: ran create external connection
    changefeed_test.go:1141: invalid regular expression: error parsing regexp: unexpected ): ``�)�QP``
    --- FAIL: TestChangefeedRandomExpressions/kafka (4.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=20</code>
 - <code>shard=10</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.2.0-rc release-blocker]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.3-rc release-blocker]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/9ce0147d-f04b-473f-851b-3f11570fe67d?testReportRun=2&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c6122f6e6f0d35249a5eef8cab22db49dc43a626](https://github.com/cockroachdb/cockroach/commits/c6122f6e6f0d35249a5eef8cab22db49dc43a626):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 1:::INT8, (-2147483648):::INT8, 0.060239776968955994:::FLOAT8, (-0.1114157984916288):::FLOAT8, '1982-04-05':::DATE, '2023-03-21 13:08:19.000819':::TIMESTAMP, '2008-06-18 18:40:02.000991+00':::TIMESTAMPTZ, NULL, false, NULL, '8&':::STRING, '\x':::BYTES, 'cf96ad00-c32f-4c1b-ba73-26a0d6759f42':::UUID, '0.0.0.0/0':::INET, '{"a": [true, false, null], "c": true, "eK0J": true}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 121791460:::INT8, 127:::INT8, (-1.0684905052185059):::FLOAT8, 1.9529404003700115:::FLOAT8, '1979-07-16':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, NULL, '-17 years -7 mons -839 days -05:01:06.040085':::INTERVAL, true, 2334215413616890294:::DECIMAL, '"':::STRING, '\xa7':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 747646839:::INT8, (-7621554303837866378):::INT8, 0.15442048013210297:::FLOAT8, '+Inf':::FLOAT8, NULL, '2016-03-23 14:08:23.000405':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, false, 487623.8014355924487:::DECIMAL, e'\x1c~\x13~yW':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '185.53.175.125/12':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, (-1594059081):::INT8, 9223372036854775807:::INT8, '+Inf':::FLOAT8, (-1.0):::FLOAT8, '1989-12-29':::DATE, '1972-02-24 06:13:13.000436':::TIMESTAMP, '2006-08-13 22:57:33.000044+00':::TIMESTAMPTZ, '-35 years -1 mons -142 days -04:13:46.024756':::INTERVAL, false, NULL, 'X':::STRING, '\xff':::BYTES, 'db34be91-996c-484c-a9f6-58754df70def':::UUID, '121.250.89.42/21':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1422):::INT8, (-1129003848):::INT8, 234280082835456225:::INT8, NULL, (-1.0):::FLOAT8, '1987-08-12':::DATE, '2009-08-28 09:16:42.000835':::TIMESTAMP, '1988-02-01 18:33:44.000973+00':::TIMESTAMPTZ, '39 years 3 mons 806 days 07:31:05.354154':::INTERVAL, NULL, 5.796428159862765544E+35:::DECIMAL, e'\'':::STRING, '\x5dc5834336f8c08e':::BYTES, '4ce87d78-340c-487b-9872-dc56862933f4':::UUID, '161.79.88.175/17':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 2147483647:::INT8, (-3694796152193747053):::INT8, 0.31461602449417114:::FLOAT8, NULL, '1970-01-01':::DATE, '2026-02-16 09:58:01.000718':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '38 years 8 mons 497 days 19:36:01.701142':::INTERVAL, true, (-1.234E+401):::DECIMAL, e'\x00':::STRING, '\x60a7190c6a725ce58d':::BYTES, NULL, '148.183.38.164/1':::INET, '{"baz": [null], "mIVIHzs": {"AviHs0QQ": 1.4215326245807496, "a": true, "foo": {"b": {"PuHFJ": [null, null], "Whw": null}, "bar": [null, null, 0.227317682463288], "baz": 0.39713716165129936}, "foobar": true}, "sznapaiI": ["c", true], "wBbdjAV9": "foobar"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-1):::INT8, (-1):::INT8, 0.0:::FLOAT8, 1.401298464324817e-45:::FLOAT8, NULL, '0001-01-01 00:00:00':::TIMESTAMP, '1992-06-18 11:22:29.000106+00':::TIMESTAMPTZ, NULL, true, (-8.344393291420802408E+28):::DECIMAL, e'\x0b\b\x1e\x07\x7faF^m':::STRING, '\x27':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '85e2:ffd7:c6fb:b956:c669:88a3:8547:9ae3/48':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-32768):::INT8, 127:::INT8, 0.08917132765054703:::FLOAT8, '-Inf':::FLOAT8, '5874897-12-31':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '59 years 7 mons 98 days 11:37:08.012949':::INTERVAL, true, NULL, '"':::STRING, '\xe946':::BYTES, 'fc26b57f-031f-457a-b969-462e15e556f0':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"XwD8j": [[null], 1.7168506518721287], "bar": {"3WaGX": false}, "c": "a", "foo": "b", "foobar": {"c": "AebYSahWa", "foo": [{"Ah68flyS": false, "PfGJ": 0.14778165874965096}, false, 0.6094049558512301], "pDnhijDX": true, "yf": null}}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (9370:::INT8, 0:::INT8, NULL, '+Inf':::FLOAT8, 'NaN':::FLOAT8, '4714-11-24 BC':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, NULL, 0.01192867770598406531:::DECIMAL, '4':::STRING, NULL, '230075fc-48c3-4969-ba7d-f49423c20c6d':::UUID, '0.0.0.0/0':::INET, '{"b": {"b": false}, "zJulKXXH": false}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (4363:::INT8, 48685991:::INT8, (-2999948665167509030):::INT8, 1.0922958850860596:::FLOAT8, 1.0:::FLOAT8, '1990-01-11':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '2014-07-07 19:55:26.000489+00':::TIMESTAMPTZ, '290 years':::INTERVAL, true, (-1.0):::DECIMAL, '"':::STRING, '\x58':::BYTES, '24aa455b-3e4f-42ea-9941-9b3601767697':::UUID, '208.31.239.222/17':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-32768):::INT8, (-6608605728222089877):::INT8, NULL, 1.401298464324817e-45:::FLOAT8, '1970-01-01':::DATE, NULL, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, (-1):::DECIMAL, e'\U00002603':::STRING, '\x2c7ab0':::BYTES, NULL, '148.156.139.178/8':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, 1646058873:::INT8, (-2147483648):::INT8, 3.4028234663852886e+38:::FLOAT8, 1.401298464324817e-45:::FLOAT8, '1995-06-23':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2028-08-27 10:35:07.000527+00':::TIMESTAMPTZ, '-10 years -7 mons -133 days -16:22:24.569821':::INTERVAL, false, 2.548453752399953026E+22:::DECIMAL, e'\x06\x1c Hi?':::STRING, '\x58':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-20459):::INT8, NULL, 2147483647:::INT8, 0.30934709310531616:::FLOAT8, (-0.28322193285821307):::FLOAT8, '4714-11-24 BC':::DATE, '1992-11-01 02:44:55.000759':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-1.234E+401):::DECIMAL, '':::STRING, '\xe6795a':::BYTES, '51a8315f-a180-4ab8-ba49-5ce368c6ab30':::UUID, 'd9c5:91d2:93cb:8778:66d3:d360:47e3:5fe9/56':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, 695905556:::INT8, 127:::INT8, 3.4028234663852886e+38:::FLOAT8, 0.6771443118726221:::FLOAT8, '1975-03-26':::DATE, '2012-12-14 05:45:55.000068':::TIMESTAMP, '2032-02-19 23:55:08.000732+00':::TIMESTAMPTZ, '58 years 445 days 13:50:57.574385':::INTERVAL, true, 7.352399370937084494E+22:::DECIMAL, 'n9Ht':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (27487:::INT8, (-23568147):::INT8, (-81964701489859977):::INT8, 0.08818909525871277:::FLOAT8, (-0.5411112340940061):::FLOAT8, '1986-01-25':::DATE, '2023-02-03 10:24:21.000647':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '56 years 4 mons 653 days 02:21:33.636451':::INTERVAL, false, (-150.7634956152123955):::DECIMAL, e'\x1f\x0e':::STRING, '\x22':::BYTES, 'eb66eda7-f2f0-4559-a211-234c8ed86415':::UUID, NULL, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32558):::INT8, 1:::INT8, 8707407984371093019:::INT8, 0.0:::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '-infinity':::DATE, '2018-09-21 10:24:36.000921':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '72 years 8 mons 275 days 21:19:15.245887':::INTERVAL, true, 4336924910902672459:::DECIMAL, '':::STRING, '\x22':::BYTES, '84769d8d-a5e4-44e0-ab95-dbd987c7840e':::UUID, '8357:cfe6:5688:5df3:8691:4579:a2fb:30bb/18':::INET, '{"ENV08": {"SIF": null}, "Zgv": null, "sO": true}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 32767:::INT8, (-3490873076504227471):::INT8, 3.4028234663852886e+38:::FLOAT8, (-1.9254426934531026):::FLOAT8, '1980-07-10':::DATE, '2032-09-29 15:50:41.000874':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 'NaN':::DECIMAL, '':::STRING, '\xfd04e4693a09a0ac':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (30510:::INT8, (-1958098649):::INT8, NULL, (-0.25865939259529114):::FLOAT8, (-0.663797986696129):::FLOAT8, '5874897-12-31':::DATE, '1996-11-26 15:10:18.000837':::TIMESTAMP, '1975-09-04 00:17:50.000051+00':::TIMESTAMPTZ, '60 years 1 mon 627 days 09:10:11.953929':::INTERVAL, true, (-798.7767823392497097):::DECIMAL, '':::STRING, '\x27':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '31.215.90.33/2':::INET, '{"5xkLApf": {"c": true, "foobar": 1.246060696290563}, "EAe": 0.03618410061566466, "foobar": {"c": "foobar", "o6Vx": "bar"}}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1149: Skipping predicate false because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._float4 AS col̓😱_43, seed._bytes AS "c*ol,_44", uuid(seed._uuid::UUID)::UUID AS col̮_45 FROM d.public.seed AS tab_21 ORDER BY seed.crdb_internal_mvcc_timestamp ASC NULLS LAST)
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._float4 AS col̓😱_43, seed._bytes AS "c*ol,_44", uuid(seed._uuid::UUID)::UUID AS col̮_45 FROM d.public.seed AS tab_21 ORDER BY seed.crdb_internal_mvcc_timestamp ASC NULLS LAST) to emit 100 events
    changefeed_test.go:1156: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE e'$\x026Ak\\':::STRING LIKE 'g-':::STRING
    changefeed_test.go:1149: Skipping predicate e'$\x026Ak\\':::STRING LIKE 'g-':::STRING because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL::TIMETZ IN ('05:54:01.306456-09:30':::TIMETZ, '23:38:39.736337+14:15':::TIMETZ, '03:41:51.704284-06:46':::TIMETZ, (NULL::TIMETZ + '-55 years -2 mons -413 days -05:21:59.435195':::INTERVAL::INTERVAL)::TIMETZ)
    changefeed_test.go:1149: Skipping predicate NULL::TIMETZ IN ('05:54:01.306456-09:30':::TIMETZ, '23:38:39.736337+14:15':::TIMETZ, '03:41:51.704284-06:46':::TIMETZ, (NULL::TIMETZ + '-55 years -2 mons -413 days -05:21:59.435195':::INTERVAL::INTERVAL)::TIMETZ) because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT EXISTS (SELECT (-169186968575583987):::INT8 AS col_100 FROM d.public.seed AS tab_34 ORDER BY seed._jsonb NULLS LAST)) AND (seed._bool AND seed._bool))
    changefeed_test.go:1149: Skipping predicate ((NOT EXISTS (SELECT (-169186968575583987):::INT8 AS col_100 FROM d.public.seed AS tab_34 ORDER BY seed._jsonb NULLS LAST)) AND (seed._bool AND seed._bool)) because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((lower(e'.\x1abY':::STRING::STRING)::STRING ~* NULL) AND NULL)
    changefeed_test.go:1149: Skipping predicate ((lower(e'.\x1abY':::STRING::STRING)::STRING ~* NULL) AND NULL) because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1149: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_isvalidtrajectory('010600004008000000010300004001000000080000004076B9C3797DEBC1B092C7BCCDC4C541EEB09D9D26B1FAC15C23F49CD7FAE5C130995EDEA6FFC141E095ACC71703BEC1CA9279DD938602C263704DB06B1CF0C170586495F59DE6C1F48DFED3D35FED4109A1F76773E900C27009BB30C0F2F7411E155C14AFA6014200E4D98E2D1FE841F851DC1EF1E3F141700C14DEECEAEF417468337CA30AFC415AA44D3EE1ABF4410C084CDEAB67D6C108707A67C47F024234177F1B874FFC414076B9C3797DEBC1B092C7BCCDC4C541EEB09D9D26B1FAC10103000040010000000500000060D2E65F2153E141E0AA36E30D2602C26230EE700F6301C2580D8718D44401429E45F12C7EB6FEC19E7CEE8F6BD2F04110C3BB44AB11F7412EFECCEC29AEF141D01B4AFC2B19CAC1F314AC7EF4E9F6C178AAAD065025FF412067B8639F38BE4160D2E65F2153E141E0AA36E30D2602C26230EE700F6301C2010300004001000000060000006E7DA6BE501EEAC131074554276DF9C185BE3ABEEE73F9C16C11B88F4449E9418F614F34EC9602C24A400FFD2586E3C17656EF2E927AFA41DC4BD65047B8F8C1706F27DCF13EDEC118994F0D2C5DFF41482753BEBAC8D8C1F29DECBDB8AFF941C05CDB7309C1D941805CEF2D2DFDA8C180011CF6E5709C416E7DA6BE501EEAC131074554276DF9C185BE3ABEEE73F9C101030000400100000008000000552EA0C2B9B8F0C1D2D7D0E4873700C26894E618F226EA4128E15CD842F5F8417A446650EB2E02C2CB84463B722CFFC192B3B83FFC1E014210264A9DB0DDEDC12090137B820DD1C12088D2D0659A0142FCB0C5237559DBC100663B2CB4C59541A6AFE4D5324DF5415228D88F07BDF041C8ACB94CE528E2C1B02B93BE30DBC7415C4EB39DD010FF4148B31E8FCF88F74150F88C31EA33DF416C58C1A8195AE3418E652588C870F841552EA0C2B9B8F0C1D2D7D0E4873700C26894E618F226EA41010300004001000000090000004CFB6E5D6832F3C1C43665D9C506F8C1D8622D543E4DEAC1DC1249C7B95AF341D7D2440FDA8B02C22EE668EE2A6D0142CCBAA05D82E5FB416451812BB8F0FDC1E6D89B50996FE2C100865A4EC22BFE410429EAEEAE43F6C110F86E924F58F841BFE63567B2E0F4C18411D644B2140042F4FB018AB25CFE41E0A955E0D5B8DBC18094A45A4E44B5C1E05CDBBC67E9F84116C1CC15311802C2FC55EBBDBC96FD41ACF5371976C8E2411B76C175A37BF0C1C8FC1A2487E0D941B4A146A7D02EF3414CFB6E5D6832F3C1C43665D9C506F8C1D8622D543E4DEAC101030000400100000009000000288F8F227244E54144AE98239EE1E0C1EEA49FDD278FE5C14CD2E54DFC8FF5414FF3C6DF4815F9C1A2BFC6629E03F741A457E8C74F7FF941A06D82A00BE8BBC1C0313B2ABA54E4C18242C1727D46F441C0E936197DECA841F02DA91E0EC1C541F8A395AF10E7E74150323FAA1679FB4138D00350ED1EE4C162DED77E18A5FBC18AB89B199126F341541A111D430BFAC1CA9B67AE92D8F7C110D458E54B21F1416B894E256DE901C28013C20DDC1EB241E0D75F1BA1C3D041B4904C3DCC4ED6C1288F8F227244E54144AE98239EE1E0C1EEA49FDD278FE5C101030000400100000008000000B8E0D3FD711DFBC138B6BB3B400DF2C160DA3C519B2501C252A32794B78201C2BF481B51B87102C288970E1A643301C2434EF7A3A1BDF1C1D230EFC09935F0C100366A5536F3D641689269D858F8D941D066231D42FAC941D6185D4F287DF2C1949AFC7EDCF5F14170332DC6BA6BF3412018BF454386C8C130C208CEF7E1C14190737032D15AFC418006ABA825840242807AC527FC88E3C19CA7CD847C570042D20BA846B4A5F0C1B8E0D3FD711DFBC138B6BB3B400DF2C160DA3C519B2501C2010300004001000000080000002EEDE95FC52DF8C164E4042566FED8C106A772176FE4ECC1701163B1E176D8414ECB50F8A5FCF6C10E6861FA6CE0F3412616DA298564F2414C3FFB38A451D5C17C9FD46B1978EF4188E660A33F91E841D8962C2235560042F2077460B056F94190F72F6A8849DC4158BFB4D60508FD41D81E324F39F701C2A0013DD03879BCC1CE3ECB12B0D1F2417438FDBABA99FD4174DDC10379F0FFC1B0D4C499DD1202428F16656DD6E3F3C12EEDE95FC52DF8C164E4042566FED8C106A772176FE4ECC1':::GEOMETRY::GEOMETRY)::BOOL
    changefeed_test.go:1142: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE st_isvalidtrajectory('010600004008000000010300004001000000080000004076B9C3797DEBC1B092C7BCCDC4C541EEB09D9D26B1FAC15C23F49CD7FAE5C130995EDEA6FFC141E095ACC71703BEC1CA9279DD938602C263704DB06B1CF0C170586495F59DE6C1F48DFED3D35FED4109A1F76773E900C27009BB30C0F2F7411E155C14AFA6014200E4D98E2D1FE841F851DC1EF1E3F141700C14DEECEAEF417468337CA30AFC415AA44D3EE1ABF4410C084CDEAB67D6C108707A67C47F024234177F1B874FFC414076B9C3797DEBC1B092C7BCCDC4C541EEB09D9D26B1FAC10103000040010000000500000060D2E65F2153E141E0AA36E30D2602C26230EE700F6301C2580D8718D44401429E45F12C7EB6FEC19E7CEE8F6BD2F04110C3BB44AB11F7412EFECCEC29AEF141D01B4AFC2B19CAC1F314AC7EF4E9F6C178AAAD065025FF412067B8639F38BE4160D2E65F2153E141E0AA36E30D2602C26230EE700F6301C2010300004001000000060000006E7DA6BE501EEAC131074554276DF9C185BE3ABEEE73F9C16C11B88F4449E9418F614F34EC9602C24A400FFD2586E3C17656EF2E927AFA41DC4BD65047B8F8C1706F27DCF13EDEC118994F0D2C5DFF41482753BEBAC8D8C1F29DECBDB8AFF941C05CDB7309C1D941805CEF2D2DFDA8C180011CF6E5709C416E7DA6BE501EEAC131074554276DF9C185BE3ABEEE73F9C101030000400100000008000000552EA0C2B9B8F0C1D2D7D0E4873700C26894E618F226EA4128E15CD842F5F8417A446650EB2E02C2CB84463B722CFFC192B3B83FFC1E014210264A9DB0DDEDC12090137B820DD1C12088D2D0659A0142FCB0C5237559DBC100663B2CB4C59541A6AFE4D5324DF5415228D88F07BDF041C8ACB94CE528E2C1B02B93BE30DBC7415C4EB39DD010FF4148B31E8FCF88F74150F88C31EA33DF416C58C1A8195AE3418E652588C870F841552EA0C2B9B8F0C1D2D7D0E4873700C26894E618F226EA41010300004001000000090000004CFB6E5D6832F3C1C43665D9C506F8C1D8622D543E4DEAC1DC1249C7B95AF341D7D2440FDA8B02C22EE668EE2A6D0142CCBAA05D82E5FB416451812BB8F0FDC1E6D89B50996FE2C100865A4EC22BFE410429EAEEAE43F6C110F86E924F58F841BFE63567B2E0F4C18411D644B2140042F4FB018AB25CFE41E0A955E0D5B8DBC18094A45A4E44B5C1E05CDBBC67E9F84116C1CC15311802C2FC55EBBDBC96FD41ACF5371976C8E2411B76C175A37BF0C1C8FC1A2487E0D941B4A146A7D02EF3414CFB6E5D6832F3C1C43665D9C506F8C1D8622D543E4DEAC101030000400100000009000000288F8F227244E54144AE98239EE1E0C1EEA49FDD278FE5C14CD2E54DFC8FF5414FF3C6DF4815F9C1A2BFC6629E03F741A457E8C74F7FF941A06D82A00BE8BBC1C0313B2ABA54E4C18242C1727D46F441C0E936197DECA841F02DA91E0EC1C541F8A395AF10E7E74150323FAA1679FB4138D00350ED1EE4C162DED77E18A5FBC18AB89B199126F341541A111D430BFAC1CA9B67AE92D8F7C110D458E54B21F1416B894E256DE901C28013C20DDC1EB241E0D75F1BA1C3D041B4904C3DCC4ED6C1288F8F227244E54144AE98239EE1E0C1EEA49FDD278FE5C101030000400100000008000000B8E0D3FD711DFBC138B6BB3B400DF2C160DA3C519B2501C252A32794B78201C2BF481B51B87102C288970E1A643301C2434EF7A3A1BDF1C1D230EFC09935F0C100366A5536F3D641689269D858F8D941D066231D42FAC941D6185D4F287DF2C1949AFC7EDCF5F14170332DC6BA6BF3412018BF454386C8C130C208CEF7E1C14190737032D15AFC418006ABA825840242807AC527FC88E3C19CA7CD847C570042D20BA846B4A5F0C1B8E0D3FD711DFBC138B6BB3B400DF2C160DA3C519B2501C2010300004001000000080000002EEDE95FC52DF8C164E4042566FED8C106A772176FE4ECC1701163B1E176D8414ECB50F8A5FCF6C10E6861FA6CE0F3412616DA298564F2414C3FFB38A451D5C17C9FD46B1978EF4188E660A33F91E841D8962C2235560042F2077460B056F94190F72F6A8849DC4158BFB4D60508FD41D81E324F39F701C2A0013DD03879BCC1CE3ECB12B0D1F2417438FDBABA99FD4174DDC10379F0FFC1B0D4C499DD1202428F16656DD6E3F3C12EEDE95FC52DF8C164E4042566FED8C106A772176FE4ECC1':::GEOMETRY::GEOMETRY)::BOOL because error pq: st_isvalidtrajectory(): expected LineString, got MultiPolygonM
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string ~ seed._string
    changefeed_test.go:1142: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string ~ seed._string because error pq: invalid regular expression: error parsing regexp: unexpected ): ``1J�)``
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (e'a\x12+7\x1czg(':::STRING::STRING % seed._string::STRING)::BOOL
    changefeed_test.go:1149: Skipping predicate (e'a\x12+7\x1czg(':::STRING::STRING % seed._string::STRING)::BOOL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('20:03:21.675664+04:34':::TIMETZ::TIMETZ || ('C21D9510/C1E5EA27':::PG_LSN::PG_LSN || '':::STRING::STRING)::STRING::STRING)::STRING SIMILAR TO seed._string
    changefeed_test.go:1142: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('20:03:21.675664+04:34':::TIMETZ::TIMETZ || ('C21D9510/C1E5EA27':::PG_LSN::PG_LSN || '':::STRING::STRING)::STRING::STRING)::STRING SIMILAR TO seed._string because error pq: invalid regular expression: error parsing regexp: unexpected ): ``^(?s:1J�))$``
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._bool OR ilike_escape(seed._string::STRING, seed._string::STRING, e'H+\'D':::STRING::STRING)::BOOL))
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._bool OR ilike_escape(seed._string::STRING, seed._string::STRING, e'H+\'D':::STRING::STRING)::BOOL)) to emit 49 events
    changefeed_test.go:1170: ilike_escape(): invalid escape string
    --- FAIL: TestChangefeedRandomExpressions/kafka (4.89s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2.0-rc release-blocker]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.3-rc release-blocker]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17122978?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17122978?buildTab=artifacts#/) on master @ [75e14fa0c5ccfa36f7d45496582bff23af2b3b32](https://github.com/cockroachdb/cockroach/commits/75e14fa0c5ccfa36f7d45496582bff23af2b3b32):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-5273):::INT8, (-821712059):::INT8, 4664354559486696975:::INT8, (-1.0):::FLOAT8, (-1.0032975192257796):::FLOAT8, '1970-11-18':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2005-10-26 14:18:07.000405+00':::TIMESTAMPTZ, '68 years 153 days 04:40:34.848565':::INTERVAL, false, NULL, e'\x07Xq\'Lhh':::STRING, '\x38c0fd':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '144.252.231.148/23':::INET, '[{"a": {"N4xAcZ": true, "baz": {"foo": null}}, "bTmIK1": 0.46459119908553986}, {"a": true, "bar": ["a"]}, ["foobar", 0.02094387615993551], false, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, NULL, (-7059302810377547298):::INT8, 1.401298464324817e-45:::FLOAT8, (-1.9497611235575718):::FLOAT8, '1977-10-09':::DATE, '2001-12-18 13:57:33.000591':::TIMESTAMP, NULL, '00:00:00':::INTERVAL, true, 0:::DECIMAL, e'W\fK':::STRING, '\x27':::BYTES, '4f022d8b-c773-4a53-9eba-f4c160596fe5':::UUID, 'd8ed:d879:73b:3ff1:d1ae:fd50:22b8:f49f/85':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-22307):::INT8, 957943817:::INT8, 9223372036854775807:::INT8, NULL, (-1.0):::FLOAT8, NULL, '2022-11-16 22:11:54.000984':::TIMESTAMP, '3000-01-01 00:00:00+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, 1489.941358437334819:::DECIMAL, e'\x00':::STRING, '\x22':::BYTES, 'a96e4cd6-999b-4b03-ab9c-a0f4653101a7':::UUID, NULL, '{"b": {"LRjUYHQL": null, "a": {"BFJOYka": true, "RTuq8LN": ["a"], "baz": true}, "foo": 0.3088351602853645, "foobar": 0.33263532036437127}, "baz": "baz"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-15531):::INT8, (-128):::INT8, 32767:::INT8, 0.27595028281211853:::FLOAT8, 0.0:::FLOAT8, '1970-01-01':::DATE, '2009-12-19 14:49:24.000427':::TIMESTAMP, NULL, '290 years':::INTERVAL, false, 766113408060.5118252:::DECIMAL, '':::STRING, '\x22':::BYTES, 'be4300fc-b9ec-4e0d-a2ce-c94b60fd2088':::UUID, 'a9f4:95af:4299:7c83:4a07:163d:8029:f50d/124':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 73575618:::INT8, 8151084182132522361:::INT8, (-0.0):::FLOAT8, '-Inf':::FLOAT8, '1991-06-19':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '1 day':::INTERVAL, true, 60588594222368110.46:::DECIMAL, e'?1N]\x10\x0fE0,':::STRING, NULL, 'f0bdf907-15ca-4194-91c0-ec705f12d62c':::UUID, '339f:7347:7361:310a:57ed:b84a:7d42:826c/68':::INET, '{"1xgwwHG": true, "S3Bc2D": {"xcPLkbSUo": "w8ECyulnZ"}, "c": {"9PGvL9HL": 4.0825424022806445}, "foobar": false}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, 0:::INT8, (-9223372036854775807):::INT8, (-0.0):::FLOAT8, (-2.162961911120915):::FLOAT8, '1970-01-01':::DATE, '1980-01-04 21:58:20.000646':::TIMESTAMP, '2003-06-14 12:43:23.000115+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 5.396414666696433114E+25:::DECIMAL, e'05E-\x1bmN\x1c``':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-9639):::INT8, (-592380349):::INT8, 32767:::INT8, 0.0:::FLOAT8, (-1.4095383969678612):::FLOAT8, '1989-05-22':::DATE, '1999-03-24 12:30:11.000885':::TIMESTAMP, '1991-02-21 05:42:01.00027+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, (-393459.3639570999081):::DECIMAL, '':::STRING, '\x32da734b43c8e3':::BYTES, '4d983769-5b67-48d9-8049-2b9191ed27eb':::UUID, '15fc:dc6:8ce0:242b:d794:e539:9364:83cd/71':::INET, '{"b": null, "baz": null, "c": 0.7391866212589066}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-1747506350):::INT8, (-3177220541444176873):::INT8, (-0.719997227191925):::FLOAT8, '+Inf':::FLOAT8, NULL, '294276-12-31 23:59:59.999999':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '79 years 10 mons 634 days 04:26:33.385891':::INTERVAL, true, 'Infinity':::DECIMAL, e'\'':::STRING, '\xa2df6061':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, NULL, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32737):::INT8, 565758622:::INT8, (-8362782108007453152):::INT8, (-0.8515405654907227):::FLOAT8, 0.35448739207418994:::FLOAT8, '5874897-12-31':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, NULL, true, (-2.689996831069048265E+30):::DECIMAL, e'\U00002603':::STRING, '\x58':::BYTES, '6ef3a7e5-f754-476f-87d6-d106ab7d725c':::UUID, NULL, '{"2Jmu0Ly": 2.0180033703487283, "PM": true, "bar": false}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1149: Skipping predicate false because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 47 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1149: Skipping predicate (seed._inet::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1149: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (CASE WHEN seed._bool THEN e'\x7f&/o\x1aJ00':::STRING ELSE seed._string END::STRING || seed._string::STRING)::STRING ~ seed._string
    changefeed_test.go:1142: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (CASE WHEN seed._bool THEN e'\x7f&/o\x1aJ00':::STRING ELSE seed._string END::STRING || seed._string::STRING)::STRING ~ seed._string because error pq: invalid regular expression: error parsing regexp: missing closing ): ``-�U
        Q(�``
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT seed._bool) OR true)
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((NOT seed._bool) OR true) to emit 100 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (0.1801498400085238:::FLOAT8 IS NOT DISTINCT FROM seed._float4)
    changefeed_test.go:1149: Skipping predicate (0.1801498400085238:::FLOAT8 IS NOT DISTINCT FROM seed._float4) because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT ((NULL::INET | 'daca:2396:bb30:1ab1:8528:7742:abbc:6a4f/40':::INET::INET)::INET::INET IN (('38.32.34.110':::INET::INET + 3197681253781405589:::INT8::INT8)::INET, NULL)))
    changefeed_test.go:1149: Skipping predicate (NOT ((NULL::INET | 'daca:2396:bb30:1ab1:8528:7742:abbc:6a4f/40':::INET::INET)::INET::INET IN (('38.32.34.110':::INET::INET + 3197681253781405589:::INT8::INT8)::INET, NULL))) because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (inet(seed._inet::INET)::INET::INET >> NULL::INET)::BOOL
    changefeed_test.go:1149: Skipping predicate (inet(seed._inet::INET)::INET::INET >> NULL::INET)::BOOL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('':::STRING::STRING % seed._string::STRING)::BOOL
    changefeed_test.go:1149: Skipping predicate ('':::STRING::STRING % seed._string::STRING)::BOOL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true AND seed._bool)
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true AND seed._bool) to emit 47 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ilike_escape(btrim((seed._date::DATE || e'u\x05\x7f':::STRING::STRING)::STRING::STRING, 'm7lZ=':::STRING::STRING)::STRING::STRING, NULL::STRING, NULL::STRING)::BOOL
    changefeed_test.go:1149: Skipping predicate ilike_escape(btrim((seed._date::DATE || e'u\x05\x7f':::STRING::STRING)::STRING::STRING, 'm7lZ=':::STRING::STRING)::STRING::STRING, NULL::STRING, NULL::STRING)::BOOL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL LIKE NULL
    changefeed_test.go:1149: Skipping predicate NULL LIKE NULL because it returned no rows
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT seed._bool) OR seed._bool)
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((NOT seed._bool) OR seed._bool) to emit 90 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((seed._bool AND seed._bool) OR true)
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((seed._bool AND seed._bool) OR true) to emit 100 events
    changefeed_test.go:1137: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND ('':::STRING SIMILAR TO seed._string))
    changefeed_test.go:1153: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND ('':::STRING SIMILAR TO seed._string)) to emit 8 events
    changefeed_test.go:1170: invalid regular expression: error parsing regexp: missing closing ): ``^(?s:qS��
        (?:�@(?:)$``
    --- FAIL: TestChangefeedRandomExpressions/kafka (3.68s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #131179 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1.5-rc release-blocker]
- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2.0-rc]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.3-rc]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/67ed3123-92a6-4610-b820-f465cab63118?testReportRun=6&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [84503526b5123563f0d3c6c6175bdaac9013497a](https://github.com/cockroachdb/cockroach/commits/84503526b5123563f0d3c6c6175bdaac9013497a):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (3060:::INT8, (-2147483648):::INT8, (-2147483648):::INT8, '-Inf':::FLOAT8, 1.0651874133036188:::FLOAT8, '1988-04-02':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '6 years 3 mons 812 days 10:22:17.438264':::INTERVAL, false, (-6.935497191400616045E+37):::DECIMAL, 'X':::STRING, '\x5e0f62830314bb':::BYTES, NULL, '5817:674a:aaf8:d44c:5881:e504:7d8c:c7ec/39':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-3168):::INT8, 310318860:::INT8, 0:::INT8, '+Inf':::FLOAT8, '-Inf':::FLOAT8, '1972-05-02':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '66 years 11 mons 730 days 23:29:05.236388':::INTERVAL, false, (-53605394.01642487552):::DECIMAL, e'\U00002603':::STRING, NULL, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (32767:::INT8, 1770051604:::INT8, (-2426862428896943573):::INT8, 2.0536205768585205:::FLOAT8, (-0.24430265562723985):::FLOAT8, 'infinity':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '290 years':::INTERVAL, false, 358583199.9054225709:::DECIMAL, e'\x00':::STRING, '\x00':::BYTES, 'c32d8079-c62c-49a7-92c0-b3f0dcaf3330':::UUID, NULL, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (3893:::INT8, 154079032:::INT8, 4468869469863663796:::INT8, 0.21678172051906586:::FLOAT8, (-0.26109459395826606):::FLOAT8, '1972-08-06':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '1982-09-01 22:44:00.000142+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-9.09575336507075824E+28):::DECIMAL, NULL, '\xe29883':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'bb10:d7ed:c7c6:9b8d:dd9f:9c7d:9911:774d/5':::INET, '["nkn1J6yEH", "c"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-3266):::INT8, (-2147483648):::INT8, (-1):::INT8, 1.0:::FLOAT8, (-0.0):::FLOAT8, '1970-01-01':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '2003-10-14 06:50:17.000834+00':::TIMESTAMPTZ, '-18 years -5 mons -896 days -01:38:40.453117':::INTERVAL, false, (-24301305249055375.92):::DECIMAL, e'\'':::STRING, '\x22':::BYTES, '9878ae7e-707b-4d82-8857-130503bef12e':::UUID, '228.52.184.117/2':::INET, 'true':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (31156:::INT8, 1:::INT8, 7787542527423571275:::INT8, 0.004635083023458719:::FLOAT8, 5e-324:::FLOAT8, '1971-06-17':::DATE, '2000-05-11 21:18:52.000581':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-16 years -6 mons -381 days -01:04:46.018192':::INTERVAL, true, 1.0:::DECIMAL, e'uf9f3\x16G\x7fk':::STRING, '\x27':::BYTES, '2ae8f3a2-fe28-4343-bb10-af7c5f4f1935':::UUID, '57.101.181.213/22':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, (-519974339):::INT8, 1:::INT8, 'NaN':::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '1975-05-12':::DATE, '0001-01-01 00:00:00':::TIMESTAMP, '1998-02-17 20:20:46.000415+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-52007451634116028.90):::DECIMAL, e'\'':::STRING, '\x47e7ba4f5986':::BYTES, '3b6d6512-61b0-4548-9e4e-0a2211c0d4e4':::UUID, '253.145.56.3/5':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, (-1366734726):::INT8, 127:::INT8, 3.4028234663852886e+38:::FLOAT8, NULL, '5874897-12-31':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, NULL, NULL, NULL, 62533028485010.08109:::DECIMAL, e'\U00002603':::STRING, '\xe8ba658fbd16':::BYTES, '2645be88-78d0-48a8-bd64-4f1bfc1f5669':::UUID, '106.113.9.201/24':::INET, '{"bar": [true], "foo": false}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, 1337160799:::INT8, (-2147483648):::INT8, (-0.9856284856796265):::FLOAT8, (-0.0):::FLOAT8, '1983-08-26':::DATE, '2014-10-12 05:46:26.000746':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '-50 years -7 mons -808 days -12:42:10.440365':::INTERVAL, false, (-300.275991325577495):::DECIMAL, e'\x00':::STRING, '\xc2':::BYTES, '84080d95-9b84-40e5-b1dd-84a4ef348965':::UUID, '0.0.0.0/0':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 2147483647:::INT8, (-981831581719457382):::INT8, (-0.26496368646621704):::FLOAT8, 1.3049651449909612:::FLOAT8, '4714-11-24 BC':::DATE, '2018-03-03 15:03:25.000251':::TIMESTAMP, '2013-09-07 16:29:22.000879+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-2.018107004258504233E+24):::DECIMAL, '':::STRING, '\x22':::BYTES, 'f1556607-f349-4457-9250-a8136e22b379':::UUID, '19.141.74.121/0':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (23:::INT8, (-2147483648):::INT8, 5615228443618377888:::INT8, 'NaN':::FLOAT8, 0.38896015677819445:::FLOAT8, '1970-07-04':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '2014-04-30 14:07:31.000098+00':::TIMESTAMPTZ, '1 day':::INTERVAL, NULL, NULL, e'\x00':::STRING, '\xe29883':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '{"6Apm2cCIB": {"c": 1.8673956243649323, "sAaI": 0.7974014708485762}, "bar": true, "ii": false, "veByaupM": true}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (24714:::INT8, 842523475:::INT8, (-1):::INT8, 0.026209698989987373:::FLOAT8, '-Inf':::FLOAT8, '1977-08-23':::DATE, '2013-05-17 07:58:33.000607':::TIMESTAMP, '2013-10-06 15:51:56.000673+00':::TIMESTAMPTZ, '-20 years -5 mons -611 days -01:25:30.020704':::INTERVAL, true, (-9.146856371848335448E+26):::DECIMAL, e'\x17\x7f"':::STRING, '\x27':::BYTES, '63d5152d-7a56-4c35-8578-c4b8d822c8ad':::UUID, '38.172.184.175':::INET, '{"bar": true, "foo": [0.43645501906067574], "ofldmU": null}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (10554:::INT8, 2100214593:::INT8, 4043070262088141054:::INT8, 0.3574737012386322:::FLOAT8, 3.4028234663852886e+38:::FLOAT8, '4714-11-24 BC':::DATE, '2030-11-29 03:56:15.000321':::TIMESTAMP, '2028-12-16 10:05:47.000228+00':::TIMESTAMPTZ, '22 years 4 mons 211 days 02:18:43.937526':::INTERVAL, false, 5.715733012835385716E+27:::DECIMAL, e'\'':::STRING, '\x31':::BYTES, '27d00c2f-0234-41f6-9803-f54df146e0b2':::UUID, '109.153.64.7/23':::INET, '{"5SkhMW04": 0.7123214097746277, "FXj2O": {"H9B": {"mHLHWIm": "a", "mb0L": null}, "c": false, "m6KS": null}, "IA": true, "c": null}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, (-1):::INT8, 977570864354592534:::INT8, '+Inf':::FLOAT8, (-0.8762813989169875):::FLOAT8, 'infinity':::DATE, '2028-12-11 00:14:39.000384':::TIMESTAMP, '1989-01-30 13:16:51.000736+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, '-Infinity':::DECIMAL, '':::STRING, '\x27':::BYTES, '5cf1b085-a6f1-48f9-ba20-2a32fb757c51':::UUID, '0.0.0.0/0':::INET, '{"3JvffQX": 1.5649575287908117, "LKLqGw": "foo", "UVZINj": {"a": "XzgAcni"}, "b": 0.13602352256975672, "mEnemA9yN": [null, 0.3031874003179271, null]}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (4227:::INT8, 985152772:::INT8, 4914713655391160442:::INT8, (-1.4411450624465942):::FLOAT8, 0.07357262316202895:::FLOAT8, NULL, '4714-11-24 00:00:00 BC':::TIMESTAMP, '1970-03-14 01:20:59.000096+00':::TIMESTAMPTZ, '-21 years -6 mons -499 days -00:47:49.142033':::INTERVAL, false, 1.887989038815547388E+25:::DECIMAL, 'wS/a':::STRING, '\x58':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '[{"ywzdZZKu": true}, "a"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-30366):::INT8, (-1667611854):::INT8, (-2678501610029878718):::INT8, (-1.5104206800460815):::FLOAT8, 3.4028234663852886e+38:::FLOAT8, '1990-05-28':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, 1:::DECIMAL, 'X':::STRING, '\x14c278da41':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-453142473):::INT8, (-9040129490009151497):::INT8, 1.0615941286087036:::FLOAT8, 0.9004547717730285:::FLOAT8, '1970-01-01':::DATE, '2009-05-20 15:56:06.000726':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 1:::DECIMAL, '}':::STRING, '\xe29883':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[{"BEui4": [false], "Q1h3bC": "xyKUABx8", "jRhqt0": false}, true, "3nMdAhJ9"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 127:::INT8, 32767:::INT8, 1.1176718473434448:::FLOAT8, 1.57725854882713:::FLOAT8, '1982-10-13':::DATE, '2003-10-09 20:20:41.000563':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '-65 years -7 mons -820 days -01:06:06.377903':::INTERVAL, NULL, (-1):::DECIMAL, e'V\x1b\x11;T^':::STRING, '\xe29883':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, 2037756160:::INT8, 1864088117122986899:::INT8, 0.0:::FLOAT8, 5e-324:::FLOAT8, '4714-11-24 BC':::DATE, '2027-03-15 00:45:02.000739':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, true, (-1.234E+401):::DECIMAL, 'QE~1':::STRING, '\x37bfe957109d21ab':::BYTES, '3c07fbbb-37b2-4247-8e81-43878cf1391e':::UUID, '255.200.32.215/29':::INET, 'false':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-26547):::INT8, (-1575969977):::INT8, 2147483647:::INT8, NULL, (-1.0):::FLOAT8, '4714-11-24 BC':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '1996-07-20 09:39:38.00056+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, true, 7.551615465250817584E+30:::DECIMAL, e'\x07>,m\x1f3|{':::STRING, '\x4dbd2b6597932601':::BYTES, 'ef3dc9c5-67fa-4afa-8a26-ba700c391f02':::UUID, '0.0.0.0/0':::INET, '[null, true, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, (-32768):::INT8, (-32768):::INT8, 1.401298464324817e-45:::FLOAT8, 0.2318866939517145:::FLOAT8, '4714-11-24 BC':::DATE, '1983-03-14 02:41:06.000065':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '-3 years -7 mons -71 days -21:19:52.367131':::INTERVAL, false, (-7.192517834613732253E+24):::DECIMAL, '':::STRING, '\xe29883':::BYTES, '2af767ff-d830-48c7-aa27-455027ebe314':::UUID, '0.0.0.0/0':::INET, '{"Zv": ["baz"], "b": ["foobar"], "bar": null, "qTjd": 3.4625712739078827, "rXF": {"Y3YG2J": false, "c": [false], "foo": "c5"}}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT 0.42643634174532874:::FLOAT8 AS "co l_1")
    changefeed_test.go:1150: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT 0.42643634174532874:::FLOAT8 AS "co l_1") to emit 100 events
    changefeed_test.go:1153: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1150: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> seed._inet::INET)::BOOL
    changefeed_test.go:1146: Skipping predicate (seed._inet::INET >> seed._inet::INET)::BOOL because it returned no rows
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE NULL
    changefeed_test.go:1146: Skipping predicate NULL because it returned no rows
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_dwithinexclusive('01040000C00600000001010000C0500A1CB5E9E3C3C1CC19C5523729FEC13846C91B37C1FA41E08EEFAEEA75B8C101010000C0BAB4F9E39E0602426456036BA6C6FA41D71C02698048F1C1C0863277CEF7A54101010000C0C8490CB70B55D041CC32CAE4978EEAC168ECF295F635E441F0555880C734C1C101010000C04050F0CC0ED1F7416CD453B2DA44F54128D81C8DE529FD4132DF9F7819EDF5C101010000C04AAF70DD3318E0C1003A21A4362C8D414C2E799929BDEE415F348D29445701C201010000C08EE765C8DB76F441A6A32D43B2A4F7C1B20C4EAE8626E3C1EA0798AD0437F7C1':::GEOMETRY::GEOMETRY, st_clipbybox2d(st_geomfromgeohash(sha1(seed._bytes::BYTES)::STRING::STRING, 2147483647:::INT8::INT8)::GEOMETRY::GEOMETRY, 'BOX(-2.2536617950143243 -0.17142462610866308,-0.809552928934357 -0.10573137236243893)':::BOX2D::BOX2D)::GEOMETRY::GEOMETRY, 0.9643877271228383:::FLOAT8::FLOAT8)::BOOL
    changefeed_test.go:1139: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE _st_dwithinexclusive('01040000C00600000001010000C0500A1CB5E9E3C3C1CC19C5523729FEC13846C91B37C1FA41E08EEFAEEA75B8C101010000C0BAB4F9E39E0602426456036BA6C6FA41D71C02698048F1C1C0863277CEF7A54101010000C0C8490CB70B55D041CC32CAE4978EEAC168ECF295F635E441F0555880C734C1C101010000C04050F0CC0ED1F7416CD453B2DA44F54128D81C8DE529FD4132DF9F7819EDF5C101010000C04AAF70DD3318E0C1003A21A4362C8D414C2E799929BDEE415F348D29445701C201010000C08EE765C8DB76F441A6A32D43B2A4F7C1B20C4EAE8626E3C1EA0798AD0437F7C1':::GEOMETRY::GEOMETRY, st_clipbybox2d(st_geomfromgeohash(sha1(seed._bytes::BYTES)::STRING::STRING, 2147483647:::INT8::INT8)::GEOMETRY::GEOMETRY, 'BOX(-2.2536617950143243 -0.17142462610866308,-0.809552928934357 -0.10573137236243893)':::BOX2D::BOX2D)::GEOMETRY::GEOMETRY, 0.9643877271228383:::FLOAT8::FLOAT8)::BOOL because error pq: st_geomfromgeohash(): invalid GeoHash: geohash decode '356a192b7913b04c54574d18c28d46e6395428ab': invalid character at index 3
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1150: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 48 events
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((false AND true) AND false)
    changefeed_test.go:1146: Skipping predicate ((false AND true) AND false) because it returned no rows
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (seed._timestamptz::TIMESTAMPTZ NOT IN (SELECT NULL::TIMESTAMPTZ AS col_67 FROM d.public.seed AS "t%6bab_28" GROUP BY seed.crdb_internal_origin_timestamp, seed._float8, seed._bytes ORDER BY seed._bytes DESC NULLS LAST)))
    changefeed_test.go:1146: Skipping predicate (NOT (seed._timestamptz::TIMESTAMPTZ NOT IN (SELECT NULL::TIMESTAMPTZ AS col_67 FROM d.public.seed AS "t%6bab_28" GROUP BY seed.crdb_internal_origin_timestamp, seed._float8, seed._bytes ORDER BY seed._bytes DESC NULLS LAST))) because it returned no rows
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1146: Skipping predicate false because it returned no rows
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT seed._string AS col_88 FROM d.public.seed AS tab_38)
    changefeed_test.go:1150: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT seed._string AS col_88 FROM d.public.seed AS tab_38) to emit 100 events
    changefeed_test.go:1153: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1134: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (seed._bool AND (seed.crdb_internal_origin_id::INT8 IN (COALESCE(seed._int2, seed._int8), (seed._date::DATE - seed._date::DATE)::INT8, seed.crdb_internal_origin_id))))
    changefeed_test.go:1150: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (NOT (seed._bool AND (seed.crdb_internal_origin_id::INT8 IN (COALESCE(seed._int2, seed._int8), (seed._date::DATE - seed._date::DATE)::INT8, seed.crdb_internal_origin_id)))) to emit 46 events
    changefeed_test.go:1167: cannot subtract infinite dates
    --- FAIL: TestChangefeedRandomExpressions/kafka (2.40s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #133049 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.3]
- #132516 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2.13-rc]
- #131898 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-24.1.6-rc release-blocker]
- #131179 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.5-rc]
- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2.0-rc]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.3-rc]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/e40aa913-9600-4a4a-97bf-1b8c9668ca66?testReportRun=12&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [8f5366d09e6cf2144ca43f9cdda7e1128a13fbf8](https://github.com/cockroachdb/cockroach/commits/8f5366d09e6cf2144ca43f9cdda7e1128a13fbf8):


```
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-32768):::INT8, (-927166900):::INT8, (-954484165732507652):::INT8, (-1.0):::FLOAT8, (-0.814715902765541):::FLOAT8, '1970-01-01':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '2030-12-08 21:31:09.00047+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, (-3.745304500834548042E+19):::DECIMAL, 'X':::STRING, '\xe29883':::BYTES, '0f866cac-20f2-4ca0-8f92-4987fb5f2892':::UUID, '91c9:65cb:6f1f:50d0:ba70:efae:3842:bfd1/82':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 2147483647:::INT8, (-2147483648):::INT8, 0.17483708262443542:::FLOAT8, (-1.0):::FLOAT8, '1988-05-19':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '290 years':::INTERVAL, true, 'NaN':::DECIMAL, '':::STRING, '\xff':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-32768):::INT8, (-3184942503412050409):::INT8, (-1.6095027923583984):::FLOAT8, 0.053672508800749585:::FLOAT8, '1970-08-02':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '2001-11-02 14:23:00.000644+00':::TIMESTAMPTZ, '-54 years -7 mons -723 days -03:36:21.725837':::INTERVAL, true, 3.842703367634909968E+33:::DECIMAL, NULL, '\xd2b57538':::BYTES, 'f7343245-6e99-4652-a39a-8ac7b2925e81':::UUID, '142.95.193.142/21':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-1):::INT8, (-32768):::INT8, (-2147483648):::INT8, (-0.11507805436849594):::FLOAT8, NULL, '1987-02-06':::DATE, '1974-07-17 08:41:31.000764':::TIMESTAMP, '1992-11-29 12:01:18.000607+00':::TIMESTAMPTZ, '6 years 10 mons 441 days 02:01:15.292229':::INTERVAL, false, (-1.201528338958214929E+34):::DECIMAL, '':::STRING, '\xff':::BYTES, 'd112bf91-196a-422f-bfdc-c9f2ad7d5bce':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[{"1IpyJ": true, "baz": false}, ["ZhAds", 0.2560597575307317], true, null]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-1198621864):::INT8, 2481295000278008879:::INT8, 1.401298464324817e-45:::FLOAT8, 5e-324:::FLOAT8, '-infinity':::DATE, '1970-12-23 18:47:17.000528':::TIMESTAMP, '2007-08-21 21:09:43.00059+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, (-1):::DECIMAL, e'\'':::STRING, '\x58':::BYTES, '904f1ed9-975a-47c2-9aa9-0e08703c60b2':::UUID, 'ea36:41e6:8ff5:40c6:3df7:1a32:819c:37e0/25':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (NULL, (-547873443):::INT8, (-7326762219265525562):::INT8, (-1.0):::FLOAT8, NULL, '1990-01-14':::DATE, '1974-09-07 09:47:34.000421':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-10 years -7 mons -744 days -08:59:22.376209':::INTERVAL, false, 2.160410314721984372E+30:::DECIMAL, '"':::STRING, '\x':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (866:::INT8, 127:::INT8, 4186269902295751585:::INT8, 1.401298464324817e-45:::FLOAT8, 1.401298464324817e-45:::FLOAT8, '1989-10-14':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '1970-03-20 04:56:34.000666+00':::TIMESTAMPTZ, '54 years 7 mons 227 days 09:42:37.062999':::INTERVAL, true, 1.0:::DECIMAL, NULL, '\xf186ddab807c':::BYTES, '9cf5916e-0974-445c-b513-3739bb2b965e':::UUID, '8db8:51f0:b879:23dd:3022:aeb3:cae7:89a3/85':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (2208:::INT8, (-1482088616):::INT8, (-4111989021073260388):::INT8, (-0.0):::FLOAT8, 0.4384708483453478:::FLOAT8, '1981-12-04':::DATE, '2019-08-18 18:20:52.000229':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 632914581265155.9252:::DECIMAL, 'X':::STRING, '\xff':::BYTES, '8c31a822-6fc0-4b52-b94a-46ea90450e2f':::UUID, '0.0.0.0/0':::INET, '[["baz", 0.43523893891569165], {"c": null}, false]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-18192):::INT8, (-1):::INT8, 1987320897176304410:::INT8, '-Inf':::FLOAT8, (-1.0):::FLOAT8, '1979-09-18':::DATE, '1994-10-05 15:10:24.000022':::TIMESTAMP, '2029-02-08 12:55:54.000017+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 85430.7673456512205:::DECIMAL, e'\x04\x01\x1fa':::STRING, '\x9839':::BYTES, '19bf35a9-92c8-4168-bb24-dc3451ff25e9':::UUID, '0.0.0.0/0':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (0:::INT8, NULL, 5843778169717409776:::INT8, NULL, 0.7204546330917863:::FLOAT8, '1970-01-01':::DATE, '2026-12-12 16:44:54.000604':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 2.978254538621351422E+33:::DECIMAL, '':::STRING, '\x27':::BYTES, 'f17cef45-f09f-4923-b767-ba8d18169c37':::UUID, NULL, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (4084:::INT8, 0:::INT8, 32767:::INT8, 1.5880622863769531:::FLOAT8, '-Inf':::FLOAT8, '1987-11-13':::DATE, '1983-03-04 17:53:51.00044':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-61 years -7 mons -263 days -23:05:59.695678':::INTERVAL, false, 0:::DECIMAL, '*':::STRING, '\xd418cba5b2ef':::BYTES, '500d4257-34fb-4c79-bc67-deb47b466550':::UUID, '48.139.69.243/21':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-22062):::INT8, 1:::INT8, (-6551753995296559897):::INT8, (-0.22093652188777924):::FLOAT8, (-1.0):::FLOAT8, 'infinity':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '1982-09-15 03:01:32.000822+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, NULL, (-5.412259799299906418E+34):::DECIMAL, e'\'':::STRING, '\xe4331582d1186b77':::BYTES, '85085cb4-8633-42ba-8548-e1d43291459a':::UUID, '175.41.218.114':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (26562:::INT8, (-932462211):::INT8, (-7409262329337605153):::INT8, '-Inf':::FLOAT8, (-1.9062178423399234):::FLOAT8, '1994-10-24':::DATE, '2001-01-01 00:00:00 BC':::TIMESTAMP, '1971-01-26 17:23:33.000542+00':::TIMESTAMPTZ, NULL, true, 'NaN':::DECIMAL, 'X':::STRING, '\xe29883':::BYTES, '87dba9a0-5971-4672-91d2-b75499581ebd':::UUID, '4178:76d:378b:3cb8:adae:e3fe:8781:85c5/57':::INET, '{"c": [{"b": true, "vRVG": {"V1E2v": 1.6937344992432009, "qIqvSL": "AKgPN1J"}}, null, true, ["bar"], 0.0027441331362771937], "l7": null}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 2147483647:::INT8, (-128):::INT8, 0.0:::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '1981-12-09':::DATE, '3000-01-01 00:00:00':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '-60 years -5 mons -928 days -07:40:22.269':::INTERVAL, false, NULL, '':::STRING, '\x00':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'f1a6:6d90:5c4d:fe8a:604a:1ec5:94f:8fa0/49':::INET, '{"baz": 0.21050372028649617}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-4717):::INT8, (-1):::INT8, 3641687848235871760:::INT8, (-0.0):::FLOAT8, 1.103966750552865:::FLOAT8, '1970-01-01':::DATE, '2025-09-01 06:57:36.000014':::TIMESTAMP, '2001-01-01 00:00:00+00 BC':::TIMESTAMPTZ, '00:00:00':::INTERVAL, true, 1.0:::DECIMAL, e'\U00002603':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '96.246.232.87/2':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (23787:::INT8, (-202385826):::INT8, (-8940941507984524864):::INT8, 0.06467178463935852:::FLOAT8, 'NaN':::FLOAT8, '1985-07-09':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, (-1.234E+401):::DECIMAL, e'\x12!XhFD':::STRING, '\x8c372537fda30693':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, '0.0.0.0/0':::INET, '[{"foobar": true}, {"baz": true}, 0.04343581391701967]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-1209959710):::INT8, (-3322052491397353860):::INT8, (-0.7121473550796509):::FLOAT8, 0.12148336986256414:::FLOAT8, '1984-11-06':::DATE, '2007-01-25 10:00:28.000252':::TIMESTAMP, '2024-09-01 23:51:20.000018+00':::TIMESTAMPTZ, '-35 years -5 mons -803 days -14:14:12.797656':::INTERVAL, true, 'NaN':::DECIMAL, NULL, '\xce004862bc':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, '0.0.0.0/0':::INET, '{"test": "json"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, (-1):::INT8, 8153836462185088213:::INT8, (-0.13207252323627472):::FLOAT8, (-1.0):::FLOAT8, '1970-01-01':::DATE, '1994-01-05 23:57:59.000194':::TIMESTAMP, NULL, NULL, false, 85542048054676.08327:::DECIMAL, e'+_\x15vY*\x1e':::STRING, '\xff':::BYTES, 'ea4b9099-58dc-413d-a2c9-53d58eae9822':::UUID, '0.0.0.0/0':::INET, '[{"Ksom1u9B5": true, "c": null}, true]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (32767:::INT8, 1984506322:::INT8, (-7837936201082396735):::INT8, 1.401298464324817e-45:::FLOAT8, 0.8665453923604285:::FLOAT8, 'infinity':::DATE, '1982-06-26 15:23:02.000571':::TIMESTAMP, '2005-11-04 18:52:22.000674+00':::TIMESTAMPTZ, '1 day':::INTERVAL, false, 6.679243318625453333E+28:::DECIMAL, e'\n/U$\rlU\x05':::STRING, '\x8ff018a079':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '{"c": [null], "foobar": "mm"}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 1781514462:::INT8, (-2147483648):::INT8, 0.9515772461891174:::FLOAT8, 0.7749177682504216:::FLOAT8, NULL, '0001-01-01 00:00:00':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '-50 years -5 mons -289 days -01:41:50.424197':::INTERVAL, true, 'NaN':::DECIMAL, '"':::STRING, '\xb35a36265d012d6607':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'ab86:c50c:6930:1763:27:af3e:2e1c:43f2/121':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-11943):::INT8, 127:::INT8, 32767:::INT8, (-0.13912880420684814):::FLOAT8, NULL, '1993-04-03':::DATE, '2020-05-07 20:45:26.000014':::TIMESTAMP, '1999-10-07 03:38:22.00014+00':::TIMESTAMPTZ, '1 mon 1 day':::INTERVAL, false, (-2.353583227018389999E+34):::DECIMAL, e'\U00002603':::STRING, '\x27':::BYTES, '49346e3e-08a7-44c9-8a95-439402e0d3e2':::UUID, '20.120.246.52/21':::INET, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 0:::INT8, (-7732380629429857092):::INT8, (-0.06843160837888718):::FLOAT8, 2.064585523845304:::FLOAT8, '1983-01-10':::DATE, '294276-12-31 23:59:59.999999':::TIMESTAMP, '1971-11-30 11:03:02.000728+00':::TIMESTAMPTZ, '00:00:00':::INTERVAL, false, 8.820626343185888247E+26:::DECIMAL, e'<\x0526 ?I+':::STRING, '\x6509bfb118':::BYTES, '7be4bc20-e918-4302-8e15-e53baf86a9d4':::UUID, '39.134.237.67/29':::INET, '{}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (127:::INT8, (-1):::INT8, (-2147483648):::INT8, 0.591670036315918:::FLOAT8, 1.7976931348623157e+308:::FLOAT8, '1973-10-20':::DATE, '4714-11-24 00:00:00 BC':::TIMESTAMP, '4714-11-24 00:00:00+00 BC':::TIMESTAMPTZ, '1 day':::INTERVAL, true, 0:::DECIMAL, '"':::STRING, '\x56a2bc7a4fa389':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, NULL, '1':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1578:::INT8, 32767:::INT8, (-128):::INT8, '-Inf':::FLOAT8, NULL, '1989-04-30':::DATE, '2017-09-08 07:43:51.000192':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '17 years 11 mons 389 days 15:14:39.61811':::INTERVAL, true, 2.177917864731254498E+35:::DECIMAL, 'W7S':::STRING, '\xee34934150':::BYTES, '24bddbdf-8129-4613-8191-bce439219688':::UUID, '24.129.156.238/14':::INET, '{"6m9WpQ6HS": false, "b": {"n8iLhCp25": null}, "bar": {"SpsovKet": true, "b": false, "c": {"bar": false}}, "baz": [true, false], "foobar": 5.930432394101086}':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-20969):::INT8, 1203156552:::INT8, 1675072257174243609:::INT8, 1.2403802871704102:::FLOAT8, 'NaN':::FLOAT8, '5874897-12-31':::DATE, '2004-01-14 05:16:14.000194':::TIMESTAMP, '2003-08-28 20:05:32.00008+00':::TIMESTAMPTZ, '-26 years -237 days -07:22:49.900537':::INTERVAL, NULL, 7.197675042772433903E+23:::DECIMAL, '"':::STRING, '\x6e62e41a2ed538cd10':::BYTES, 'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, NULL);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES ((-128):::INT8, 1687718985:::INT8, 396635460572004532:::INT8, (-1.1010527610778809):::FLOAT8, 'NaN':::FLOAT8, '4714-11-24 BC':::DATE, '2024-07-04 23:39:28.0002':::TIMESTAMP, '2010-05-11 04:44:16.000588+00':::TIMESTAMPTZ, '83 years 2 mons 140 days 01:58:10.516078':::INTERVAL, NULL, (-1.234E+401):::DECIMAL, e'\x00':::STRING, '\x751f982ea5869ca3':::BYTES, 'a9a22b0e-7d2e-40b5-ba66-e7cf95e1a235':::UUID, 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET, '[{"b": 1.7260838099907247}, {"IKDmYnngc": ["kW", true], "b": true, "baz": 1.499870932122527}, false, "MXERZ", "VZpjb6C"]':::JSONB);
        INSERT INTO seed (_int2, _int4, _int8, _float4, _float8, _date, _timestamp, _timestamptz, _interval, _bool, _decimal, _string, _bytes, _uuid, _inet, _jsonb) VALUES (1:::INT8, (-1815019306):::INT8, (-6786876321013776619):::INT8, '+Inf':::FLOAT8, 1.148673477078468:::FLOAT8, '1982-08-08':::DATE, '2007-04-27 03:57:12.000498':::TIMESTAMP, '0001-01-01 00:00:00+00':::TIMESTAMPTZ, '46 years 11 mons 380 days 11:22:58.690549':::INTERVAL, false, NULL, e'\x0fY+!':::STRING, '\x':::BYTES, '00000000-0000-0000-0000-000000000000':::UUID, 'b6c6:5435:be53:cc8b:d6e1:6c3:242b:f996/29':::INET, '{"a": "b", "baz": null, "foo": null}':::JSONB);
        DELETE FROM seed WHERE rowid NOT IN (SELECT rowid FROM seed ORDER BY rowid LIMIT 100);
        ALTER TABLE seed ADD COLUMN _enum greeting;
        UPDATE seed SET _enum = enum_range('hi'::greeting)[rowid%7];
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1148: Skipping predicate false because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((true OR true) OR EXISTS (SELECT seed._string AS "\\x39col_17", (-1.6363264493993965):::FLOAT8 AS "\\u7407 col͉_18" FROM d.public.seed AS tab_5 JOIN d.public.seed AS "t ab_6" JOIN d.public.seed AS tab_7 ON ("t ab_6"._jsonb) = (tab_7._jsonb) JOIN d.public.seed AS "ta b_8" ON ((tab_7._jsonb) = ("ta b_8"._jsonb)) AND (("t ab_6"._jsonb) = ("ta b_8"._jsonb)) ON ((tab_5._float8) = ("ta b_8"._float8)) AND ((tab_5._int8) = ("ta b_8"._int4)) ORDER BY seed._float4))
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE ((true OR true) OR EXISTS (SELECT seed._string AS "\\x39col_17", (-1.6363264493993965):::FLOAT8 AS "\\u7407 col͉_18" FROM d.public.seed AS tab_5 JOIN d.public.seed AS "t ab_6" JOIN d.public.seed AS tab_7 ON ("t ab_6"._jsonb) = (tab_7._jsonb) JOIN d.public.seed AS "ta b_8" ON ((tab_7._jsonb) = ("ta b_8"._jsonb)) AND (("t ab_6"._jsonb) = ("ta b_8"._jsonb)) ON ((tab_5._float8) = ("ta b_8"._float8)) AND ((tab_5._int8) = ("ta b_8"._int4)) ORDER BY seed._float4)) to emit 100 events
    changefeed_test.go:1155: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string LIKE seed._string
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string LIKE seed._string to emit 89 events
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 44 events
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT NULL AS "Col_54", e'\x17:\f\x0f':::STRING AS "c\\uFF58ol_55" FROM d.public.seed AS tab_22 ORDER BY seed._int8 ASC)
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT NULL AS "Col_54", e'\x17:\f\x0f':::STRING AS "c\\uFF58ol_55" FROM d.public.seed AS tab_22 ORDER BY seed._int8 ASC) to emit 100 events
    changefeed_test.go:1155: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._string NOT LIKE format(e'<Fxy\x10\x0fK':::STRING::STRING, seed.crdb_internal_origin_timestamp)::STRING
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._string NOT LIKE format(e'<Fxy\x10\x0fK':::STRING::STRING, seed.crdb_internal_origin_timestamp)::STRING to emit 91 events
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((false AND (true OR ((false AND EXISTS (SELECT (-8568568745840160225):::INT8 AS col_80, e'\U0007B718\U000D3E1E\U0006B33A\U00046BF1' COLLATE da_DK AS col_81 FROM d.public.seed AS """tab_30" JOIN d.public.seed AS tab_31 ON ("""tab_30".crdb_internal_origin_timestamp) = (tab_31._decimal) ORDER BY seed._int2 ASC, seed._int8 DESC NULLS FIRST)) OR ('FAA657CE/77D0D0E8':::PG_LSN < 'D95C8C34/4546FE4B':::PG_LSN)))) OR NULL)
    changefeed_test.go:1148: Skipping predicate ((false AND (true OR ((false AND EXISTS (SELECT (-8568568745840160225):::INT8 AS col_80, e'\U0007B718\U000D3E1E\U0006B33A\U00046BF1' COLLATE da_DK AS col_81 FROM d.public.seed AS """tab_30" JOIN d.public.seed AS tab_31 ON ("""tab_30".crdb_internal_origin_timestamp) = (tab_31._decimal) ORDER BY seed._int2 ASC, seed._int8 DESC NULLS FIRST)) OR ('FAA657CE/77D0D0E8':::PG_LSN < 'D95C8C34/4546FE4B':::PG_LSN)))) OR NULL) because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed.crdb_internal_origin_id::INT8 IN ((seed._date::DATE - seed._date::DATE)::INT8,)
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed.crdb_internal_origin_id::INT8 IN ((seed._date::DATE - seed._date::DATE)::INT8,) to emit 96 events
    changefeed_test.go:1169: cannot subtract infinite dates
    --- FAIL: TestChangefeedRandomExpressions/kafka (2.63s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=12</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134813 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot T-cdc branch-release-24.1.7-rc release-blocker]
- #133049 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.3]
- #132516 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-23.2.13-rc]
- #131898 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc X-duplicate branch-release-24.1.6-rc release-blocker]
- #131179 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.5-rc]
- #129173 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2.0-rc]
- #128772 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.3-rc]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #126889 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1.2-rc]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedRandomExpressions [failed](https://tanzanite.cluster.engflow.com/invocations/default/68fb4ff7-dce9-40aa-ad5b-8ef4af94e560?testReportRun=18&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [67caf19d3998bb3ca1ada7e3c14486d505b68012](https://github.com/cockroachdb/cockroach/commits/67caf19d3998bb3ca1ada7e3c14486d505b68012):


```
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND seed._bool) to emit 43 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE seed._bool
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE seed._bool to emit 43 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._inet::INET >> '3d33:e6b4:24f9:a9cf:733c:328e:932b:3581/115':::INET::INET)::BOOL
    changefeed_test.go:1148: Skipping predicate (seed._inet::INET >> '3d33:e6b4:24f9:a9cf:733c:328e:932b:3581/115':::INET::INET)::BOOL because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE B'001010'::BIT(6) IN (CASE WHEN true THEN B'000101' ELSE B'001001' END,)
    changefeed_test.go:1148: Skipping predicate B'001010'::BIT(6) IN (CASE WHEN true THEN B'000101' ELSE B'001001' END,) because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE EXISTS (SELECT bpchar('':::VOID::VOID)::BPCHAR AS "co l_54", seed._int8 AS col_55, seed._decimal AS col_56)
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE EXISTS (SELECT bpchar('':::VOID::VOID)::BPCHAR AS "co l_54", seed._int8 AS col_55, seed._decimal AS col_56) to emit 100 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1155: Test tolerating create changefeed error: failed to start feed for job 0: pq: sub-query expressions not supported by CDC
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE true
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE true to emit 100 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE '2K[JN_':::STRING NOT LIKE e'\x110u':::STRING
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE '2K[JN_':::STRING NOT LIKE e'\x110u':::STRING to emit 100 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL = B'01101111101100011111')
    changefeed_test.go:1148: Skipping predicate (NULL = B'01101111101100011111') because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (true OR ('224.236.152.183/19':::INET::INET << (('77.203.252.90/7':::INET::INET & '64.254.23.178/5':::INET::INET)::INET::INET - seed._int8::INT8)::INET::INET)::BOOL)
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (true OR ('224.236.152.183/19':::INET::INET << (('77.203.252.90/7':::INET::INET & '64.254.23.178/5':::INET::INET)::INET::INET - seed._int8::INT8)::INET::INET)::BOOL) to emit 100 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE false
    changefeed_test.go:1148: Skipping predicate false because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (seed._string !~* (seed._string::STRING || '01040000A0E610000008000000010100008080751BB33F2C04C0DE14923BF37355403AFDB3A3FF96FCC10101000080E8344D7C5D9163407D4BDED95F7B52C0FCF89138E316EC41010100008040C84EED5C0B40C0E825963FB9023540A0473966C74CB0C10101000080460B286C8CE86040547C368529C554C0067256521CF4FDC1010100008038D4A1B13BB63B409A4DFF7A41FE5040103B9D97583ED84101010000805E0F24DD23426640E64E8CF25E284140054A526D7CC7F6C1010100008055A7843D482766C0287C4342988E33C030F485CB230FEE410101000080C4531E88F76747C060BB58A54B334B40CA28B7E59D2D0042':::GEOGRAPHY::GEOGRAPHY)::STRING))
    changefeed_test.go:1141: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT (seed._string !~* (seed._string::STRING || '01040000A0E610000008000000010100008080751BB33F2C04C0DE14923BF37355403AFDB3A3FF96FCC10101000080E8344D7C5D9163407D4BDED95F7B52C0FCF89138E316EC41010100008040C84EED5C0B40C0E825963FB9023540A0473966C74CB0C10101000080460B286C8CE86040547C368529C554C0067256521CF4FDC1010100008038D4A1B13BB63B409A4DFF7A41FE5040103B9D97583ED84101010000805E0F24DD23426640E64E8CF25E284140054A526D7CC7F6C1010100008055A7843D482766C0287C4342988E33C030F485CB230FEE410101000080C4531E88F76747C060BB58A54B334B40CA28B7E59D2D0042':::GEOGRAPHY::GEOGRAPHY)::STRING)) because error pq: invalid regular expression: error parsing regexp: missing argument to repetition operator: ``?``
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ('cac1:c342:845:7b22:5e35:1938:7ccb:a6af/97':::INET::INET << (seed._int8::INT8 + NULL::INET)::INET::INET)::BOOL
    changefeed_test.go:1148: Skipping predicate ('cac1:c342:845:7b22:5e35:1938:7ccb:a6af/97':::INET::INET << (seed._int8::INT8 + NULL::INET)::INET::INET)::BOOL because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (e'\x0fG\x13':::STRING::STRING % '7':::STRING::STRING)::BOOL
    changefeed_test.go:1148: Skipping predicate (e'\x0fG\x13':::STRING::STRING % '7':::STRING::STRING)::BOOL because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NOT EXISTS (SELECT seed._string AS "cOl_133" FROM d.public.seed AS t😳ab_47))
    changefeed_test.go:1148: Skipping predicate (NOT EXISTS (SELECT seed._string AS "cOl_133" FROM d.public.seed AS t😳ab_47)) because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (NULL AND (NOT (true AND false)))
    changefeed_test.go:1148: Skipping predicate (NULL AND (NOT (true AND false))) because it returned no rows
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT (seed._string SIMILAR TO seed._string)) OR seed._bool)
    changefeed_test.go:1141: Skipping query SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE ((NOT (seed._string SIMILAR TO seed._string)) OR seed._bool) because error pq: invalid regular expression: error parsing regexp: missing argument to repetition operator: ``?``
    changefeed_test.go:1136: SELECT array_to_string(IFNULL(array_agg(distinct rowid),'{}'),'|') FROM seed WHERE (seed._bool AND (seed._string LIKE seed._string))
    changefeed_test.go:1152: Expecting statement CREATE CHANGEFEED WITH schema_change_policy='stop' AS SELECT rowid FROM seed WHERE (seed._bool AND (seed._string LIKE seed._string)) to emit 39 events
    testfeed_test.go:301: creating external connection
    testfeed_test.go:304: ran create external connection
    changefeed_test.go:1169: LIKE regexp compilation failed: LIKE pattern must not end with escape character
    --- FAIL: TestChangefeedRandomExpressions/kafka (3.15s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=18</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #135269 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc]
- #134813 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc]
- #133049 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.3]
- #127642 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.2]
- #124738 ccl/changefeedccl: TestChangefeedRandomExpressions failed [A-cdc C-test-failure O-robot P-3 T-cdc branch-release-24.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedRandomExpressions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1158,7 +1158,6 @@ func TestChangefeedRandomExpressions(t *testing.T) {
 				}
 				continue
 			}
-			numNonTrivialTestRuns++
 			assertedPayloads := make([]string, len(expectedRowIDs))
 			for i, id := range expectedRowIDs {
 				assertedPayloads[i] = fmt.Sprintf(`seed: [%s]->{"rowid": %s}`, id, id)
@@ -1166,12 +1165,26 @@ func TestChangefeedRandomExpressions(t *testing.T) {
 			err = assertPayloadsBaseErr(context.Background(), seedFeed, assertedPayloads, false, false)
 			closeFeedIgnoreError(t, seedFeed)
 			if err != nil {
+				code := pgerror.GetPGCode(err)
+				// Skip errors that may come up during SQL execution. If the SQL query
+				// didn't fail with these errors, it's likely because the query was built in
+				// a way that did not have to execute on the row that caused the error, but
+				// the CDC query did.
+				switch code {
+				case pgcode.ConfigFile,
+					pgcode.DatetimeFieldOverflow,
+					pgcode.InvalidEscapeCharacter,
+					pgcode.InvalidEscapeSequence,
+					pgcode.InvalidParameterValue,
+					pgcode.InvalidRegularExpression:
+					t.Logf("Skipping statement %s because it encountered pgerror %s: %s", createStmt, code, err)
+					continue
+				}
 				t.Fatal(err)
 			}
+			numNonTrivialTestRuns++
 		}
-		if n > 100 {
-			require.Greater(t, numNonTrivialTestRuns, 1)
-		}
+		require.Greater(t, numNonTrivialTestRuns, 1)
 		t.Logf("%d predicates checked: all had the same result in SELECT and CHANGEFEED", numNonTrivialTestRuns)
 
 	}

